diff --git a/minecraft/server/packs/resources/ResourceMetadata.java b/minecraft/server/packs/resources/ResourceMetadata.java
index e47645c..6902d99 100644
--- a/minecraft/server/packs/resources/ResourceMetadata.java
+++ b/minecraft/server/packs/resources/ResourceMetadata.java
@@ -71,7 +71,7 @@ public interface ResourceMetadata {
         return resourcemetadata_builder.build();
     }
 
-    private default <T> void copySection(ResourceMetadata.Builder resourcemetadata_builder, MetadataSectionSerializer<T> metadatasectionserializer) {
+    private <T> void copySection(ResourceMetadata.Builder resourcemetadata_builder, MetadataSectionSerializer<T> metadatasectionserializer) {
         this.getSection(metadatasectionserializer).ifPresent((object) -> {
             resourcemetadata_builder.put(metadatasectionserializer, object);
         });
@@ -94,7 +94,7 @@ public interface ResourceMetadata {
             return immutablemap.isEmpty() ? ResourceMetadata.EMPTY : new ResourceMetadata() {
                 @Override
                 public <T> Optional<T> getSection(MetadataSectionSerializer<T> metadatasectionserializer) {
-                    return Optional.ofNullable(immutablemap.get(metadatasectionserializer));
+                    return (Optional<T>) Optional.ofNullable(immutablemap.get(metadatasectionserializer));
                 }
             };
         }
