diff --git a/minecraft/world/level/biome/OverworldBiomeBuilder.java b/minecraft/world/level/biome/OverworldBiomeBuilder.java
index c28be66..c9cfe15 100644
--- a/minecraft/world/level/biome/OverworldBiomeBuilder.java
+++ b/minecraft/world/level/biome/OverworldBiomeBuilder.java
@@ -450,7 +450,7 @@ public final class OverworldBiomeBuilder {
 
         for (int i = 0; i < aclimate_parameter.length; ++i) {
             if (d1 < (double) aclimate_parameter[i].max()) {
-                return i.makeConcatWithConstants < invokedynamic > (i);
+                return "" + i;
             }
         }
 
