diff --git a/minecraft/data/structures/NbtToSnbt.java b/minecraft/data/structures/NbtToSnbt.java
index e84c6a4..3e4508c 100644
--- a/minecraft/data/structures/NbtToSnbt.java
+++ b/minecraft/data/structures/NbtToSnbt.java
@@ -46,7 +46,7 @@ public class NbtToSnbt implements DataProvider {
 
             list.add(CompletableFuture.supplyAsync(() -> {
                 try {
-                    Stream stream = Files.walk(path1);
+                    Stream<Path> stream = Files.walk(path1);
 
                     CompletableFuture completablefuture;
 
