diff --git a/minecraft/world/level/storage/DimensionDataStorage.java b/minecraft/world/level/storage/DimensionDataStorage.java
index e9731d4..5a736ba 100644
--- a/minecraft/world/level/storage/DimensionDataStorage.java
+++ b/minecraft/world/level/storage/DimensionDataStorage.java
@@ -44,7 +44,7 @@ public class DimensionDataStorage {
         if (t0 != null) {
             return t0;
         } else {
-            T t1 = (SavedData) saveddata_factory.constructor().get();
+            T t1 = saveddata_factory.constructor().get();
 
             this.set(s, t1);
             return t1;
@@ -60,7 +60,7 @@ public class DimensionDataStorage {
             this.cache.put(s, saveddata);
         }
 
-        return saveddata;
+        return (T) saveddata;
     }
 
     @Nullable
@@ -71,7 +71,7 @@ public class DimensionDataStorage {
             if (file.exists()) {
                 CompoundTag compoundtag = this.readTagFromDisk(s, datafixtypes, SharedConstants.getCurrentVersion().getDataVersion().getVersion());
 
-                return (SavedData) function.apply(compoundtag.getCompound("data"));
+                return function.apply(compoundtag.getCompound("data"));
             }
         } catch (Exception exception) {
             DimensionDataStorage.LOGGER.error("Error loading saved data: {}", s, exception);
