diff --git a/minecraft/world/level/storage/LevelStorageSource.java b/minecraft/world/level/storage/LevelStorageSource.java
index 36f4827..03c4c72 100644
--- a/minecraft/world/level/storage/LevelStorageSource.java
+++ b/minecraft/world/level/storage/LevelStorageSource.java
@@ -148,7 +148,7 @@ public class LevelStorageSource {
     }
 
     public static WorldDataConfiguration readDataConfig(Dynamic<?> dynamic) {
-        DataResult dataresult = WorldDataConfiguration.CODEC.parse(dynamic);
+        DataResult<WorldDataConfiguration> dataresult = WorldDataConfiguration.CODEC.parse(dynamic);
         Logger logger = LevelStorageSource.LOGGER;
 
         Objects.requireNonNull(logger);
@@ -190,7 +190,7 @@ public class LevelStorageSource {
             throw new LevelStorageException(Component.translatable("selectWorld.load_folder_access"));
         } else {
             try {
-                Stream stream = Files.list(this.baseDir);
+                Stream<Path> stream = Files.list(this.baseDir);
 
                 LevelStorageSource.LevelCandidates levelstoragesource_levelcandidates;
 
@@ -528,7 +528,7 @@ public class LevelStorageSource {
         }
 
         public Path getLevelPath(LevelResource levelresource) {
-            Map map = this.resources;
+            Map<LevelResource, Path> map = this.resources;
             LevelStorageSource.LevelDirectory levelstoragesource_leveldirectory = this.levelDirectory;
 
             Objects.requireNonNull(this.levelDirectory);
