diff --git a/minecraft/world/entity/animal/allay/AllayAi.java b/minecraft/world/entity/animal/allay/AllayAi.java
index bd52c1d..d0f64c4 100644
--- a/minecraft/world/entity/animal/allay/AllayAi.java
+++ b/minecraft/world/entity/animal/allay/AllayAi.java
@@ -81,7 +81,7 @@ public class AllayAi {
         Optional<GlobalPos> optional = brain.getMemory(MemoryModuleType.LIKED_NOTEBLOCK_POSITION);
 
         if (optional.isEmpty()) {
-            brain.setMemory(MemoryModuleType.LIKED_NOTEBLOCK_POSITION, (Object) globalpos);
+            brain.setMemory(MemoryModuleType.LIKED_NOTEBLOCK_POSITION, globalpos);
             brain.setMemory(MemoryModuleType.LIKED_NOTEBLOCK_COOLDOWN_TICKS, (int) 600);
         } else if (((GlobalPos) optional.get()).equals(globalpos)) {
             brain.setMemory(MemoryModuleType.LIKED_NOTEBLOCK_COOLDOWN_TICKS, (int) 600);
