diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 45fca81..9550e4d 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -401,6 +401,23 @@ public class ServerGamePacketListenerImpl
                 double d2 = clampHorizontal(packet.getZ());
                 float f = Mth.wrapDegrees(packet.getYRot());
                 float f1 = Mth.wrapDegrees(packet.getXRot());
+                // dish
+                // Not sure if converting the passangers to a stream, then filtering out non players then converting it back to an ArrayList is good for performance.
+                java.util.ArrayList<net.ouja.api.entity.Player> players = rootVehicle.getPassengers().stream().map((e) -> {
+                    if (e instanceof Player) return new net.ouja.dish.entity.DishPlayer(((ServerPlayer) e));
+                    return null;
+                }).filter(Objects::nonNull).collect(Collectors.toCollection(java.util.ArrayList::new));
+                double xOld = clampHorizontal(this.vehicleLastGoodX);
+                double yOld = clampHorizontal(this.vehicleLastGoodY);
+                double zOld = clampHorizontal(this.vehicleLastGoodZ);
+                net.ouja.api.event.vehicle.VehicleMoveEvent event = new net.ouja.api.event.vehicle.VehicleMoveEvent(players, d, d1, d2, xOld, yOld, zOld);
+                net.ouja.dish.plugins.RegisteredEvents.callEvent(event);
+                if (event.isCancelled()) {
+                    rootVehicle.absMoveTo(xOld, yOld, zOld, f, f1);
+                    this.send(new ClientboundMoveVehiclePacket(rootVehicle));
+                    return;
+                }
+                // dish end
                 double d3 = d - this.vehicleFirstGoodX;
                 double d4 = d1 - this.vehicleFirstGoodY;
                 double d5 = d2 - this.vehicleFirstGoodZ;
