diff --git a/src/main/java/net/minecraft/server/network/ServerLoginPacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
index 1638819..19ffca9 100644
--- a/src/main/java/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
@@ -136,6 +136,13 @@ public class ServerLoginPacketListenerImpl implements ServerLoginPacketListener,
     private void verifyLoginAndFinishConnectionSetup(GameProfile profile) {
         PlayerList playerList = this.server.getPlayerList();
         Component component = playerList.canPlayerLogin(this.connection.getRemoteAddress(), profile);
+        // dish
+        net.ouja.api.event.player.PlayerLoginEvent event = new net.ouja.api.event.player.PlayerLoginEvent(new net.ouja.api.auth.GameProfile(profile.getName(), profile.getId()), component != null ? net.ouja.dish.network.chat.DishComponent.fromComponent(component) : null);
+        net.ouja.dish.plugins.RegisteredEvents.callEvent(event);
+        if (event.getErrorMessage() != null) {
+            component = net.ouja.dish.network.chat.DishComponent.toComponent(event.getErrorMessage()); // dish -- return custom error message
+        }
+        // dish end
         if (component != null) {
             this.disconnect(component);
         } else {
