diff --git a/minecraft/world/entity/SpawnPlacements.java b/minecraft/world/entity/SpawnPlacements.java
index fa96a1a..e9e4464 100644
--- a/minecraft/world/entity/SpawnPlacements.java
+++ b/minecraft/world/entity/SpawnPlacements.java
@@ -68,7 +68,7 @@ public class SpawnPlacements {
     public static <T extends Entity> boolean checkSpawnRules(EntityType<T> entitytype, ServerLevelAccessor serverlevelaccessor, MobSpawnType mobspawntype, BlockPos blockpos, RandomSource randomsource) {
         SpawnPlacements.Data spawnplacements_data = (SpawnPlacements.Data) SpawnPlacements.DATA_BY_TYPE.get(entitytype);
 
-        return spawnplacements_data == null || spawnplacements_data.predicate.test(entitytype, serverlevelaccessor, mobspawntype, blockpos, randomsource);
+        return spawnplacements_data == null || spawnplacements_data.predicate.test((EntityType)entitytype, serverlevelaccessor, mobspawntype, blockpos, randomsource);
     }
 
     static {
