diff --git a/minecraft/resources/ResourceKey.java b/minecraft/resources/ResourceKey.java
index 915d5bb..e9ce338 100644
--- a/minecraft/resources/ResourceKey.java
+++ b/minecraft/resources/ResourceKey.java
@@ -47,7 +47,7 @@ public class ResourceKey<T> {
     }
 
     public <E> Optional<ResourceKey<E>> cast(ResourceKey<? extends Registry<E>> resourcekey) {
-        return this.isFor(resourcekey) ? Optional.of(this) : Optional.empty();
+        return this.isFor(resourcekey) ? Optional.of((ResourceKey<E>) this) : Optional.empty();
     }
 
     public ResourceLocation location() {
