diff --git a/minecraft/world/entity/animal/Bee.java b/minecraft/world/entity/animal/Bee.java
index 218a761..e438161 100644
--- a/minecraft/world/entity/animal/Bee.java
+++ b/minecraft/world/entity/animal/Bee.java
@@ -1298,7 +1298,6 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
     private static class BeeBecomeAngryTargetGoal extends NearestAttackableTargetGoal<Player> {
 
         BeeBecomeAngryTargetGoal(Bee bee) {
-            Objects.requireNonNull(bee);
             super(bee, Player.class, 10, true, false, bee::isAngryAt);
         }
 
