diff --git a/minecraft/world/entity/animal/Panda.java b/minecraft/world/entity/animal/Panda.java
index a7aab79..935485f 100644
--- a/minecraft/world/entity/animal/Panda.java
+++ b/minecraft/world/entity/animal/Panda.java
@@ -873,10 +873,7 @@ public class Panda extends Animal {
         private final Panda panda;
 
         public PandaAvoidGoal(Panda panda, Class<T> oclass, float f, double d0, double d1) {
-            Predicate predicate = EntitySelector.NO_SPECTATORS;
-
-            Objects.requireNonNull(predicate);
-            super(panda, oclass, f, d0, d1, predicate::test);
+            super(panda, oclass, f, d0, d1, EntitySelector.NO_SPECTATORS::test);
             this.panda = panda;
         }
 
