diff --git a/minecraft/world/entity/animal/Sheep.java b/minecraft/world/entity/animal/Sheep.java
index 6c78f61..c73b701 100644
--- a/minecraft/world/entity/animal/Sheep.java
+++ b/minecraft/world/entity/animal/Sheep.java
@@ -52,6 +52,7 @@ import net.minecraft.world.inventory.MenuType;
 import net.minecraft.world.inventory.TransientCraftingContainer;
 import net.minecraft.world.item.DyeColor;
 import net.minecraft.world.item.DyeItem;
+import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
 import net.minecraft.world.item.crafting.CraftingRecipe;
@@ -379,7 +380,7 @@ public class Sheep extends Animal implements Shearable {
         DyeColor dyecolor = ((Sheep) animal).getColor();
         DyeColor dyecolor1 = ((Sheep) animal1).getColor();
         CraftingContainer craftingcontainer = makeContainer(dyecolor, dyecolor1);
-        Optional optional = this.level().getRecipeManager().getRecipeFor(RecipeType.CRAFTING, craftingcontainer, this.level()).map((recipeholder) -> {
+        Optional<Item> optional = this.level().getRecipeManager().getRecipeFor(RecipeType.CRAFTING, craftingcontainer, this.level()).map((recipeholder) -> {
             return ((CraftingRecipe) recipeholder.value()).assemble(craftingcontainer, this.level().registryAccess());
         }).map(ItemStack::getItem);
 
