diff --git a/minecraft/advancements/critereon/LightningStrikeTrigger.java b/minecraft/advancements/critereon/LightningStrikeTrigger.java
index ec7d5ec..01ea6df 100644
--- a/minecraft/advancements/critereon/LightningStrikeTrigger.java
+++ b/minecraft/advancements/critereon/LightningStrikeTrigger.java
@@ -56,7 +56,7 @@ public class LightningStrikeTrigger extends SimpleCriterionTrigger<LightningStri
                 return false;
             } else {
                 if (this.bystander.isPresent()) {
-                    Stream stream = list.stream();
+                    Stream<LootContext> stream = list.stream();
                     ContextAwarePredicate contextawarepredicate = (ContextAwarePredicate) this.bystander.get();
 
                     Objects.requireNonNull(contextawarepredicate);
