diff --git a/minecraft/advancements/AdvancementRequirements.java b/minecraft/advancements/AdvancementRequirements.java
index 779fa58..24c0475 100644
--- a/minecraft/advancements/AdvancementRequirements.java
+++ b/minecraft/advancements/AdvancementRequirements.java
@@ -164,7 +164,7 @@ public record AdvancementRequirements(String[][] requirements) {
         for (int j = 0; j < i; ++j) {
             String[] astring1 = astring[j];
             JsonArray jsonarray1 = new JsonArray();
-            Stream stream = Arrays.stream(astring1);
+            Stream<String> stream = Arrays.stream(astring1);
 
             Objects.requireNonNull(jsonarray1);
             stream.forEach(jsonarray1::add);
