diff --git a/minecraft/world/level/levelgen/structure/pools/StructureTemplatePool.java b/minecraft/world/level/levelgen/structure/pools/StructureTemplatePool.java
index 56c8296..2f65969 100644
--- a/minecraft/world/level/levelgen/structure/pools/StructureTemplatePool.java
+++ b/minecraft/world/level/levelgen/structure/pools/StructureTemplatePool.java
@@ -30,7 +30,7 @@ public class StructureTemplatePool {
     private static final int SIZE_UNSET = Integer.MIN_VALUE;
     private static final MutableObject<Codec<Holder<StructureTemplatePool>>> CODEC_REFERENCE = new MutableObject();
     public static final Codec<StructureTemplatePool> DIRECT_CODEC = RecordCodecBuilder.create((instance) -> {
-        MutableObject mutableobject = StructureTemplatePool.CODEC_REFERENCE;
+        MutableObject<Codec<Holder<StructureTemplatePool>>> mutableobject = StructureTemplatePool.CODEC_REFERENCE;
 
         Objects.requireNonNull(mutableobject);
         return instance.group(ExtraCodecs.lazyInitializedCodec(mutableobject::getValue).fieldOf("fallback").forGetter(StructureTemplatePool::getFallback), Codec.mapPair(StructurePoolElement.CODEC.fieldOf("element"), Codec.intRange(1, 150).fieldOf("weight")).codec().listOf().fieldOf("elements").forGetter((structuretemplatepool) -> {
@@ -108,11 +108,7 @@ public class StructureTemplatePool {
     }
 
     static {
-        RegistryFileCodec registryfilecodec = RegistryFileCodec.create(Registries.TEMPLATE_POOL, StructureTemplatePool.DIRECT_CODEC);
-        MutableObject mutableobject = StructureTemplatePool.CODEC_REFERENCE;
-
-        Objects.requireNonNull(mutableobject);
-        CODEC = (Codec) Util.make(registryfilecodec, mutableobject::setValue);
+        CODEC = (Codec) Util.make(RegistryFileCodec.create(Registries.TEMPLATE_POOL, StructureTemplatePool.DIRECT_CODEC), StructureTemplatePool.CODEC_REFERENCE::setValue);
     }
 
     public static enum Projection implements StringRepresentable {
