diff --git a/minecraft/world/level/block/piston/PistonMovingBlockEntity.java b/minecraft/world/level/block/piston/PistonMovingBlockEntity.java
index cb73c5e..2900f5f 100644
--- a/minecraft/world/level/block/piston/PistonMovingBlockEntity.java
+++ b/minecraft/world/level/block/piston/PistonMovingBlockEntity.java
@@ -178,7 +178,7 @@ public class PistonMovingBlockEntity extends BlockEntity {
     private static void moveEntityByPiston(Direction direction, Entity entity, double d0, Direction direction1) {
         PistonMovingBlockEntity.NOCLIP.set(direction);
         entity.move(MoverType.PISTON, new Vec3(d0 * (double) direction1.getStepX(), d0 * (double) direction1.getStepY(), d0 * (double) direction1.getStepZ()));
-        PistonMovingBlockEntity.NOCLIP.set((Object) null);
+        PistonMovingBlockEntity.NOCLIP.set(null);
     }
 
     private static void moveStuckEntities(Level level, BlockPos blockpos, float f, PistonMovingBlockEntity pistonmovingblockentity) {
