diff --git a/minecraft/util/profiling/jfr/parse/JfrStatsResult.java b/minecraft/util/profiling/jfr/parse/JfrStatsResult.java
index 30e70d8..c42bae1 100644
--- a/minecraft/util/profiling/jfr/parse/JfrStatsResult.java
+++ b/minecraft/util/profiling/jfr/parse/JfrStatsResult.java
@@ -25,8 +25,8 @@ public record JfrStatsResult(Instant recordingStarted, Instant recordingEnded, D
         Map<ChunkStatus, List<ChunkGenStat>> map = (Map) this.chunkGenStats.stream().collect(Collectors.groupingBy(ChunkGenStat::status));
 
         return map.entrySet().stream().map((entry) -> {
-            return Pair.of((ChunkStatus) entry.getKey(), TimedStatSummary.summary((List) entry.getValue()));
-        }).sorted(Comparator.comparing((pair) -> {
+            return Pair.of((ChunkStatus) entry.getKey(), TimedStatSummary.summary( entry.getValue()));
+        }).sorted(Comparator.<Pair<ChunkStatus, TimedStatSummary<ChunkGenStat>>, Duration>comparing((pair) -> {
             return ((TimedStatSummary) pair.getSecond()).totalDuration();
         }).reversed()).toList();
     }
