diff --git a/minecraft/util/InclusiveRange.java b/minecraft/util/InclusiveRange.java
index 68f94e3..abfd38a 100644
--- a/minecraft/util/InclusiveRange.java
+++ b/minecraft/util/InclusiveRange.java
@@ -7,12 +7,12 @@ public record InclusiveRange<T extends Comparable<T>> (T minInclusive, T maxIncl
 
     public static final Codec<InclusiveRange<Integer>> INT = codec(Codec.INT);
 
-    public InclusiveRange(T t0, T t1) {
-        if (t0.compareTo(t1) > 0) {
+    public InclusiveRange(T minInclusive, T maxInclusive) {
+        if (minInclusive.compareTo(maxInclusive) > 0) {
             throw new IllegalArgumentException("min_inclusive must be less than or equal to max_inclusive");
         } else {
-            this.minInclusive = t0;
-            this.maxInclusive = t1;
+            this.minInclusive = minInclusive;
+            this.maxInclusive = maxInclusive;
         }
     }
 
