diff --git a/minecraft/util/LowerCaseEnumTypeAdapterFactory.java b/minecraft/util/LowerCaseEnumTypeAdapterFactory.java
index 2601bbb..923335c 100644
--- a/minecraft/util/LowerCaseEnumTypeAdapterFactory.java
+++ b/minecraft/util/LowerCaseEnumTypeAdapterFactory.java
@@ -19,7 +19,7 @@ public class LowerCaseEnumTypeAdapterFactory implements TypeAdapterFactory {
 
     @Nullable
     public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> typetoken) {
-        Class<T> oclass = typetoken.getRawType();
+        Class<T> oclass = (Class<T>) typetoken.getRawType();
 
         if (!oclass.isEnum()) {
             return null;
@@ -29,7 +29,7 @@ public class LowerCaseEnumTypeAdapterFactory implements TypeAdapterFactory {
             int i = aobject.length;
 
             for (int j = 0; j < i; ++j) {
-                T t0 = aobject[j];
+                T t0 = (T) aobject[j];
 
                 map.put(this.toLowercase(t0), t0);
             }
