diff --git a/minecraft/util/profiling/FilledProfileResults.java b/minecraft/util/profiling/FilledProfileResults.java
index d0fee4d..8019a70 100644
--- a/minecraft/util/profiling/FilledProfileResults.java
+++ b/minecraft/util/profiling/FilledProfileResults.java
@@ -48,7 +48,7 @@ public class FilledProfileResults implements ProfileResults {
         }
     };
     private static final Splitter SPLITTER = Splitter.on('\u001e');
-    private static final Comparator<Entry<String, FilledProfileResults.CounterCollector>> COUNTER_ENTRY_COMPARATOR = Entry.comparingByValue(Comparator.comparingLong((filledprofileresults_countercollector) -> {
+    private static final Comparator<Entry<String, FilledProfileResults.CounterCollector>> COUNTER_ENTRY_COMPARATOR = Entry.<String, CounterCollector>comparingByValue(Comparator.comparingLong((filledprofileresults_countercollector) -> {
         return filledprofileresults_countercollector.totalValue;
     })).reversed();
     private final Map<String, ? extends ProfilerPathEntry> entries;
