diff --git a/minecraft/world/entity/animal/goat/Goat.java b/minecraft/world/entity/animal/goat/Goat.java
index 17dba9e..4d8fec9 100644
--- a/minecraft/world/entity/animal/goat/Goat.java
+++ b/minecraft/world/entity/animal/goat/Goat.java
@@ -182,7 +182,7 @@ public class Goat extends Animal {
 
     @Override
     public Brain<Goat> getBrain() {
-        return super.getBrain();
+        return (Brain<Goat>) super.getBrain();
     }
 
     @Override
