diff --git a/minecraft/world/level/levelgen/structure/pools/SinglePoolElement.java b/minecraft/world/level/levelgen/structure/pools/SinglePoolElement.java
index da0e8b8..e63d442 100644
--- a/minecraft/world/level/levelgen/structure/pools/SinglePoolElement.java
+++ b/minecraft/world/level/levelgen/structure/pools/SinglePoolElement.java
@@ -122,7 +122,7 @@ public class SinglePoolElement extends StructurePoolElement {
 
     @VisibleForTesting
     static void sortBySelectionPriority(List<StructureTemplate.StructureBlockInfo> list) {
-        list.sort(Comparator.comparingInt((structuretemplate_structureblockinfo) -> {
+        list.sort(Comparator.<StructureTemplate.StructureBlockInfo>comparingInt((structuretemplate_structureblockinfo) -> {
             return (Integer) Optionull.mapOrDefault(structuretemplate_structureblockinfo.nbt(), (compoundtag) -> {
                 return compoundtag.getInt("selection_priority");
             }, 0);
