diff --git a/minecraft/world/level/material/Fluids.java b/minecraft/world/level/material/Fluids.java
index 3e7c413..3950745 100644
--- a/minecraft/world/level/material/Fluids.java
+++ b/minecraft/world/level/material/Fluids.java
@@ -16,7 +16,7 @@ public class Fluids {
     public Fluids() {}
 
     private static <T extends Fluid> T register(String s, T t0) {
-        return (Fluid) Registry.register(BuiltInRegistries.FLUID, s, t0);
+        return Registry.register(BuiltInRegistries.FLUID, s, t0);
     }
 
     static {
