diff --git a/minecraft/network/chat/Style.java b/minecraft/network/chat/Style.java
index e5b3ab8..7444a93 100644
--- a/minecraft/network/chat/Style.java
+++ b/minecraft/network/chat/Style.java
@@ -64,7 +64,7 @@ public class Style {
     final ResourceLocation font;
 
     private static Style create(Optional<TextColor> optional, Optional<Boolean> optional1, Optional<Boolean> optional2, Optional<Boolean> optional3, Optional<Boolean> optional4, Optional<Boolean> optional5, Optional<String> optional6, Optional<ResourceLocation> optional7) {
-        return new Style((TextColor) optional.orElse((Object) null), (Boolean) optional1.orElse((Object) null), (Boolean) optional2.orElse((Object) null), (Boolean) optional3.orElse((Object) null), (Boolean) optional4.orElse((Object) null), (Boolean) optional5.orElse((Object) null), (ClickEvent) null, (HoverEvent) null, (String) optional6.orElse((Object) null), (ResourceLocation) optional7.orElse((Object) null));
+        return new Style((TextColor) optional.orElse(null), (Boolean) optional1.orElse(null), (Boolean) optional2.orElse(null), (Boolean) optional3.orElse(null), (Boolean) optional4.orElse(null), (Boolean) optional5.orElse(null), (ClickEvent) null, (HoverEvent) null, (String) optional6.orElse(null), (ResourceLocation) optional7.orElse(null));
     }
 
     Style(@Nullable TextColor textcolor, @Nullable Boolean obool, @Nullable Boolean obool1, @Nullable Boolean obool2, @Nullable Boolean obool3, @Nullable Boolean obool4, @Nullable ClickEvent clickevent, @Nullable HoverEvent hoverevent, @Nullable String s, @Nullable ResourceLocation resourcelocation) {
@@ -293,11 +293,11 @@ public class Style {
     public String toString() {
         final StringBuilder stringbuilder = new StringBuilder("{");
 
-        class 1Collector {
+        class Collector {
 
             private boolean isNotFirst;
 
-            _Collector/* $FF was: 1Collector*/() {
+            Collector/* $FF was: 1Collector*/() {
 }
 
             private void prependSeparator() {
@@ -331,18 +331,18 @@ public class Style {
             }
         }
 
-        1Collector 1collector = new 1Collector();
-
-        1collector.addValueString("color", this.color);
-        1collector.addFlagString("bold", this.bold);
-        1collector.addFlagString("italic", this.italic);
-        1collector.addFlagString("underlined", this.underlined);
-        1collector.addFlagString("strikethrough", this.strikethrough);
-        1collector.addFlagString("obfuscated", this.obfuscated);
-        1collector.addValueString("clickEvent", this.clickEvent);
-        1collector.addValueString("hoverEvent", this.hoverEvent);
-        1collector.addValueString("insertion", this.insertion);
-        1collector.addValueString("font", this.font);
+        Collector collector = new Collector();
+
+        collector.addValueString("color", this.color);
+        collector.addFlagString("bold", this.bold);
+        collector.addFlagString("italic", this.italic);
+        collector.addFlagString("underlined", this.underlined);
+        collector.addFlagString("strikethrough", this.strikethrough);
+        collector.addFlagString("obfuscated", this.obfuscated);
+        collector.addValueString("clickEvent", this.clickEvent);
+        collector.addValueString("hoverEvent", this.hoverEvent);
+        collector.addValueString("insertion", this.insertion);
+        collector.addValueString("font", this.font);
         stringbuilder.append("}");
         return stringbuilder.toString();
     }
