diff --git a/minecraft/util/ByIdMap.java b/minecraft/util/ByIdMap.java
index 6e6d233..d7a6fa6 100644
--- a/minecraft/util/ByIdMap.java
+++ b/minecraft/util/ByIdMap.java
@@ -20,7 +20,7 @@ public class ByIdMap {
             int i = at.length;
 
             for (int j = 0; j < i; ++j) {
-                T t0 = aobject[j];
+                T t0 = (T) aobject[j];
                 int k = tointfunction.applyAsInt(t0);
                 T t1 = int2objectmap.put(k, t0);
 
@@ -47,14 +47,14 @@ public class ByIdMap {
         if (i == 0) {
             throw new IllegalArgumentException("Empty value list");
         } else {
-            T[] at1 = (Object[]) at.clone();
+            T[] at1 = at.clone();
 
             Arrays.fill(at1, (Object) null);
             Object[] aobject = at;
             int j = at.length;
 
             for (int k = 0; k < j; ++k) {
-                T t0 = aobject[k];
+                T t0 = (T) aobject[k];
                 int l = tointfunction.applyAsInt(t0);
 
                 if (l < 0 || l >= i) {
