diff --git a/minecraft/network/chat/contents/TranslatableContents.java b/minecraft/network/chat/contents/TranslatableContents.java
index ffcda51..b8b0667 100644
--- a/minecraft/network/chat/contents/TranslatableContents.java
+++ b/minecraft/network/chat/contents/TranslatableContents.java
@@ -94,7 +94,7 @@ public class TranslatableContents implements ComponentContents {
     }
 
     private static TranslatableContents create(String s, Optional<String> optional, Optional<List<Object>> optional1) {
-        return new TranslatableContents(s, (String) optional.orElse((Object) null), adjustArgs(optional1));
+        return new TranslatableContents(s, (String) optional.orElse(null), adjustArgs(optional1));
     }
 
     public TranslatableContents(String s, @Nullable String s1, Object[] aobject) {
