diff --git a/minecraft/server/commands/ForceLoadCommand.java b/minecraft/server/commands/ForceLoadCommand.java
index 4713746..8e6b339 100644
--- a/minecraft/server/commands/ForceLoadCommand.java
+++ b/minecraft/server/commands/ForceLoadCommand.java
@@ -146,16 +146,17 @@ public class ForceLoadCommand {
                 if (j2 == 0) {
                     throw (flag ? ForceLoadCommand.ERROR_ALL_ADDED : ForceLoadCommand.ERROR_NONE_REMOVED).create();
                 } else {
+                    ChunkPos finalChunkpos = chunkpos;
                     if (j2 == 1) {
                         commandsourcestack.sendSuccess(() -> {
-                            return Component.translatable("commands.forceload." + (flag ? "added" : "removed") + ".single", chunkpos, resourcekey.location());
+                            return Component.translatable("commands.forceload." + (flag ? "added" : "removed") + ".single", finalChunkpos, resourcekey.location());
                         }, true);
                     } else {
                         ChunkPos chunkpos1 = new ChunkPos(i1, j1);
                         ChunkPos chunkpos2 = new ChunkPos(k1, l1);
 
                         commandsourcestack.sendSuccess(() -> {
-                            return Component.translatable("commands.forceload." + (flag ? "added" : "removed") + ".multiple", chunkpos, resourcekey.location(), chunkpos1, chunkpos2);
+                            return Component.translatable("commands.forceload." + (flag ? "added" : "removed") + ".multiple", finalChunkpos, resourcekey.location(), chunkpos1, chunkpos2);
                         }, true);
                     }
 
