diff --git a/minecraft/world/level/block/RespawnAnchorBlock.java b/minecraft/world/level/block/RespawnAnchorBlock.java
index a1c0beb..eabc0fa 100644
--- a/minecraft/world/level/block/RespawnAnchorBlock.java
+++ b/minecraft/world/level/block/RespawnAnchorBlock.java
@@ -127,7 +127,7 @@ public class RespawnAnchorBlock extends Block {
 
     private void explode(BlockState blockstate, Level level, final BlockPos blockpos) {
         level.removeBlock(blockpos, false);
-        Stream stream = Direction.Plane.HORIZONTAL.stream();
+        Stream<Direction> stream = Direction.Plane.HORIZONTAL.stream();
 
         Objects.requireNonNull(blockpos);
         boolean flag = stream.map(blockpos::relative).anyMatch((blockpos1) -> {
