diff --git a/minecraft/world/entity/animal/frog/Frog.java b/minecraft/world/entity/animal/frog/Frog.java
index 636b912..4a86a1e 100644
--- a/minecraft/world/entity/animal/frog/Frog.java
+++ b/minecraft/world/entity/animal/frog/Frog.java
@@ -100,7 +100,7 @@ public class Frog extends Animal implements VariantHolder<FrogVariant> {
 
     @Override
     public Brain<Frog> getBrain() {
-        return super.getBrain();
+        return (Brain<Frog>) super.getBrain();
     }
 
     @Override
@@ -245,7 +245,7 @@ public class Frog extends Animal implements VariantHolder<FrogVariant> {
     @Override
     public void spawnChildFromBreeding(ServerLevel serverlevel, Animal animal) {
         this.finalizeSpawnChildFromBreeding(serverlevel, animal, (AgeableMob) null);
-        this.getBrain().setMemory(MemoryModuleType.IS_PREGNANT, (Object) Unit.INSTANCE);
+        this.getBrain().setMemory(MemoryModuleType.IS_PREGNANT, Unit.INSTANCE);
     }
 
     @Override
