diff --git a/src/main/java/net/minecraft/world/item/BlockItem.java b/src/main/java/net/minecraft/world/item/BlockItem.java
index c039f42..231c6ad 100644
--- a/src/main/java/net/minecraft/world/item/BlockItem.java
+++ b/src/main/java/net/minecraft/world/item/BlockItem.java
@@ -35,6 +35,14 @@ import net.minecraft.world.level.block.state.properties.Property;
 import net.minecraft.world.level.gameevent.GameEvent;
 import net.minecraft.world.phys.shapes.CollisionContext;
 
+// dish
+import net.minecraft.server.level.ServerLevel;
+import net.ouja.dish.entity.DishPlayer;
+import net.ouja.dish.world.DishLevel;
+import net.ouja.api.event.player.PlayerPlaceBlockEvent;
+import net.ouja.dish.plugins.RegisteredEvents;
+// dish end
+
 public class BlockItem extends Item {
 
     public static final String BLOCK_ENTITY_TAG = "BlockEntityTag";
@@ -68,7 +76,15 @@ public class BlockItem extends Item {
             return InteractionResult.FAIL;
         } else {
             BlockPlaceContext blockplacecontext1 = this.updatePlacementContext(blockplacecontext);
-
+            // dish
+            DishPlayer dishPlayer = new DishPlayer((ServerPlayer)blockplacecontext1.getPlayer());
+            net.ouja.api.world.level.block.Block dishBlock = this.getBlock().toDish(blockplacecontext1.getClickedPos(), blockplacecontext1.getLevel());
+            DishLevel dishLevel = new DishLevel((ServerLevel)blockplacecontext1.getLevel());
+            net.ouja.api.world.level.block.BlockPos dishBlockPos = new net.ouja.api.world.level.block.BlockPos(blockplacecontext1.getClickedPos().getX(), blockplacecontext1.getClickedPos().getY(), blockplacecontext1.getClickedPos().getZ(), dishLevel);
+            PlayerPlaceBlockEvent event = new PlayerPlaceBlockEvent(dishPlayer, dishBlock, dishBlockPos);
+            RegisteredEvents.callEvent(event);
+            if (event.isCancelled()) return InteractionResult.FAIL;
+            // dish end
             if (blockplacecontext1 == null) {
                 return InteractionResult.FAIL;
             } else {
