diff --git a/minecraft/world/level/storage/loot/ValidationContext.java b/minecraft/world/level/storage/loot/ValidationContext.java
index 94e5b38..ac1f924 100644
--- a/minecraft/world/level/storage/loot/ValidationContext.java
+++ b/minecraft/world/level/storage/loot/ValidationContext.java
@@ -28,7 +28,7 @@ public class ValidationContext {
     }
 
     public ValidationContext enterElement(String s, LootDataId<?> lootdataid) {
-        ImmutableSet<LootDataId<?>> immutableset = ImmutableSet.builder().addAll(this.visitedElements).add(lootdataid).build();
+        ImmutableSet<LootDataId<?>> immutableset = ImmutableSet.<LootDataId<?>>builder().addAll(this.visitedElements).add(lootdataid).build();
 
         return new ValidationContext(this.reporter.forChild(s), this.params, this.resolver, immutableset);
     }
