diff --git a/minecraft/advancements/AdvancementTree.java b/minecraft/advancements/AdvancementTree.java
index 32ac672..3ab7ce7 100644
--- a/minecraft/advancements/AdvancementTree.java
+++ b/minecraft/advancements/AdvancementTree.java
@@ -68,7 +68,7 @@ public class AdvancementTree {
     }
 
     public void addAll(Collection<AdvancementHolder> collection) {
-        ArrayList arraylist = new ArrayList(collection);
+        ArrayList<AdvancementHolder> arraylist = new ArrayList(collection);
 
         while (!arraylist.isEmpty()) {
             if (!arraylist.removeIf(this::tryInsert)) {
@@ -85,7 +85,7 @@ public class AdvancementTree {
         Map map = this.nodes;
 
         Objects.requireNonNull(this.nodes);
-        AdvancementNode advancementnode = (AdvancementNode) optional.map(map::get).orElse((Object) null);
+        AdvancementNode advancementnode = (AdvancementNode) optional.map(map::get).orElse(null);
 
         if (advancementnode == null && optional.isPresent()) {
             return false;
