diff --git a/minecraft/advancements/Advancement.java b/minecraft/advancements/Advancement.java
index a777c61..83730c9 100644
--- a/minecraft/advancements/Advancement.java
+++ b/minecraft/advancements/Advancement.java
@@ -25,7 +25,7 @@ import net.minecraft.world.level.storage.loot.LootDataResolver;
 
 public record Advancement(Optional<ResourceLocation> parent, Optional<DisplayInfo> display, AdvancementRewards rewards, Map<String, Criterion<?>> criteria, AdvancementRequirements requirements, boolean sendsTelemetryEvent, Optional<Component> name) {
 
-    private static final Codec<Map<String, Criterion<?>>> CRITERIA_CODEC = ExtraCodecs.validate((Codec) Codec.unboundedMap(Codec.STRING, Criterion.CODEC), (map) -> {
+    private static final Codec<Map<String, Criterion<?>>> CRITERIA_CODEC = ExtraCodecs.validate(Codec.unboundedMap(Codec.STRING, Criterion.CODEC), (map) -> {
         return map.isEmpty() ? DataResult.error(() -> {
             return "Advancement criteria cannot be empty";
         }) : DataResult.success(map);
