diff --git a/minecraft/advancements/CriteriaTriggers.java b/minecraft/advancements/CriteriaTriggers.java
index ce6ba61..818c2ff 100644
--- a/minecraft/advancements/CriteriaTriggers.java
+++ b/minecraft/advancements/CriteriaTriggers.java
@@ -103,7 +103,7 @@ public class CriteriaTriggers {
     public CriteriaTriggers() {}
 
     private static <T extends CriterionTrigger<?>> T register(String s, T t0) {
-        return (CriterionTrigger) Registry.register(BuiltInRegistries.TRIGGER_TYPES, s, t0);
+        return Registry.register(BuiltInRegistries.TRIGGER_TYPES, s, t0);
     }
 
     public static CriterionTrigger<?> bootstrap(Registry<CriterionTrigger<?>> registry) {
