diff --git a/minecraft/advancements/DisplayInfo.java b/minecraft/advancements/DisplayInfo.java
index 036464e..0e75719 100644
--- a/minecraft/advancements/DisplayInfo.java
+++ b/minecraft/advancements/DisplayInfo.java
@@ -103,7 +103,7 @@ public class DisplayInfo {
         }
 
         friendlybytebuf.writeInt(i);
-        Optional optional = this.background;
+        Optional<ResourceLocation> optional = this.background;
 
         Objects.requireNonNull(friendlybytebuf);
         optional.ifPresent(friendlybytebuf::writeResourceLocation);
