diff --git a/minecraft/world/level/levelgen/structure/templatesystem/RuleTestType.java b/minecraft/world/level/levelgen/structure/templatesystem/RuleTestType.java
index 73fa5f8..af88364 100644
--- a/minecraft/world/level/levelgen/structure/templatesystem/RuleTestType.java
+++ b/minecraft/world/level/levelgen/structure/templatesystem/RuleTestType.java
@@ -16,7 +16,7 @@ public interface RuleTestType<P extends RuleTest> {
     Codec<P> codec();
 
     static <P extends RuleTest> RuleTestType<P> register(String s, Codec<P> codec) {
-        return (RuleTestType) Registry.register(BuiltInRegistries.RULE_TEST, s, () -> {
+        return Registry.register(BuiltInRegistries.RULE_TEST, s, () -> {
             return codec;
         });
     }
