diff --git a/minecraft/world/level/levelgen/structure/templatesystem/StructureProcessorType.java b/minecraft/world/level/levelgen/structure/templatesystem/StructureProcessorType.java
index 78ff130..de20adb 100644
--- a/minecraft/world/level/levelgen/structure/templatesystem/StructureProcessorType.java
+++ b/minecraft/world/level/levelgen/structure/templatesystem/StructureProcessorType.java
@@ -29,7 +29,7 @@ public interface StructureProcessorType<P extends StructureProcessor> {
     Codec<P> codec();
 
     static <P extends StructureProcessor> StructureProcessorType<P> register(String s, Codec<P> codec) {
-        return (StructureProcessorType) Registry.register(BuiltInRegistries.STRUCTURE_PROCESSOR, s, () -> {
+        return Registry.register(BuiltInRegistries.STRUCTURE_PROCESSOR, s, () -> {
             return codec;
         });
     }
