diff --git a/minecraft/data/Main.java b/minecraft/data/Main.java
index c4d8604..d02fb2f 100644
--- a/minecraft/data/Main.java
+++ b/minecraft/data/Main.java
@@ -98,7 +98,7 @@ public class Main {
 
     private static <T extends DataProvider> DataProvider.Factory<T> bindRegistries(BiFunction<PackOutput, CompletableFuture<HolderLookup.Provider>, T> bifunction, CompletableFuture<HolderLookup.Provider> completablefuture) {
         return (packoutput) -> {
-            return (DataProvider) bifunction.apply(packoutput, completablefuture);
+            return bifunction.apply(packoutput, completablefuture);
         };
     }
 
