diff --git a/minecraft/util/datafix/fixes/FixProjectileStoredItem.java b/minecraft/util/datafix/fixes/FixProjectileStoredItem.java
index 2cd78a5..e0e1e0e 100644
--- a/minecraft/util/datafix/fixes/FixProjectileStoredItem.java
+++ b/minecraft/util/datafix/fixes/FixProjectileStoredItem.java
@@ -55,7 +55,7 @@ public class FixProjectileStoredItem extends DataFix {
 
         return (typed) -> {
             return typed.updateTyped(opticfinder, type1, (typed1) -> {
-                return fixprojectilestoreditem_subfixer.fix(typed1, type1);
+                return fixprojectilestoreditem_subfixer.fix(typed1, (Type)type1);
             });
         };
     }
