diff --git a/minecraft/server/packs/VanillaPackResources.java b/minecraft/server/packs/VanillaPackResources.java
index de9de18..43226e1 100644
--- a/minecraft/server/packs/VanillaPackResources.java
+++ b/minecraft/server/packs/VanillaPackResources.java
@@ -157,7 +157,7 @@ public class VanillaPackResources implements PackResources {
 
                 label46:
                 {
-                    Object object;
+                    T object;
 
                     try {
                         T t0 = AbstractPackResources.getMetadataFromStream(metadatasectionserializer, inputstream);
