diff --git a/minecraft/world/entity/monster/warden/AngerManagement.java b/minecraft/world/entity/monster/warden/AngerManagement.java
index 9d27002..a8843cb 100644
--- a/minecraft/world/entity/monster/warden/AngerManagement.java
+++ b/minecraft/world/entity/monster/warden/AngerManagement.java
@@ -175,7 +175,7 @@ public class AngerManagement {
 
     @Nullable
     private Entity getTopSuspect() {
-        return (Entity) this.suspects.stream().filter(this.filter).findFirst().orElse((Object) null);
+        return (Entity) this.suspects.stream().filter(this.filter).findFirst().orElse(null);
     }
 
     public int getActiveAnger(@Nullable Entity entity) {
