diff --git a/minecraft/data/tags/UpdateOneTwentyOneEntityTypeTagsProvider.java b/minecraft/data/tags/UpdateOneTwentyOneEntityTypeTagsProvider.java
index b535059..6436cf9 100644
--- a/minecraft/data/tags/UpdateOneTwentyOneEntityTypeTagsProvider.java
+++ b/minecraft/data/tags/UpdateOneTwentyOneEntityTypeTagsProvider.java
@@ -17,8 +17,8 @@ public class UpdateOneTwentyOneEntityTypeTagsProvider extends IntrinsicHolderTag
 
     @Override
     protected void addTags(HolderLookup.Provider holderlookup_provider) {
-        this.tag(EntityTypeTags.FALL_DAMAGE_IMMUNE).add((Object) EntityType.BREEZE);
-        this.tag(EntityTypeTags.DEFLECTS_ARROWS).add((Object) EntityType.BREEZE);
-        this.tag(EntityTypeTags.DEFLECTS_TRIDENTS).add((Object) EntityType.BREEZE);
+        this.tag(EntityTypeTags.FALL_DAMAGE_IMMUNE).add(EntityType.BREEZE);
+        this.tag(EntityTypeTags.DEFLECTS_ARROWS).add(EntityType.BREEZE);
+        this.tag(EntityTypeTags.DEFLECTS_TRIDENTS).add(EntityType.BREEZE);
     }
 }
