diff --git a/minecraft/world/entity/animal/Cat.java b/minecraft/world/entity/animal/Cat.java
index 7128672..3a4680c 100644
--- a/minecraft/world/entity/animal/Cat.java
+++ b/minecraft/world/entity/animal/Cat.java
@@ -648,10 +648,7 @@ public class Cat extends TamableAnimal implements VariantHolder<CatVariant> {
         private final Cat cat;
 
         public CatAvoidEntityGoal(Cat cat, Class<T> oclass, float f, double d0, double d1) {
-            Predicate predicate = EntitySelector.NO_CREATIVE_OR_SPECTATOR;
-
-            Objects.requireNonNull(predicate);
-            super(cat, oclass, f, d0, d1, predicate::test);
+            super(cat, oclass, f, d0, d1, EntitySelector.NO_CREATIVE_OR_SPECTATOR::test);
             this.cat = cat;
         }
 
