diff --git a/minecraft/world/entity/animal/Ocelot.java b/minecraft/world/entity/animal/Ocelot.java
index 21bd583..495a8da 100644
--- a/minecraft/world/entity/animal/Ocelot.java
+++ b/minecraft/world/entity/animal/Ocelot.java
@@ -315,10 +315,7 @@ public class Ocelot extends Animal {
         private final Ocelot ocelot;
 
         public OcelotAvoidEntityGoal(Ocelot ocelot, Class<T> oclass, float f, double d0, double d1) {
-            Predicate predicate = EntitySelector.NO_CREATIVE_OR_SPECTATOR;
-
-            Objects.requireNonNull(predicate);
-            super(ocelot, oclass, f, d0, d1, predicate::test);
+            super(ocelot, oclass, f, d0, d1, EntitySelector.NO_CREATIVE_OR_SPECTATOR::test);
             this.ocelot = ocelot;
         }
 
