diff --git a/minecraft/core/registries/BuiltInRegistries.java b/minecraft/core/registries/BuiltInRegistries.java
index 0ad4922..3ed8619 100644
--- a/minecraft/core/registries/BuiltInRegistries.java
+++ b/minecraft/core/registries/BuiltInRegistries.java
@@ -323,7 +323,7 @@ public class BuiltInRegistries {
         BuiltInRegistries.LOADERS.put(resourcelocation, () -> {
             return builtinregistries_registrybootstrap.run(r0);
         });
-        BuiltInRegistries.WRITABLE_REGISTRY.register(resourcekey, (Object) r0, lifecycle);
+        BuiltInRegistries.WRITABLE_REGISTRY.register((ResourceKey) resourcekey, r0, lifecycle);
         return r0;
     }
 
