diff --git a/minecraft/server/level/ServerChunkCache.java b/minecraft/server/level/ServerChunkCache.java
index 8b5924b..84432e7 100644
--- a/minecraft/server/level/ServerChunkCache.java
+++ b/minecraft/server/level/ServerChunkCache.java
@@ -177,12 +177,12 @@ public class ServerChunkCache extends ChunkSource {
             if (chunkholder == null) {
                 return null;
             } else {
-                Either<ChunkAccess, ChunkHolder.ChunkLoadingFailure> either = (Either) chunkholder.getFutureIfPresent(ChunkStatus.FULL).getNow((Object) null);
+                Either<ChunkAccess, ChunkHolder.ChunkLoadingFailure> either = (Either) chunkholder.getFutureIfPresent(ChunkStatus.FULL).getNow(null);
 
                 if (either == null) {
                     return null;
                 } else {
-                    ChunkAccess chunkaccess1 = (ChunkAccess) either.left().orElse((Object) null);
+                    ChunkAccess chunkaccess1 = (ChunkAccess) either.left().orElse( null);
 
                     if (chunkaccess1 != null) {
                         this.storeInCache(k, chunkaccess1, ChunkStatus.FULL);
@@ -317,7 +317,7 @@ public class ServerChunkCache extends ChunkSource {
         } else if (!this.level.shouldTickBlocksAt(i)) {
             return false;
         } else {
-            Either<LevelChunk, ChunkHolder.ChunkLoadingFailure> either = (Either) chunkholder.getTickingChunkFuture().getNow((Object) null);
+            Either<LevelChunk, ChunkHolder.ChunkLoadingFailure> either = (Either) chunkholder.getTickingChunkFuture().getNow(null);
 
             return either != null && either.left().isPresent();
         }
@@ -592,7 +592,7 @@ public class ServerChunkCache extends ChunkSource {
         }
 
         @Override
-        protected boolean pollTask() {
+        public boolean pollTask() {
             if (ServerChunkCache.this.runDistanceManagerUpdates()) {
                 return true;
             } else {
