diff --git a/minecraft/data/worldgen/features/MiscOverworldFeatures.java b/minecraft/data/worldgen/features/MiscOverworldFeatures.java
index b062c48..f37a16d 100644
--- a/minecraft/data/worldgen/features/MiscOverworldFeatures.java
+++ b/minecraft/data/worldgen/features/MiscOverworldFeatures.java
@@ -58,8 +58,8 @@ public class MiscOverworldFeatures {
         FeatureUtils.register(bootstapcontext, MiscOverworldFeatures.BONUS_CHEST, Feature.BONUS_CHEST);
         FeatureUtils.register(bootstapcontext, MiscOverworldFeatures.VOID_START_PLATFORM, Feature.VOID_START_PLATFORM);
         FeatureUtils.register(bootstapcontext, MiscOverworldFeatures.DESERT_WELL, Feature.DESERT_WELL);
-        FeatureUtils.register(bootstapcontext, MiscOverworldFeatures.SPRING_LAVA_OVERWORLD, Feature.SPRING, new SpringConfiguration(Fluids.LAVA.defaultFluidState(), true, 4, 1, HolderSet.direct(Block::builtInRegistryHolder, (Object[])(Blocks.STONE, Blocks.GRANITE, Blocks.DIORITE, Blocks.ANDESITE, Blocks.DEEPSLATE, Blocks.TUFF, Blocks.CALCITE, Blocks.DIRT))));
-        FeatureUtils.register(bootstapcontext, MiscOverworldFeatures.SPRING_LAVA_FROZEN, Feature.SPRING, new SpringConfiguration(Fluids.LAVA.defaultFluidState(), true, 4, 1, HolderSet.direct(Block::builtInRegistryHolder, (Object[])(Blocks.SNOW_BLOCK, Blocks.POWDER_SNOW, Blocks.PACKED_ICE))));
-        FeatureUtils.register(bootstapcontext, MiscOverworldFeatures.SPRING_WATER, Feature.SPRING, new SpringConfiguration(Fluids.WATER.defaultFluidState(), true, 4, 1, HolderSet.direct(Block::builtInRegistryHolder, (Object[])(Blocks.STONE, Blocks.GRANITE, Blocks.DIORITE, Blocks.ANDESITE, Blocks.DEEPSLATE, Blocks.TUFF, Blocks.CALCITE, Blocks.DIRT, Blocks.SNOW_BLOCK, Blocks.POWDER_SNOW, Blocks.PACKED_ICE))));
+        FeatureUtils.register(bootstapcontext, MiscOverworldFeatures.SPRING_LAVA_OVERWORLD, Feature.SPRING, new SpringConfiguration(Fluids.LAVA.defaultFluidState(), true, 4, 1, HolderSet.direct(Block::builtInRegistryHolder, Blocks.STONE, Blocks.GRANITE, Blocks.DIORITE, Blocks.ANDESITE, Blocks.DEEPSLATE, Blocks.TUFF, Blocks.CALCITE, Blocks.DIRT)));
+        FeatureUtils.register(bootstapcontext, MiscOverworldFeatures.SPRING_LAVA_FROZEN, Feature.SPRING, new SpringConfiguration(Fluids.LAVA.defaultFluidState(), true, 4, 1, HolderSet.direct(Block::builtInRegistryHolder, Blocks.SNOW_BLOCK, Blocks.POWDER_SNOW, Blocks.PACKED_ICE)));
+        FeatureUtils.register(bootstapcontext, MiscOverworldFeatures.SPRING_WATER, Feature.SPRING, new SpringConfiguration(Fluids.WATER.defaultFluidState(), true, 4, 1, HolderSet.direct(Block::builtInRegistryHolder, Blocks.STONE, Blocks.GRANITE, Blocks.DIORITE, Blocks.ANDESITE, Blocks.DEEPSLATE, Blocks.TUFF, Blocks.CALCITE, Blocks.DIRT, Blocks.SNOW_BLOCK, Blocks.POWDER_SNOW, Blocks.PACKED_ICE)));
     }
 }
