diff --git a/minecraft/world/level/block/state/properties/BlockStateProperties.java b/minecraft/world/level/block/state/properties/BlockStateProperties.java
index a6bf3b7..8c29be0 100644
--- a/minecraft/world/level/block/state/properties/BlockStateProperties.java
+++ b/minecraft/world/level/block/state/properties/BlockStateProperties.java
@@ -40,7 +40,7 @@ public class BlockStateProperties {
     public static final BooleanProperty BLOOM = BooleanProperty.create("bloom");
     public static final BooleanProperty SHRIEKING = BooleanProperty.create("shrieking");
     public static final BooleanProperty CAN_SUMMON = BooleanProperty.create("can_summon");
-    public static final EnumProperty<Direction.Axis> HORIZONTAL_AXIS = EnumProperty.create("axis", Direction.Axis.class, (Enum[])(Direction.Axis.X, Direction.Axis.Z));
+    public static final EnumProperty<Direction.Axis> HORIZONTAL_AXIS = EnumProperty.create("axis", Direction.Axis.class, Direction.Axis.X, Direction.Axis.Z);
     public static final EnumProperty<Direction.Axis> AXIS = EnumProperty.create("axis", Direction.Axis.class);
     public static final BooleanProperty UP = BooleanProperty.create("up");
     public static final BooleanProperty DOWN = BooleanProperty.create("down");
