diff --git a/minecraft/world/level/gameevent/EntityPositionSource.java b/minecraft/world/level/gameevent/EntityPositionSource.java
index d79dc3e..3d37220 100644
--- a/minecraft/world/level/gameevent/EntityPositionSource.java
+++ b/minecraft/world/level/gameevent/EntityPositionSource.java
@@ -47,8 +47,8 @@ public class EntityPositionSource implements PositionSource {
     }
 
     private void resolveEntity(Level level) {
-        ((Optional) this.entityOrUuidOrId.map(Optional::of, (either) -> {
-            Function function = (uuid) -> {
+        (this.entityOrUuidOrId.map(Optional::of, (either) -> {
+            Function<UUID, Entity> function = (uuid) -> {
                 Entity entity;
 
                 if (level instanceof ServerLevel) {
