diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index a517582..e5021fa 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -80,6 +80,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
     @Nullable
     private DebugSampleSubscriptionTracker debugSampleSubscriptionTracker;
     private final ServerLinks serverLinks;
+    private final net.ouja.dish.Dish dishServer = new net.ouja.dish.Dish(this); // dish
 
     public DedicatedServer(
         Thread serverThread,
@@ -125,6 +126,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
 
         LOGGER.info("Loading properties");
         DedicatedServerProperties properties = this.settings.getProperties();
+        this.dishServer.initializePlugins(); // dish
         if (this.isSingleplayer()) {
             this.setLocalIp("127.0.0.1");
         } else {
