diff --git a/minecraft/util/datafix/schemas/V705.java b/minecraft/util/datafix/schemas/V705.java
index 7be984f..b8bb457 100644
--- a/minecraft/util/datafix/schemas/V705.java
+++ b/minecraft/util/datafix/schemas/V705.java
@@ -15,7 +15,7 @@ public class V705 extends NamespacedSchema {
 
     protected static final HookFunction ADD_NAMES = new HookFunction() {
         public <T> T apply(DynamicOps<T> dynamicops, T t0) {
-            return V99.addNames(new Dynamic(dynamicops, t0), V704.ITEM_TO_BLOCKENTITY, "minecraft:armor_stand");
+            return V99.addNames(new Dynamic<>(dynamicops, t0), V704.ITEM_TO_BLOCKENTITY, "minecraft:armor_stand");
         }
     };
 
