diff --git a/minecraft/commands/execution/tasks/BuildContexts.java b/minecraft/commands/execution/tasks/BuildContexts.java
index fd76ad6..0b2b14c 100644
--- a/minecraft/commands/execution/tasks/BuildContexts.java
+++ b/minecraft/commands/execution/tasks/BuildContexts.java
@@ -74,11 +74,11 @@ public class BuildContexts<T extends ExecutionCommandSource<T>> {
                             executioncommandsource = (ExecutionCommandSource) iterator.next();
 
                             try {
-                                Collection<T> collection = ContextChain.runModifier(commandcontext, executioncommandsource, ExecutionCommandSource.resultConsumer(), flag1);
+                                Collection<T> collection = ContextChain.runModifier(commandcontext, (T) executioncommandsource, ExecutionCommandSource.resultConsumer(), flag1);
                                 Iterator iterator1 = collection.iterator();
 
                                 while (iterator1.hasNext()) {
-                                    T t0 = (ExecutionCommandSource) iterator1.next();
+                                    T t0 = (T) iterator1.next();
 
                                     list2.add(t0);
                                     if (list2.size() >= j) {
@@ -113,12 +113,12 @@ public class BuildContexts<T extends ExecutionCommandSource<T>> {
 
             while (iterator.hasNext()) {
                 executioncommandsource = (ExecutionCommandSource) iterator.next();
-                customcommandexecutor.run(executioncommandsource, contextchain, flag1, executioncontrol);
+                customcommandexecutor.run((T) executioncommandsource, contextchain, flag1, executioncontrol);
             }
         } else {
             ExecuteCommand<T> executecommand = new ExecuteCommand<>(this.commandInput, flag1, commandcontext1);
 
-            ContinuationTask.schedule(executioncontext, i, (List) list1, (k, executioncommandsource1) -> {
+            ContinuationTask.schedule(executioncontext, i, list1, (k, executioncommandsource1) -> {
                 return new CommandQueueEntry<>(k, executecommand.bind(executioncommandsource1));
             });
         }
