diff --git a/minecraft/world/level/entity/EntitySectionStorage.java b/minecraft/world/level/entity/EntitySectionStorage.java
index ebe716f..f8b6a45 100644
--- a/minecraft/world/level/entity/EntitySectionStorage.java
+++ b/minecraft/world/level/entity/EntitySectionStorage.java
@@ -86,7 +86,7 @@ public class EntitySectionStorage<T extends EntityAccess> {
 
     public Stream<EntitySection<T>> getExistingSectionsInChunk(long i) {
         LongStream longstream = this.getExistingSectionPositionsInChunk(i);
-        Long2ObjectMap long2objectmap = this.sections;
+        Long2ObjectMap<EntitySection<T>> long2objectmap = this.sections;
 
         Objects.requireNonNull(this.sections);
         return longstream.mapToObj(long2objectmap::get).filter(Objects::nonNull);
