diff --git a/src/main/java/net/minecraft/world/entity/item/ItemEntity.java b/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
index 8cadd05..9bb4f42 100644
--- a/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
+++ b/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
@@ -341,6 +341,20 @@ public class ItemEntity extends Entity implements TraceableEntity {
             ItemStack item = this.getItem();
             Item item1 = item.getItem();
             int count = item.getCount();
+            // dish
+            // TODO: 12/11/2023 Check performance
+            net.ouja.dish.entity.DishEntity cachedThrower = new net.ouja.dish.entity.DishEntity((net.minecraft.world.entity.LivingEntity)this.cachedThrower);
+            net.ouja.api.entity.ItemEntity itemEntity = new net.ouja.api.entity.ItemEntity(
+                item.toDish(),
+                this.pickupDelay,
+                this.age,
+                cachedThrower,
+                this.thrower
+            );
+            net.ouja.api.event.player.PlayerAttemptPickUpItem event = new net.ouja.api.event.player.PlayerAttemptPickUpItem(new net.ouja.dish.entity.DishPlayer((net.minecraft.server.level.ServerPlayer)entity), itemEntity);
+            net.ouja.dish.plugins.RegisteredEvents.callEvent(event);
+            // dish end
+
             if (this.pickupDelay == 0 && (this.target == null || this.target.equals(entity.getUUID())) && entity.getInventory().add(item)) {
                 entity.take(this, count);
                 if (item.isEmpty()) {
