diff --git a/minecraft/world/level/saveddata/maps/MapItemSavedData.java b/minecraft/world/level/saveddata/maps/MapItemSavedData.java
index 455311f..bf504c3 100644
--- a/minecraft/world/level/saveddata/maps/MapItemSavedData.java
+++ b/minecraft/world/level/saveddata/maps/MapItemSavedData.java
@@ -15,6 +15,7 @@ import net.minecraft.core.BlockPos;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.nbt.ListTag;
 import net.minecraft.nbt.NbtOps;
+import net.minecraft.nbt.Tag;
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.protocol.Packet;
 import net.minecraft.network.protocol.game.ClientboundMapItemDataPacket;
@@ -86,7 +87,7 @@ public class MapItemSavedData extends SavedData {
     }
 
     public static MapItemSavedData load(CompoundTag compoundtag) {
-        DataResult dataresult = DimensionType.parseLegacy(new Dynamic(NbtOps.INSTANCE, compoundtag.get("dimension")));
+        DataResult<ResourceKey<Level>> dataresult = DimensionType.parseLegacy(new Dynamic<>(NbtOps.INSTANCE, compoundtag.get("dimension")));
         Logger logger = MapItemSavedData.LOGGER;
 
         Objects.requireNonNull(logger);
@@ -129,7 +130,7 @@ public class MapItemSavedData extends SavedData {
 
     @Override
     public CompoundTag save(CompoundTag compoundtag) {
-        DataResult dataresult = ResourceLocation.CODEC.encodeStart(NbtOps.INSTANCE, this.dimension.location());
+        DataResult<Tag> dataresult = ResourceLocation.CODEC.encodeStart(NbtOps.INSTANCE, this.dimension.location());
         Logger logger = MapItemSavedData.LOGGER;
 
         Objects.requireNonNull(logger);
