diff --git a/minecraft/core/Holder.java b/minecraft/core/Holder.java
index 6c997ec..3fd0325 100644
--- a/minecraft/core/Holder.java
+++ b/minecraft/core/Holder.java
@@ -114,7 +114,7 @@ public interface Holder<T> {
         }
 
         public static <T> Holder.Reference<T> createStandAlone(HolderOwner<T> holderowner, ResourceKey<T> resourcekey) {
-            return new Holder.Reference<>(Holder.Reference.Type.STAND_ALONE, holderowner, resourcekey, (Object) null);
+            return new Holder.Reference(Holder.Reference.Type.STAND_ALONE, holderowner, resourcekey, (Object) null);
         }
 
         /** @deprecated */
