diff --git a/minecraft/core/IdMapper.java b/minecraft/core/IdMapper.java
index 9f3ba4d..e0513ca 100644
--- a/minecraft/core/IdMapper.java
+++ b/minecraft/core/IdMapper.java
@@ -30,7 +30,7 @@ public class IdMapper<T> implements IdMap<T> {
         this.tToId.put(t0, i);
 
         while (this.idToT.size() <= i) {
-            this.idToT.add((Object) null);
+            this.idToT.add(null);
         }
 
         this.idToT.set(i, t0);
