diff --git a/minecraft/core/RegistryCodecs.java b/minecraft/core/RegistryCodecs.java
index 77a904e..e16afda 100644
--- a/minecraft/core/RegistryCodecs.java
+++ b/minecraft/core/RegistryCodecs.java
@@ -41,7 +41,7 @@ public class RegistryCodecs {
             Iterator iterator = registry.iterator();
 
             while (iterator.hasNext()) {
-                T t0 = iterator.next();
+                T t0 = (T) iterator.next();
 
                 builder.add(new RegistryCodecs.RegistryEntry<>((ResourceKey) registry.getResourceKey(t0).get(), registry.getId(t0), t0));
             }
