diff --git a/minecraft/world/level/storage/loot/functions/SetStewEffectFunction.java b/minecraft/world/level/storage/loot/functions/SetStewEffectFunction.java
index 772601a..ad31990 100644
--- a/minecraft/world/level/storage/loot/functions/SetStewEffectFunction.java
+++ b/minecraft/world/level/storage/loot/functions/SetStewEffectFunction.java
@@ -40,8 +40,9 @@ public class SetStewEffectFunction extends LootItemConditionalFunction {
             setsteweffectfunction_effectentry = (SetStewEffectFunction.EffectEntry) iterator.next();
         } while (set.add(setsteweffectfunction_effectentry.effect()));
 
+        EffectEntry finalSetsteweffectfunction_effectentry = setsteweffectfunction_effectentry;
         return DataResult.error(() -> {
-            return "Encountered duplicate mob effect: '" + setsteweffectfunction_effectentry.effect() + "'";
+            return "Encountered duplicate mob effect: '" + finalSetsteweffectfunction_effectentry.effect() + "'";
         });
     });
     public static final Codec<SetStewEffectFunction> CODEC = RecordCodecBuilder.create((instance) -> {
