diff --git a/minecraft/Util.java b/minecraft/Util.java
index 01c0e84..dac473a 100644
--- a/minecraft/Util.java
+++ b/minecraft/Util.java
@@ -33,6 +33,7 @@ import java.nio.file.Path;
 import java.nio.file.spi.FileSystemProvider;
 import java.security.AccessController;
 import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
 import java.time.Duration;
 import java.time.Instant;
 import java.time.ZonedDateTime;
@@ -114,7 +115,7 @@ public class Util {
     }
 
     public static <T extends Comparable<T>> String getPropertyName(Property<T> property, Object object) {
-        return property.getName((Comparable) object);
+        return property.getName((T) object);
     }
 
     public static String makeDescriptionId(String s, @Nullable ResourceLocation resourcelocation) {
@@ -287,7 +288,7 @@ public class Util {
 
             thread.setName(s);
 
-            Object object;
+            V object;
 
             try {
                 object = supplier.get();
@@ -341,8 +342,8 @@ public class Util {
     public static <T> T findPreviousInIterable(Iterable<T> iterable, @Nullable T t0) {
         Iterator<T> iterator = iterable.iterator();
 
-        Object object;
-        Object object1;
+        T object;
+        T object1;
 
         for (object1 = null; iterator.hasNext(); object1 = object) {
             object = iterator.next();
@@ -411,7 +412,7 @@ public class Util {
         list.forEach((completablefuture) -> {
             int i = list1.size();
 
-            list1.add((Object) null);
+            list1.add(null);
             acompletablefuture[i] = completablefuture.whenComplete((object, throwable) -> {
                 if (throwable != null) {
                     consumer.accept(throwable);
@@ -806,7 +807,7 @@ public class Util {
         BlockingQueue<Runnable> blockingqueue = new LinkedBlockingQueue();
 
         Objects.requireNonNull(blockingqueue);
-        Object object = function.apply(blockingqueue::add);
+        T object = function.apply(blockingqueue::add);
 
         while (!predicate.test(object)) {
             try {
@@ -874,7 +875,7 @@ public class Util {
         Optional<PartialResult<T>> optional = dataresult.error();
 
         if (optional.isPresent()) {
-            throw (Throwable) function.apply(((PartialResult) optional.get()).message());
+            throw function.apply((optional.get()).message());
         } else {
             return dataresult.result().orElseThrow();
         }
@@ -912,7 +913,7 @@ public class Util {
 
         public void openUrl(URL url) {
             try {
-                Process process = (Process) AccessController.doPrivileged(() -> {
+                Process process = (Process) AccessController.doPrivileged((PrivilegedExceptionAction<Process>) () -> {
                     return Runtime.getRuntime().exec(this.getOpenUrlArguments(url));
                 });
 
