diff --git a/minecraft/world/level/levelgen/structure/pools/alias/RandomGroup.java b/minecraft/world/level/levelgen/structure/pools/alias/RandomGroup.java
index a9b1b2d..c8416c7 100644
--- a/minecraft/world/level/levelgen/structure/pools/alias/RandomGroup.java
+++ b/minecraft/world/level/levelgen/structure/pools/alias/RandomGroup.java
@@ -19,7 +19,7 @@ record RandomGroup(SimpleWeightedRandomList<List<PoolAliasBinding>> groups) impl
     @Override
     public void forEachResolved(RandomSource randomsource, BiConsumer<ResourceKey<StructureTemplatePool>, ResourceKey<StructureTemplatePool>> biconsumer) {
         this.groups.getRandom(randomsource).ifPresent((weightedentry_wrapper) -> {
-            ((List) weightedentry_wrapper.getData()).forEach((poolaliasbinding) -> {
+            (weightedentry_wrapper.getData()).forEach((poolaliasbinding) -> {
                 poolaliasbinding.forEachResolved(randomsource, biconsumer);
             });
         });
@@ -28,7 +28,7 @@ record RandomGroup(SimpleWeightedRandomList<List<PoolAliasBinding>> groups) impl
     @Override
     public Stream<ResourceKey<StructureTemplatePool>> allTargets() {
         return this.groups.unwrap().stream().flatMap((weightedentry_wrapper) -> {
-            return ((List) weightedentry_wrapper.getData()).stream();
+            return (weightedentry_wrapper.getData()).stream();
         }).flatMap(PoolAliasBinding::allTargets);
     }
 
