diff --git a/minecraft/world/entity/monster/Slime.java b/minecraft/world/entity/monster/Slime.java
index 1894505..e286fe3 100644
--- a/minecraft/world/entity/monster/Slime.java
+++ b/minecraft/world/entity/monster/Slime.java
@@ -191,7 +191,7 @@ public class Slime extends Mob implements Enemy {
 
     @Override
     public EntityType<? extends Slime> getType() {
-        return super.getType();
+        return (EntityType<? extends Slime>) super.getType();
     }
 
     @Override
