diff --git a/minecraft/world/entity/ai/sensing/IsInWaterSensor.java b/minecraft/world/entity/ai/sensing/IsInWaterSensor.java
index 31e5c3c..851c84d 100644
--- a/minecraft/world/entity/ai/sensing/IsInWaterSensor.java
+++ b/minecraft/world/entity/ai/sensing/IsInWaterSensor.java
@@ -19,7 +19,7 @@ public class IsInWaterSensor extends Sensor<LivingEntity> {
     @Override
     protected void doTick(ServerLevel serverlevel, LivingEntity livingentity) {
         if (livingentity.isInWater()) {
-            livingentity.getBrain().setMemory(MemoryModuleType.IS_IN_WATER, (Object) Unit.INSTANCE);
+            livingentity.getBrain().setMemory(MemoryModuleType.IS_IN_WATER, Unit.INSTANCE);
         } else {
             livingentity.getBrain().eraseMemory(MemoryModuleType.IS_IN_WATER);
         }
