diff --git a/minecraft/world/entity/ai/sensing/NearestLivingEntitySensor.java b/minecraft/world/entity/ai/sensing/NearestLivingEntitySensor.java
index 3a59b8b..394f59e 100644
--- a/minecraft/world/entity/ai/sensing/NearestLivingEntitySensor.java
+++ b/minecraft/world/entity/ai/sensing/NearestLivingEntitySensor.java
@@ -27,8 +27,8 @@ public class NearestLivingEntitySensor<T extends LivingEntity> extends Sensor<T>
         list.sort(Comparator.comparingDouble(t0::distanceToSqr));
         Brain<?> brain = t0.getBrain();
 
-        brain.setMemory(MemoryModuleType.NEAREST_LIVING_ENTITIES, (Object) list);
-        brain.setMemory(MemoryModuleType.NEAREST_VISIBLE_LIVING_ENTITIES, (Object) (new NearestVisibleLivingEntities(t0, list)));
+        brain.setMemory(MemoryModuleType.NEAREST_LIVING_ENTITIES, list);
+        brain.setMemory(MemoryModuleType.NEAREST_VISIBLE_LIVING_ENTITIES, (new NearestVisibleLivingEntities(t0, list)));
     }
 
     protected int radiusXZ() {
