diff --git a/minecraft/world/entity/ai/sensing/SensorType.java b/minecraft/world/entity/ai/sensing/SensorType.java
index 95347e5..b3c86f9 100644
--- a/minecraft/world/entity/ai/sensing/SensorType.java
+++ b/minecraft/world/entity/ai/sensing/SensorType.java
@@ -63,7 +63,7 @@ public class SensorType<U extends Sensor<?>> {
     }
 
     public U create() {
-        return (Sensor) this.factory.get();
+        return this.factory.get();
     }
 
     private static <U extends Sensor<?>> SensorType<U> register(String s, Supplier<U> supplier) {
