diff --git a/minecraft/world/item/enchantment/EnchantmentCategory.java b/minecraft/world/item/enchantment/EnchantmentCategory.java
index 255baf4..d0a489c 100644
--- a/minecraft/world/item/enchantment/EnchantmentCategory.java
+++ b/minecraft/world/item/enchantment/EnchantmentCategory.java
@@ -144,7 +144,7 @@ public enum EnchantmentCategory {
     VANISHABLE {
         @Override
         public boolean canEnchant(Item item) {
-            return item instanceof Vanishable || Block.byItem(item) instanceof Vanishable || null.BREAKABLE.canEnchant(item);
+            return item instanceof Vanishable || Block.byItem(item) instanceof Vanishable || BREAKABLE.canEnchant(item);
         }
     };
 
