diff --git a/minecraft/world/entity/monster/piglin/Piglin.java b/minecraft/world/entity/monster/piglin/Piglin.java
index 78b80c3..dded89d 100644
--- a/minecraft/world/entity/monster/piglin/Piglin.java
+++ b/minecraft/world/entity/monster/piglin/Piglin.java
@@ -220,7 +220,7 @@ public class Piglin extends AbstractPiglin implements CrossbowAttackMob, Invento
 
     @Override
     public Brain<Piglin> getBrain() {
-        return super.getBrain();
+        return (Brain<Piglin>) super.getBrain();
     }
 
     @Override
@@ -416,7 +416,7 @@ public class Piglin extends AbstractPiglin implements CrossbowAttackMob, Invento
 
     @Override
     protected SoundEvent getAmbientSound() {
-        return this.level().isClientSide ? null : (SoundEvent) PiglinAi.getSoundForCurrentActivity(this).orElse((Object) null);
+        return this.level().isClientSide ? null : (SoundEvent) PiglinAi.getSoundForCurrentActivity(this).orElse(null);
     }
 
     @Override
