diff --git a/minecraft/world/entity/ai/behavior/JumpOnBed.java b/minecraft/world/entity/ai/behavior/JumpOnBed.java
index 5d42e45..96f0275 100644
--- a/minecraft/world/entity/ai/behavior/JumpOnBed.java
+++ b/minecraft/world/entity/ai/behavior/JumpOnBed.java
@@ -77,7 +77,7 @@ public class JumpOnBed extends Behavior<Mob> {
     }
 
     private void startWalkingTowardsBed(Mob mob, BlockPos blockpos) {
-        mob.getBrain().setMemory(MemoryModuleType.WALK_TARGET, (Object) (new WalkTarget(blockpos, this.speedModifier, 0)));
+        mob.getBrain().setMemory(MemoryModuleType.WALK_TARGET, (new WalkTarget(blockpos, this.speedModifier, 0)));
     }
 
     private boolean nearBed(ServerLevel serverlevel, Mob mob) {
