diff --git a/minecraft/world/level/timers/TimerCallbacks.java b/minecraft/world/level/timers/TimerCallbacks.java
index a34cb5d..e9aedc7 100644
--- a/minecraft/world/level/timers/TimerCallbacks.java
+++ b/minecraft/world/level/timers/TimerCallbacks.java
@@ -13,7 +13,7 @@ import org.slf4j.Logger;
 public class TimerCallbacks<C> {
 
     private static final Logger LOGGER = LogUtils.getLogger();
-    public static final TimerCallbacks<MinecraftServer> SERVER_CALLBACKS = (new TimerCallbacks<>()).register(new FunctionCallback.Serializer()).register(new FunctionTagCallback.Serializer());
+    public static final TimerCallbacks<MinecraftServer> SERVER_CALLBACKS = (new TimerCallbacks()).register(new FunctionCallback.Serializer()).register(new FunctionTagCallback.Serializer());
     private final Map<ResourceLocation, TimerCallback.Serializer<C, ?>> idToSerializer = Maps.newHashMap();
     private final Map<Class<?>, TimerCallback.Serializer<C, ?>> classToSerializer = Maps.newHashMap();
 
