diff --git a/minecraft/world/level/levelgen/feature/FeatureCountTracker.java b/minecraft/world/level/levelgen/feature/FeatureCountTracker.java
index df0c344..a16f2ff 100644
--- a/minecraft/world/level/levelgen/feature/FeatureCountTracker.java
+++ b/minecraft/world/level/levelgen/feature/FeatureCountTracker.java
@@ -69,7 +69,7 @@ public class FeatureCountTracker {
                 Logger logger = FeatureCountTracker.LOGGER;
                 String s2 = String.format(Locale.ROOT, "%10d ", integer1);
                 String s3 = String.format(Locale.ROOT, "%10f ", (double) integer1 / (double) integer);
-                Optional optional = featurecounttracker_featuredata.topFeature();
+                Optional<PlacedFeature> optional = featurecounttracker_featuredata.topFeature();
 
                 Objects.requireNonNull(registry);
                 logger.debug(s1 + " " + s2 + s3 + optional.flatMap(registry::getResourceKey).map(ResourceKey::location) + " " + featurecounttracker_featuredata.feature().feature() + " " + featurecounttracker_featuredata.feature());
