diff --git a/minecraft/world/level/levelgen/feature/SculkPatchFeature.java b/minecraft/world/level/levelgen/feature/SculkPatchFeature.java
index 797b927..9e1ca14 100644
--- a/minecraft/world/level/levelgen/feature/SculkPatchFeature.java
+++ b/minecraft/world/level/levelgen/feature/SculkPatchFeature.java
@@ -79,7 +79,7 @@ public class SculkPatchFeature extends Feature<SculkPatchConfiguration> {
         } else if (!blockstate.isAir() && (!blockstate.is(Blocks.WATER) || !blockstate.getFluidState().isSource())) {
             return false;
         } else {
-            Stream stream = Direction.stream();
+            Stream<Direction> stream = Direction.stream();
 
             Objects.requireNonNull(blockpos);
             return stream.map(blockpos::relative).anyMatch((blockpos1) -> {
