diff --git a/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplate.java b/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplate.java
index 570c9fa..f839bae 100644
--- a/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplate.java
+++ b/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplate.java
@@ -143,7 +143,7 @@ public class StructureTemplate {
     }
 
     private static List<StructureTemplate.StructureBlockInfo> buildInfoList(List<StructureTemplate.StructureBlockInfo> list, List<StructureTemplate.StructureBlockInfo> list1, List<StructureTemplate.StructureBlockInfo> list2) {
-        Comparator<StructureTemplate.StructureBlockInfo> comparator = Comparator.comparingInt((structuretemplate_structureblockinfo) -> {
+        Comparator<StructureTemplate.StructureBlockInfo> comparator = Comparator.<StructureTemplate.StructureBlockInfo>comparingInt((structuretemplate_structureblockinfo) -> {
             return structuretemplate_structureblockinfo.pos.getY();
         }).thenComparingInt((structuretemplate_structureblockinfo) -> {
             return structuretemplate_structureblockinfo.pos.getX();
