diff --git a/minecraft/world/level/storage/loot/LootDataManager.java b/minecraft/world/level/storage/loot/LootDataManager.java
index f83a2bc..8a210b0 100644
--- a/minecraft/world/level/storage/loot/LootDataManager.java
+++ b/minecraft/world/level/storage/loot/LootDataManager.java
@@ -89,7 +89,7 @@ public class LootDataManager implements PreparableReloadListener, LootDataResolv
             @Nullable
             @Override
             public <T> T getElement(LootDataId<T> lootdataid) {
-                return map1.get(lootdataid);
+                return (T) map1.get(lootdataid);
             }
         });
 
@@ -104,13 +104,13 @@ public class LootDataManager implements PreparableReloadListener, LootDataResolv
     }
 
     private static <T> void castAndValidate(ValidationContext validationcontext, LootDataId<T> lootdataid, Object object) {
-        lootdataid.type().runValidation(validationcontext, lootdataid, object);
+        lootdataid.type().runValidation(validationcontext, lootdataid, (T) object);
     }
 
     @Nullable
     @Override
     public <T> T getElement(LootDataId<T> lootdataid) {
-        return this.elements.get(lootdataid);
+        return (T) this.elements.get(lootdataid);
     }
 
     public Collection<ResourceLocation> getKeys(LootDataType<?> lootdatatype) {
