diff --git a/minecraft/util/CrudeIncrementalIntIdentityHashBiMap.java b/minecraft/util/CrudeIncrementalIntIdentityHashBiMap.java
index 75f7681..a4ed982 100644
--- a/minecraft/util/CrudeIncrementalIntIdentityHashBiMap.java
+++ b/minecraft/util/CrudeIncrementalIntIdentityHashBiMap.java
@@ -19,9 +19,9 @@ public class CrudeIncrementalIntIdentityHashBiMap<K> implements IdMap<K> {
     private int size;
 
     private CrudeIncrementalIntIdentityHashBiMap(int i) {
-        this.keys = new Object[i];
+        this.keys = (K[]) new Object[i];
         this.values = new int[i];
-        this.byId = new Object[i];
+        this.byId = (K[]) new Object[i];
     }
 
     private CrudeIncrementalIntIdentityHashBiMap(K[] ak, int[] aint, K[] ak1, int i, int j) {
@@ -180,6 +180,6 @@ public class CrudeIncrementalIntIdentityHashBiMap<K> implements IdMap<K> {
     }
 
     public CrudeIncrementalIntIdentityHashBiMap<K> copy() {
-        return new CrudeIncrementalIntIdentityHashBiMap<>((Object[]) this.keys.clone(), (int[]) this.values.clone(), (Object[]) this.byId.clone(), this.nextId, this.size);
+        return new CrudeIncrementalIntIdentityHashBiMap<>(this.keys.clone(), (int[]) this.values.clone(), this.byId.clone(), this.nextId, this.size);
     }
 }
