diff --git a/minecraft/util/TaskChainer.java b/minecraft/util/TaskChainer.java
index caa382f..8537bf3 100644
--- a/minecraft/util/TaskChainer.java
+++ b/minecraft/util/TaskChainer.java
@@ -16,7 +16,7 @@ public interface TaskChainer {
             @Override
             public <T> void append(CompletableFuture<T> completablefuture, Consumer<T> consumer) {
                 completablefuture.thenAcceptAsync(consumer, executor).exceptionally((throwable) -> {
-                    null.LOGGER.error("Task failed", throwable);
+                    LOGGER.error("Task failed", throwable);
                     return null;
                 });
             }
