diff --git a/minecraft/network/chat/numbers/BlankFormat.java b/minecraft/network/chat/numbers/BlankFormat.java
index cd84559..852b342 100644
--- a/minecraft/network/chat/numbers/BlankFormat.java
+++ b/minecraft/network/chat/numbers/BlankFormat.java
@@ -13,7 +13,7 @@ public class BlankFormat implements NumberFormat {
 
         @Override
         public MapCodec<BlankFormat> mapCodec() {
-            return null.CODEC;
+            return CODEC;
         }
 
         public void writeToStream(FriendlyByteBuf friendlybytebuf, BlankFormat blankformat) {}
