diff --git a/minecraft/world/level/entity/PersistentEntitySectionManager.java b/minecraft/world/level/entity/PersistentEntitySectionManager.java
index c352199..57e2c92 100644
--- a/minecraft/world/level/entity/PersistentEntitySectionManager.java
+++ b/minecraft/world/level/entity/PersistentEntitySectionManager.java
@@ -264,7 +264,7 @@ public class PersistentEntitySectionManager<T extends EntityAccess> implements A
 
         while ((chunkentities = (ChunkEntities) this.loadingInbox.poll()) != null) {
             chunkentities.getEntities().forEach((entityaccess) -> {
-                this.addEntity(entityaccess, true);
+                this.addEntity((T) entityaccess, true);
             });
             this.chunkLoadStatuses.put(chunkentities.getPos().toLong(), PersistentEntitySectionManager.ChunkLoadStatus.LOADED);
         }
