diff --git a/minecraft/data/worldgen/features/PileFeatures.java b/minecraft/data/worldgen/features/PileFeatures.java
index 55ebe1c..317b65b 100644
--- a/minecraft/data/worldgen/features/PileFeatures.java
+++ b/minecraft/data/worldgen/features/PileFeatures.java
@@ -4,6 +4,7 @@ import net.minecraft.data.worldgen.BootstapContext;
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.util.random.SimpleWeightedRandomList;
 import net.minecraft.world.level.block.Blocks;
+import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.levelgen.feature.ConfiguredFeature;
 import net.minecraft.world.level.levelgen.feature.Feature;
 import net.minecraft.world.level.levelgen.feature.configurations.BlockPileConfiguration;
@@ -25,7 +26,7 @@ public class PileFeatures {
         FeatureUtils.register(bootstapcontext, PileFeatures.PILE_HAY, Feature.BLOCK_PILE, new BlockPileConfiguration(new RotatedBlockProvider(Blocks.HAY_BLOCK)));
         FeatureUtils.register(bootstapcontext, PileFeatures.PILE_MELON, Feature.BLOCK_PILE, new BlockPileConfiguration(BlockStateProvider.simple(Blocks.MELON)));
         FeatureUtils.register(bootstapcontext, PileFeatures.PILE_SNOW, Feature.BLOCK_PILE, new BlockPileConfiguration(BlockStateProvider.simple(Blocks.SNOW)));
-        FeatureUtils.register(bootstapcontext, PileFeatures.PILE_ICE, Feature.BLOCK_PILE, new BlockPileConfiguration(new WeightedStateProvider(SimpleWeightedRandomList.builder().add(Blocks.BLUE_ICE.defaultBlockState(), 1).add(Blocks.PACKED_ICE.defaultBlockState(), 5))));
-        FeatureUtils.register(bootstapcontext, PileFeatures.PILE_PUMPKIN, Feature.BLOCK_PILE, new BlockPileConfiguration(new WeightedStateProvider(SimpleWeightedRandomList.builder().add(Blocks.PUMPKIN.defaultBlockState(), 19).add(Blocks.JACK_O_LANTERN.defaultBlockState(), 1))));
+        FeatureUtils.register(bootstapcontext, PileFeatures.PILE_ICE, Feature.BLOCK_PILE, new BlockPileConfiguration(new WeightedStateProvider(SimpleWeightedRandomList.<BlockState>builder().add(Blocks.BLUE_ICE.defaultBlockState(), 1).add(Blocks.PACKED_ICE.defaultBlockState(), 5))));
+        FeatureUtils.register(bootstapcontext, PileFeatures.PILE_PUMPKIN, Feature.BLOCK_PILE, new BlockPileConfiguration(new WeightedStateProvider(SimpleWeightedRandomList.<BlockState>builder().add(Blocks.PUMPKIN.defaultBlockState(), 19).add(Blocks.JACK_O_LANTERN.defaultBlockState(), 1))));
     }
 }
