diff --git a/minecraft/commands/arguments/blocks/BlockPredicateArgument.java b/minecraft/commands/arguments/blocks/BlockPredicateArgument.java
index f4da391..7cd8a30 100644
--- a/minecraft/commands/arguments/blocks/BlockPredicateArgument.java
+++ b/minecraft/commands/arguments/blocks/BlockPredicateArgument.java
@@ -99,7 +99,7 @@ public class BlockPredicateArgument implements ArgumentType<BlockPredicateArgume
                         return false;
                     }
 
-                    Comparable<?> comparable = (Comparable) property.getValue((String) entry.getValue()).orElse((Object) null);
+                    Comparable<?> comparable = (Comparable) property.getValue((String) entry.getValue()).orElse(null);
 
                     if (comparable == null) {
                         return false;
