diff --git a/minecraft/data/models/blockstates/VariantProperties.java b/minecraft/data/models/blockstates/VariantProperties.java
index 76bc4ee..5fdc0e7 100644
--- a/minecraft/data/models/blockstates/VariantProperties.java
+++ b/minecraft/data/models/blockstates/VariantProperties.java
@@ -6,10 +6,10 @@ import net.minecraft.resources.ResourceLocation;
 public class VariantProperties {
 
     public static final VariantProperty<VariantProperties.Rotation> X_ROT = new VariantProperty<>("x", (variantproperties_rotation) -> {
-        return new JsonPrimitive(variantproperties_rotation1.value);
+        return new JsonPrimitive(variantproperties_rotation.value);
     });
     public static final VariantProperty<VariantProperties.Rotation> Y_ROT = new VariantProperty<>("y", (variantproperties_rotation) -> {
-        return new JsonPrimitive(variantproperties_rotation1.value);
+        return new JsonPrimitive(variantproperties_rotation.value);
     });
     public static final VariantProperty<ResourceLocation> MODEL = new VariantProperty<>("model", (resourcelocation) -> {
         return new JsonPrimitive(resourcelocation.toString());
