diff --git a/minecraft/commands/arguments/OperationArgument.java b/minecraft/commands/arguments/OperationArgument.java
index b08e604..01f682f 100644
--- a/minecraft/commands/arguments/OperationArgument.java
+++ b/minecraft/commands/arguments/OperationArgument.java
@@ -55,7 +55,7 @@ public class OperationArgument implements ArgumentType<OperationArgument.Operati
     }
 
     private static OperationArgument.Operation getOperation(String s) throws CommandSyntaxException {
-        return (OperationArgument.Operation) (s.equals("><") ? (score, score1) -> {
+        return (s.equals("><") ? (score, score1) -> {
             int i = score.getScore();
 
             score.setScore(score1.getScore());
