diff --git a/minecraft/nbt/TextComponentTagVisitor.java b/minecraft/nbt/TextComponentTagVisitor.java
index b735efb..760de50 100644
--- a/minecraft/nbt/TextComponentTagVisitor.java
+++ b/minecraft/nbt/TextComponentTagVisitor.java
@@ -21,7 +21,7 @@ public class TextComponentTagVisitor implements TagVisitor {
 
     private static final Logger LOGGER = LogUtils.getLogger();
     private static final int INLINE_LIST_THRESHOLD = 8;
-    private static final ByteCollection INLINE_ELEMENT_TYPES = new ByteOpenHashSet(Arrays.asList(1, 2, 3, 4, 5, 6));
+    private static final ByteCollection INLINE_ELEMENT_TYPES = new ByteOpenHashSet(Arrays.asList((byte)1, (byte)2, (byte)3, (byte)4, (byte)5, (byte)6));
     private static final ChatFormatting SYNTAX_HIGHLIGHTING_KEY = ChatFormatting.AQUA;
     private static final ChatFormatting SYNTAX_HIGHLIGHTING_STRING = ChatFormatting.GREEN;
     private static final ChatFormatting SYNTAX_HIGHLIGHTING_NUMBER = ChatFormatting.GOLD;
