diff --git a/minecraft/resources/RegistryDataLoader.java b/minecraft/resources/RegistryDataLoader.java
index 29b1460..f994f12 100644
--- a/minecraft/resources/RegistryDataLoader.java
+++ b/minecraft/resources/RegistryDataLoader.java
@@ -121,7 +121,7 @@ public class RegistryDataLoader {
 
         map1.entrySet().stream().sorted(Entry.comparingByKey()).forEach((entry) -> {
             printwriter.printf("> Errors in registry %s:%n", entry.getKey());
-            ((Map) entry.getValue()).entrySet().stream().sorted(Entry.comparingByKey()).forEach((entry1) -> {
+            (entry.getValue()).entrySet().stream().sorted(Entry.comparingByKey()).forEach((entry1) -> {
                 printwriter.printf(">> Errors in element %s:%n", entry1.getKey());
                 ((Exception) entry1.getValue()).printStackTrace(printwriter);
             });
