diff --git a/minecraft/world/level/block/entity/StructureBlockEntity.java b/minecraft/world/level/block/entity/StructureBlockEntity.java
index 8a1d604..f0f4e44 100644
--- a/minecraft/world/level/block/entity/StructureBlockEntity.java
+++ b/minecraft/world/level/block/entity/StructureBlockEntity.java
@@ -302,7 +302,7 @@ public class StructureBlockEntity extends BlockEntity {
     }
 
     private Stream<BlockPos> getRelatedCorners(BlockPos blockpos, BlockPos blockpos1) {
-        Stream stream = BlockPos.betweenClosedStream(blockpos, blockpos1).filter((blockpos2) -> {
+        Stream<BlockPos> stream = BlockPos.betweenClosedStream(blockpos, blockpos1).filter((blockpos2) -> {
             return this.level.getBlockState(blockpos2).is(Blocks.STRUCTURE_BLOCK);
         });
         Level level = this.level;
