diff --git a/minecraft/world/entity/monster/breeze/Slide.java b/minecraft/world/entity/monster/breeze/Slide.java
index bf417d0..de0e3cb 100644
--- a/minecraft/world/entity/monster/breeze/Slide.java
+++ b/minecraft/world/entity/monster/breeze/Slide.java
@@ -27,7 +27,7 @@ public class Slide extends Behavior<Breeze> {
     }
 
     protected void start(ServerLevel serverlevel, Breeze breeze, long i) {
-        LivingEntity livingentity = (LivingEntity) breeze.getBrain().getMemory(MemoryModuleType.ATTACK_TARGET).orElse((Object) null);
+        LivingEntity livingentity = (LivingEntity) breeze.getBrain().getMemory(MemoryModuleType.ATTACK_TARGET).orElse(null);
 
         if (livingentity != null) {
             boolean flag = breeze.withinOuterCircleRange(livingentity.position());
@@ -48,7 +48,7 @@ public class Slide extends Behavior<Breeze> {
             }
 
             if (vec3 != null) {
-                breeze.getBrain().setMemory(MemoryModuleType.WALK_TARGET, (Object) (new WalkTarget(BlockPos.containing(vec3), 0.6F, 1)));
+                breeze.getBrain().setMemory(MemoryModuleType.WALK_TARGET, (new WalkTarget(BlockPos.containing(vec3), 0.6F, 1)));
             }
 
         }
