diff --git a/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java b/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java
index e9c791d..50b00a2 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayerGameMode.java
@@ -241,7 +241,20 @@ public class ServerPlayerGameMode {
         } else {
             BlockEntity blockentity = this.level.getBlockEntity(blockpos);
             Block block = blockstate.getBlock();
-
+            // dish
+            try {
+                net.ouja.dish.world.block.DishBlockInfo dishBlockInfo = new net.ouja.dish.world.block.DishBlockInfo(block, blockpos, this.level);
+                if (dishBlockInfo.getBlock() != null) {
+                    net.ouja.dish.entity.DishPlayer dishPlayer = new net.ouja.dish.entity.DishPlayer(this.player);
+                    net.ouja.api.event.player.PlayerBreakBlockEvent event = new net.ouja.api.event.player.PlayerBreakBlockEvent(dishPlayer, dishBlockInfo.getBlock());
+                    net.ouja.dish.plugins.RegisteredEvents.callEvent(event);
+                    if (event.isCancelled()) return false;
+                }
+            } catch (Exception e) {
+                // TODO: 10/31/2023 make proper error logging
+                e.printStackTrace();
+            }
+            // dish end
             if (block instanceof GameMasterBlock && !this.player.canUseGameMasterBlocks()) {
                 this.level.sendBlockUpdated(blockpos, blockstate, blockstate, 3);
                 return false;
