diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
index 161c17a..4ff6878 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
@@ -1533,6 +1533,11 @@ public class ServerPlayer extends Player {
         if (this.gameMode.getGameModeForPlayer() == GameType.SPECTATOR) {
             this.setCamera(entity);
         } else {
+            // dish start
+            net.ouja.api.event.player.PlayerAttackEvent event = new net.ouja.api.event.player.PlayerAttackEvent(new net.ouja.dish.entity.DishPlayer(this), new net.ouja.dish.entity.DishEntity(entity));
+            net.ouja.dish.plugins.RegisteredEvents.callEvent(event);
+            if (event.isCancelled()) return;
+            // dish end
             super.attack(entity);
         }
 
