diff --git a/minecraft/world/level/levelgen/structure/pools/StructurePoolElementType.java b/minecraft/world/level/levelgen/structure/pools/StructurePoolElementType.java
index 01b4826..91d457b 100644
--- a/minecraft/world/level/levelgen/structure/pools/StructurePoolElementType.java
+++ b/minecraft/world/level/levelgen/structure/pools/StructurePoolElementType.java
@@ -15,7 +15,7 @@ public interface StructurePoolElementType<P extends StructurePoolElement> {
     Codec<P> codec();
 
     static <P extends StructurePoolElement> StructurePoolElementType<P> register(String s, Codec<P> codec) {
-        return (StructurePoolElementType) Registry.register(BuiltInRegistries.STRUCTURE_POOL_ELEMENT, s, () -> {
+        return Registry.register(BuiltInRegistries.STRUCTURE_POOL_ELEMENT, s, () -> {
             return codec;
         });
     }
