diff --git a/src/main/java/net/ouja/dish/entity/neutral/DishSpider.java b/src/main/java/net/ouja/dish/entity/neutral/DishSpider.java
new file mode 100644
index 0000000..5d94b42
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/neutral/DishSpider.java
@@ -0,0 +1,19 @@
+package net.ouja.dish.entity.neutral;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.monster.Spider;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishSpider extends DishEntity implements net.ouja.api.entity.neutral.Spider {
+    private Spider spider;
+
+    public DishSpider(Entity entity) {
+        super(entity);
+        this.spider = (Spider)entity;
+    }
+
+    @Override
+    public boolean isClimbing() {
+        return this.spider.isClimbing();
+    }
+}
