diff --git a/src/main/java/net/ouja/dish/entity/neutral/DishDolphin.java b/src/main/java/net/ouja/dish/entity/neutral/DishDolphin.java
new file mode 100644
index 0000000..01e034c
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/neutral/DishDolphin.java
@@ -0,0 +1,14 @@
+package net.ouja.dish.entity.neutral;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.animal.Dolphin;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishDolphin extends DishEntity {
+    private Dolphin dolphin;
+
+    public DishDolphin(Entity entity) {
+        super(entity);
+        this.dolphin = (Dolphin)entity;
+    }
+}
