diff --git a/src/main/java/net/ouja/dish/entity/neutral/DishZombifiedPiglin.java b/src/main/java/net/ouja/dish/entity/neutral/DishZombifiedPiglin.java
new file mode 100644
index 0000000..ef0bf7e
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/neutral/DishZombifiedPiglin.java
@@ -0,0 +1,14 @@
+package net.ouja.dish.entity.neutral;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.monster.ZombifiedPiglin;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishZombifiedPiglin extends DishEntity implements net.ouja.api.entity.neutral.ZombifiedPiglin {
+    private ZombifiedPiglin zombifiedPiglin;
+
+    public DishZombifiedPiglin(Entity entity) {
+        super(entity);
+        this.zombifiedPiglin = (ZombifiedPiglin)entity;
+    }
+}
