diff --git a/src/main/java/net/ouja/dish/entity/neutral/DishEnderman.java b/src/main/java/net/ouja/dish/entity/neutral/DishEnderman.java
new file mode 100644
index 0000000..ea6bb1d
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/neutral/DishEnderman.java
@@ -0,0 +1,14 @@
+package net.ouja.dish.entity.neutral;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.monster.EnderMan;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishEnderman extends DishEntity {
+    private EnderMan enderMan;
+
+    public DishEnderman(Entity entity) {
+        super(entity);
+        this.enderMan = (EnderMan)entity;
+    }
+}
