diff --git a/src/main/java/net/ouja/dish/entity/neutral/DishLlama.java b/src/main/java/net/ouja/dish/entity/neutral/DishLlama.java
new file mode 100644
index 0000000..404089f
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/neutral/DishLlama.java
@@ -0,0 +1,20 @@
+package net.ouja.dish.entity.neutral;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.animal.horse.Llama;
+import net.ouja.api.entity.neutral.LlamaVariants;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishLlama extends DishEntity implements net.ouja.api.entity.neutral.Llama {
+    private Llama llama;
+
+    public DishLlama(Entity entity) {
+        super(entity);
+        this.llama = (Llama)entity;
+    }
+
+    @Override
+    public LlamaVariants getVariant() {
+        return LlamaVariants.valueOf(this.llama.getVariant().name());
+    }
+}
