diff --git a/minecraft/world/item/enchantment/EnchantmentHelper.java b/minecraft/world/item/enchantment/EnchantmentHelper.java
index 4c605ee..c93c79d 100644
--- a/minecraft/world/item/enchantment/EnchantmentHelper.java
+++ b/minecraft/world/item/enchantment/EnchantmentHelper.java
@@ -391,7 +391,7 @@ public class EnchantmentHelper {
             List<EnchantmentInstance> list1 = getAvailableEnchantmentResults(i, itemstack, flag);
 
             if (!list1.isEmpty()) {
-                Optional optional = WeightedRandom.getRandomItem(randomsource, list1);
+                Optional<EnchantmentInstance> optional = WeightedRandom.getRandomItem(randomsource, list1);
 
                 Objects.requireNonNull(list);
                 optional.ifPresent(list::add);
