diff --git a/minecraft/world/level/storage/loot/functions/SetNameFunction.java b/minecraft/world/level/storage/loot/functions/SetNameFunction.java
index f768b42..05dd977 100644
--- a/minecraft/world/level/storage/loot/functions/SetNameFunction.java
+++ b/minecraft/world/level/storage/loot/functions/SetNameFunction.java
@@ -78,7 +78,7 @@ public class SetNameFunction extends LootItemConditionalFunction {
     @Override
     public ItemStack run(ItemStack itemstack, LootContext lootcontext) {
         this.name.ifPresent((component) -> {
-            itemstack.setHoverName((Component) createResolver(lootcontext, (LootContext.EntityTarget) this.resolutionContext.orElse((Object) null)).apply(component));
+            itemstack.setHoverName((Component) createResolver(lootcontext, (LootContext.EntityTarget) this.resolutionContext.orElse(null)).apply(component));
         });
         return itemstack;
     }
