diff --git a/minecraft/network/protocol/common/ClientboundResourcePackPushPacket.java b/minecraft/network/protocol/common/ClientboundResourcePackPushPacket.java
index a313762..495600d 100644
--- a/minecraft/network/protocol/common/ClientboundResourcePackPushPacket.java
+++ b/minecraft/network/protocol/common/ClientboundResourcePackPushPacket.java
@@ -10,15 +10,15 @@ public record ClientboundResourcePackPushPacket(UUID id, String url, String hash
 
     public static final int MAX_HASH_LENGTH = 40;
 
-    public ClientboundResourcePackPushPacket(UUID uuid, String s, String s1, boolean flag, @Nullable Component component) {
-        if (s1.length() > 40) {
-            throw new IllegalArgumentException("Hash is too long (max 40, was " + s1.length() + ")");
+    public ClientboundResourcePackPushPacket(UUID id, String url, String hash, boolean required, @Nullable Component prompt) {
+        if (hash.length() > 40) {
+            throw new IllegalArgumentException("Hash is too long (max 40, was " + hash.length() + ")");
         } else {
-            this.id = uuid;
-            this.url = s;
-            this.hash = s1;
-            this.required = flag;
-            this.prompt = component;
+            this.id = id;
+            this.url = url;
+            this.hash = hash;
+            this.required = required;
+            this.prompt = prompt;
         }
     }
 
