diff --git a/minecraft/network/protocol/common/ServerboundCustomPayloadPacket.java b/minecraft/network/protocol/common/ServerboundCustomPayloadPacket.java
index 9cd8d93..6d5cf27 100644
--- a/minecraft/network/protocol/common/ServerboundCustomPayloadPacket.java
+++ b/minecraft/network/protocol/common/ServerboundCustomPayloadPacket.java
@@ -12,7 +12,7 @@ import net.minecraft.resources.ResourceLocation;
 public record ServerboundCustomPayloadPacket(CustomPacketPayload payload) implements Packet<ServerCommonPacketListener> {
 
     private static final int MAX_PAYLOAD_SIZE = 32767;
-    private static final Map<ResourceLocation, FriendlyByteBuf.Reader<? extends CustomPacketPayload>> KNOWN_TYPES = ImmutableMap.builder().put(BrandPayload.ID, BrandPayload::new).build();
+    private static final Map<ResourceLocation, FriendlyByteBuf.Reader<? extends CustomPacketPayload>> KNOWN_TYPES = ImmutableMap.<ResourceLocation, FriendlyByteBuf.Reader<? extends CustomPacketPayload>>builder().put(BrandPayload.ID, BrandPayload::new).build();
 
     public ServerboundCustomPayloadPacket(FriendlyByteBuf friendlybytebuf) {
         this(readPayload(friendlybytebuf.readResourceLocation(), friendlybytebuf));
