diff --git a/minecraft/commands/Commands.java b/minecraft/commands/Commands.java
index 2b82036..3e00705 100644
--- a/minecraft/commands/Commands.java
+++ b/minecraft/commands/Commands.java
@@ -351,7 +351,7 @@ public class Commands {
 
                 executioncontext1.close();
             } finally {
-                Commands.CURRENT_EXECUTION_CONTEXT.set((Object) null);
+                Commands.CURRENT_EXECUTION_CONTEXT.set(null);
             }
         } else {
             consumer.accept(executioncontext);
