diff --git a/minecraft/util/datafix/fixes/BlockEntityUUIDFix.java b/minecraft/util/datafix/fixes/BlockEntityUUIDFix.java
index 2425128..517a91a 100644
--- a/minecraft/util/datafix/fixes/BlockEntityUUIDFix.java
+++ b/minecraft/util/datafix/fixes/BlockEntityUUIDFix.java
@@ -20,8 +20,8 @@ public class BlockEntityUUIDFix extends AbstractUUIDFix {
 
     private Dynamic<?> updateSkull(Dynamic<?> dynamic) {
         return (Dynamic) dynamic.get("Owner").get().map((dynamic1) -> {
-            return (Dynamic) replaceUUIDString(dynamic1, "Id", "Id").orElse(dynamic1);
-        }).map((dynamic1) -> {
+            return replaceUUIDString(dynamic1, "Id", "Id").orElse(dynamic1);
+        }).<Dynamic<?>>map((dynamic1) -> {
             return dynamic.remove("Owner").set("SkullOwner", dynamic1);
         }).result().orElse(dynamic);
     }
