diff --git a/minecraft/util/datafix/fixes/EntityVariantFix.java b/minecraft/util/datafix/fixes/EntityVariantFix.java
index 8bc39d6..bb8b5cd 100644
--- a/minecraft/util/datafix/fixes/EntityVariantFix.java
+++ b/minecraft/util/datafix/fixes/EntityVariantFix.java
@@ -25,7 +25,7 @@ public class EntityVariantFix extends NamedEntityFix {
         return dynamic.map((object) -> {
             DynamicOps<T> dynamicops = dynamic.getOps();
             Function<T, T> function1 = (object1) -> {
-                return ((Dynamic) function.apply(new Dynamic(dynamicops, object1))).getValue();
+                return (T) ((Dynamic) function.apply(new Dynamic(dynamicops, object1))).getValue();
             };
 
             return dynamicops.get(object, s).map((object1) -> {
