diff --git a/minecraft/util/datafix/fixes/EntityRedundantChanceTagsFix.java b/minecraft/util/datafix/fixes/EntityRedundantChanceTagsFix.java
index bf39de5..5813553 100644
--- a/minecraft/util/datafix/fixes/EntityRedundantChanceTagsFix.java
+++ b/minecraft/util/datafix/fixes/EntityRedundantChanceTagsFix.java
@@ -34,7 +34,7 @@ public class EntityRedundantChanceTagsFix extends DataFix {
     }
 
     private static boolean isZeroList(OptionalDynamic<?> optionaldynamic, int i) {
-        Codec codec = EntityRedundantChanceTagsFix.FLOAT_LIST_CODEC;
+        Codec<List<Float>> codec = EntityRedundantChanceTagsFix.FLOAT_LIST_CODEC;
 
         Objects.requireNonNull(codec);
         return (Boolean) optionaldynamic.flatMap(codec::parse).map((list) -> {
