diff --git a/minecraft/util/datafix/fixes/EntityBlockStateFix.java b/minecraft/util/datafix/fixes/EntityBlockStateFix.java
index 426520b..c426802 100644
--- a/minecraft/util/datafix/fixes/EntityBlockStateFix.java
+++ b/minecraft/util/datafix/fixes/EntityBlockStateFix.java
@@ -333,7 +333,7 @@ public class EntityBlockStateFix extends DataFix {
 
         return typed.update(type.finder(), type1, (either) -> {
             int i = (Integer) either.map((pair) -> {
-                return (Integer) ((Either) pair.getSecond()).map((integer) -> {
+                return (Integer) (pair.getSecond()).map((integer) -> {
                     return integer;
                 }, EntityBlockStateFix::getBlockId);
             }, (unit) -> {
@@ -355,7 +355,7 @@ public class EntityBlockStateFix extends DataFix {
         Dynamic<?> dynamic = (Dynamic) typed.getOrCreate(DSL.remainderFinder());
 
         return typed.update(type.finder(), type1, (pair) -> {
-            int i = (Integer) ((Either) pair.getSecond()).map((integer) -> {
+            int i = (Integer)  (pair.getSecond()).map((integer) -> {
                 return integer;
             }, EntityBlockStateFix::getBlockId);
             int j = dynamic.get(s1).asInt(0) & 15;
