diff --git a/minecraft/util/datafix/fixes/RecipesRenameningFix.java b/minecraft/util/datafix/fixes/RecipesRenameningFix.java
index 9b70466..1785390 100644
--- a/minecraft/util/datafix/fixes/RecipesRenameningFix.java
+++ b/minecraft/util/datafix/fixes/RecipesRenameningFix.java
@@ -5,7 +5,7 @@ import java.util.Map;
 
 public class RecipesRenameningFix {
 
-    public static final Map<String, String> RECIPES = ImmutableMap.builder().put("minecraft:acacia_bark", "minecraft:acacia_wood").put("minecraft:birch_bark", "minecraft:birch_wood").put("minecraft:dark_oak_bark", "minecraft:dark_oak_wood").put("minecraft:jungle_bark", "minecraft:jungle_wood").put("minecraft:oak_bark", "minecraft:oak_wood").put("minecraft:spruce_bark", "minecraft:spruce_wood").build();
+    public static final Map<String, String> RECIPES = ImmutableMap.<String, String>builder().put("minecraft:acacia_bark", "minecraft:acacia_wood").put("minecraft:birch_bark", "minecraft:birch_wood").put("minecraft:dark_oak_bark", "minecraft:dark_oak_wood").put("minecraft:jungle_bark", "minecraft:jungle_wood").put("minecraft:oak_bark", "minecraft:oak_wood").put("minecraft:spruce_bark", "minecraft:spruce_wood").build();
 
     public RecipesRenameningFix() {}
 }
