diff --git a/minecraft/util/datafix/fixes/RemapChunkStatusFix.java b/minecraft/util/datafix/fixes/RemapChunkStatusFix.java
index 83a5954..55e2c39 100644
--- a/minecraft/util/datafix/fixes/RemapChunkStatusFix.java
+++ b/minecraft/util/datafix/fixes/RemapChunkStatusFix.java
@@ -33,11 +33,11 @@ public class RemapChunkStatusFix extends DataFix {
     }
 
     private <T> Dynamic<T> fixStatus(Dynamic<T> dynamic) {
-        Optional optional = dynamic.asString().result().map(NamespacedSchema::ensureNamespaced).map(this.mapper);
+        Optional<String> optional = dynamic.asString().result().map(NamespacedSchema::ensureNamespaced).map(this.mapper);
 
         Objects.requireNonNull(dynamic);
         Optional<Dynamic<T>> optional1 = optional.map(dynamic::createString);
 
-        return (Dynamic) DataFixUtils.orElse(optional1, dynamic);
+        return DataFixUtils.orElse(optional1, dynamic);
     }
 }
