diff --git a/minecraft/util/datafix/fixes/TrappedChestBlockEntityFix.java b/minecraft/util/datafix/fixes/TrappedChestBlockEntityFix.java
index 592398d..2b623f1 100644
--- a/minecraft/util/datafix/fixes/TrappedChestBlockEntityFix.java
+++ b/minecraft/util/datafix/fixes/TrappedChestBlockEntityFix.java
@@ -80,7 +80,7 @@ public class TrappedChestBlockEntityFix extends DataFix {
                             Dynamic<?> dynamic = (Dynamic) typed1.get(DSL.remainderFinder());
                             int k = dynamic.get("xPos").asInt(0);
                             int l = dynamic.get("zPos").asInt(0);
-                            TaggedChoiceType<String> taggedchoicetype = this.getInputSchema().findChoiceType(References.BLOCK_ENTITY);
+                            TaggedChoiceType<String> taggedchoicetype = (TaggedChoiceType<String>) this.getInputSchema().findChoiceType(References.BLOCK_ENTITY);
 
                             return typed1.updateTyped(opticfinder, (typed3) -> {
                                 return typed3.updateTyped(taggedchoicetype.finder(), (typed4) -> {
