diff --git a/minecraft/util/datafix/fixes/WallPropertyFix.java b/minecraft/util/datafix/fixes/WallPropertyFix.java
index 46cb6ca..3e42daf 100644
--- a/minecraft/util/datafix/fixes/WallPropertyFix.java
+++ b/minecraft/util/datafix/fixes/WallPropertyFix.java
@@ -31,10 +31,10 @@ public class WallPropertyFix extends DataFix {
 
     private static <T> Dynamic<T> fixWallProperty(Dynamic<T> dynamic, String s) {
         return dynamic.update(s, (dynamic1) -> {
-            Optional optional = dynamic1.asString().result().map(WallPropertyFix::mapProperty);
+            Optional<String> optional = dynamic1.asString().result().map(WallPropertyFix::mapProperty);
 
             Objects.requireNonNull(dynamic1);
-            return (Dynamic) DataFixUtils.orElse(optional.map(dynamic1::createString), dynamic1);
+            return DataFixUtils.orElse(optional.map(dynamic1::createString), dynamic1);
         });
     }
 
