diff --git a/minecraft/world/item/crafting/SimpleCraftingRecipeSerializer.java b/minecraft/world/item/crafting/SimpleCraftingRecipeSerializer.java
index af2ee71..c776094 100644
--- a/minecraft/world/item/crafting/SimpleCraftingRecipeSerializer.java
+++ b/minecraft/world/item/crafting/SimpleCraftingRecipeSerializer.java
@@ -14,10 +14,7 @@ public class SimpleCraftingRecipeSerializer<T extends CraftingRecipe> implements
     public SimpleCraftingRecipeSerializer(SimpleCraftingRecipeSerializer.Factory<T> simplecraftingrecipeserializer_factory) {
         this.constructor = simplecraftingrecipeserializer_factory;
         this.codec = RecordCodecBuilder.create((instance) -> {
-            P1 p1 = instance.group(CraftingBookCategory.CODEC.fieldOf("category").orElse(CraftingBookCategory.MISC).forGetter(CraftingRecipe::category));
-
-            Objects.requireNonNull(simplecraftingrecipeserializer_factory);
-            return p1.apply(instance, simplecraftingrecipeserializer_factory::create);
+            return instance.group(CraftingBookCategory.CODEC.fieldOf("category").orElse(CraftingBookCategory.MISC).forGetter(CraftingRecipe::category)).apply(instance, simplecraftingrecipeserializer_factory::create);
         });
     }
 
