diff --git a/minecraft/util/datafix/fixes/DyeItemRenameFix.java b/minecraft/util/datafix/fixes/DyeItemRenameFix.java
index 4ebd706..aa3ffd5 100644
--- a/minecraft/util/datafix/fixes/DyeItemRenameFix.java
+++ b/minecraft/util/datafix/fixes/DyeItemRenameFix.java
@@ -5,7 +5,7 @@ import java.util.Map;
 
 public class DyeItemRenameFix {
 
-    public static final Map<String, String> RENAMED_IDS = ImmutableMap.builder().put("minecraft:cactus_green", "minecraft:green_dye").put("minecraft:rose_red", "minecraft:red_dye").put("minecraft:dandelion_yellow", "minecraft:yellow_dye").build();
+    public static final Map<String, String> RENAMED_IDS = ImmutableMap.<String, String>builder().put("minecraft:cactus_green", "minecraft:green_dye").put("minecraft:rose_red", "minecraft:red_dye").put("minecraft:dandelion_yellow", "minecraft:yellow_dye").build();
 
     public DyeItemRenameFix() {}
 }
