diff --git a/minecraft/util/datafix/fixes/EntityRenameFix.java b/minecraft/util/datafix/fixes/EntityRenameFix.java
index 816e675..88c45f4 100644
--- a/minecraft/util/datafix/fixes/EntityRenameFix.java
+++ b/minecraft/util/datafix/fixes/EntityRenameFix.java
@@ -20,8 +20,8 @@ public abstract class EntityRenameFix extends DataFix {
     }
 
     public TypeRewriteRule makeRule() {
-        TaggedChoiceType<String> taggedchoicetype = this.getInputSchema().findChoiceType(References.ENTITY);
-        TaggedChoiceType<String> taggedchoicetype1 = this.getOutputSchema().findChoiceType(References.ENTITY);
+        TaggedChoiceType<String> taggedchoicetype = (TaggedChoiceType<String>) this.getInputSchema().findChoiceType(References.ENTITY);
+        TaggedChoiceType<String> taggedchoicetype1 = (TaggedChoiceType<String>) this.getOutputSchema().findChoiceType(References.ENTITY);
 
         return this.fixTypeEverywhere(this.name, taggedchoicetype, taggedchoicetype1, (dynamicops) -> {
             return (pair) -> {
