diff --git a/minecraft/util/datafix/fixes/EntityShulkerRotationFix.java b/minecraft/util/datafix/fixes/EntityShulkerRotationFix.java
index bab3fad..ee4fede 100644
--- a/minecraft/util/datafix/fixes/EntityShulkerRotationFix.java
+++ b/minecraft/util/datafix/fixes/EntityShulkerRotationFix.java
@@ -21,7 +21,7 @@ public class EntityShulkerRotationFix extends NamedEntityFix {
 
         if (!list.isEmpty()) {
             list.set(0, (Double) list.get(0) - 180.0D);
-            Stream stream = list.stream();
+            Stream<Double> stream = list.stream();
 
             Objects.requireNonNull(dynamic);
             return dynamic.set("Rotation", dynamic.createList(stream.map(dynamic::createDouble)));
