diff --git a/minecraft/util/datafix/fixes/FeatureFlagRemoveFix.java b/minecraft/util/datafix/fixes/FeatureFlagRemoveFix.java
index 999b847..cd8d251 100644
--- a/minecraft/util/datafix/fixes/FeatureFlagRemoveFix.java
+++ b/minecraft/util/datafix/fixes/FeatureFlagRemoveFix.java
@@ -12,6 +12,7 @@ import java.util.Objects;
 import java.util.Optional;
 import java.util.Set;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class FeatureFlagRemoveFix extends DataFix {
 
@@ -33,7 +34,7 @@ public class FeatureFlagRemoveFix extends DataFix {
     private <T> Dynamic<T> fixTag(Dynamic<T> dynamic) {
         List<Dynamic<T>> list = (List) dynamic.get("removed_features").asStream().collect(Collectors.toCollection(ArrayList::new));
         Dynamic<T> dynamic1 = dynamic.update("enabled_features", (dynamic2) -> {
-            Optional optional = dynamic2.asStreamOpt().result().map((stream) -> {
+            Optional<Stream> optional = dynamic2.asStreamOpt().result().map((stream) -> {
                 return stream.filter((dynamic3) -> {
                     Optional<String> optional1 = dynamic3.asString().result();
 
@@ -52,7 +53,7 @@ public class FeatureFlagRemoveFix extends DataFix {
             });
 
             Objects.requireNonNull(dynamic);
-            return (Dynamic) DataFixUtils.orElse(optional.map(dynamic::createList), dynamic2);
+            return DataFixUtils.orElse(optional.map(dynamic::createList), dynamic2);
         });
 
         if (!list.isEmpty()) {
