diff --git a/minecraft/util/datafix/fixes/ItemSpawnEggFix.java b/minecraft/util/datafix/fixes/ItemSpawnEggFix.java
index 69eef4c..b5cdd11 100644
--- a/minecraft/util/datafix/fixes/ItemSpawnEggFix.java
+++ b/minecraft/util/datafix/fixes/ItemSpawnEggFix.java
@@ -124,8 +124,9 @@ public class ItemSpawnEggFix extends DataFix {
                     Typed<?> typed2 = typed.getOrCreateTyped(opticfinder2);
                     Typed<?> typed3 = typed2.getOrCreateTyped(opticfinder3);
                     Typed<?> typed4 = typed3.getOrCreateTyped(opticfinder4);
+                    Dynamic<?> finalDynamic = dynamic;
                     Typed<?> typed5 = Util.writeAndReadTypedOrThrow(typed4, type1, (dynamic1) -> {
-                        return dynamic1.set("id", dynamic.createString(s));
+                        return dynamic1.set("id", finalDynamic.createString(s));
                     });
 
                     typed1 = typed.set(opticfinder2, typed2.set(opticfinder3, typed3.set(opticfinder4, typed5)));
