diff --git a/minecraft/util/datafix/fixes/JigsawRotationFix.java b/minecraft/util/datafix/fixes/JigsawRotationFix.java
index fc1e161..8c98c8f 100644
--- a/minecraft/util/datafix/fixes/JigsawRotationFix.java
+++ b/minecraft/util/datafix/fixes/JigsawRotationFix.java
@@ -11,7 +11,7 @@ import java.util.Optional;
 
 public class JigsawRotationFix extends DataFix {
 
-    private static final Map<String, String> RENAMES = ImmutableMap.builder().put("down", "down_south").put("up", "up_north").put("north", "north_up").put("south", "south_up").put("west", "west_up").put("east", "east_up").build();
+    private static final Map<String, String> RENAMES = ImmutableMap.<String, String>builder().put("down", "down_south").put("up", "up_north").put("north", "north_up").put("south", "south_up").put("west", "west_up").put("east", "east_up").build();
 
     public JigsawRotationFix(Schema schema, boolean flag) {
         super(schema, flag);
