diff --git a/minecraft/util/datafix/fixes/LevelUUIDFix.java b/minecraft/util/datafix/fixes/LevelUUIDFix.java
index 57ed59f..55c6c81 100644
--- a/minecraft/util/datafix/fixes/LevelUUIDFix.java
+++ b/minecraft/util/datafix/fixes/LevelUUIDFix.java
@@ -50,7 +50,7 @@ public class LevelUUIDFix extends AbstractUUIDFix {
                 return pair.mapSecond((dynamic2) -> {
                     return dynamic2.update("Players", (dynamic3) -> {
                         return dynamic2.createList(dynamic3.asStream().map((dynamic4) -> {
-                            return (Dynamic) createUUIDFromML(dynamic4).orElseGet(() -> {
+                            return createUUIDFromML(dynamic4).orElseGet(() -> {
                                 LevelUUIDFix.LOGGER.warn("CustomBossEvents contains invalid UUIDs.");
                                 return dynamic4;
                             });
