diff --git a/minecraft/util/datafix/fixes/ScoreboardDisplaySlotFix.java b/minecraft/util/datafix/fixes/ScoreboardDisplaySlotFix.java
index f85f7fa..cb2a7bb 100644
--- a/minecraft/util/datafix/fixes/ScoreboardDisplaySlotFix.java
+++ b/minecraft/util/datafix/fixes/ScoreboardDisplaySlotFix.java
@@ -16,7 +16,7 @@ import javax.annotation.Nullable;
 
 public class ScoreboardDisplaySlotFix extends DataFix {
 
-    private static final Map<String, String> SLOT_RENAMES = ImmutableMap.builder().put("slot_0", "list").put("slot_1", "sidebar").put("slot_2", "below_name").put("slot_3", "sidebar.team.black").put("slot_4", "sidebar.team.dark_blue").put("slot_5", "sidebar.team.dark_green").put("slot_6", "sidebar.team.dark_aqua").put("slot_7", "sidebar.team.dark_red").put("slot_8", "sidebar.team.dark_purple").put("slot_9", "sidebar.team.gold").put("slot_10", "sidebar.team.gray").put("slot_11", "sidebar.team.dark_gray").put("slot_12", "sidebar.team.blue").put("slot_13", "sidebar.team.green").put("slot_14", "sidebar.team.aqua").put("slot_15", "sidebar.team.red").put("slot_16", "sidebar.team.light_purple").put("slot_17", "sidebar.team.yellow").put("slot_18", "sidebar.team.white").build();
+    private static final Map<String, String> SLOT_RENAMES = ImmutableMap.<String, String>builder().put("slot_0", "list").put("slot_1", "sidebar").put("slot_2", "below_name").put("slot_3", "sidebar.team.black").put("slot_4", "sidebar.team.dark_blue").put("slot_5", "sidebar.team.dark_green").put("slot_6", "sidebar.team.dark_aqua").put("slot_7", "sidebar.team.dark_red").put("slot_8", "sidebar.team.dark_purple").put("slot_9", "sidebar.team.gold").put("slot_10", "sidebar.team.gray").put("slot_11", "sidebar.team.dark_gray").put("slot_12", "sidebar.team.blue").put("slot_13", "sidebar.team.green").put("slot_14", "sidebar.team.aqua").put("slot_15", "sidebar.team.red").put("slot_16", "sidebar.team.light_purple").put("slot_17", "sidebar.team.yellow").put("slot_18", "sidebar.team.white").build();
 
     public ScoreboardDisplaySlotFix(Schema schema) {
         super(schema, false);
@@ -37,10 +37,10 @@ public class ScoreboardDisplaySlotFix extends DataFix {
                     return dynamic.update("DisplaySlots", (dynamic1) -> {
                         return dynamic1.updateMapValues((pair) -> {
                             return pair.mapFirst((dynamic2) -> {
-                                Optional optional = dynamic2.asString().result().map(ScoreboardDisplaySlotFix::rename);
+                                Optional<String> optional = dynamic2.asString().result().map(ScoreboardDisplaySlotFix::rename);
 
                                 Objects.requireNonNull(dynamic2);
-                                return (Dynamic) DataFixUtils.orElse(optional.map(dynamic2::createString), dynamic2);
+                                return DataFixUtils.orElse(optional.map(dynamic2::createString), dynamic2);
                             });
                         });
                     });
