diff --git a/minecraft/resources/RegistryOps.java b/minecraft/resources/RegistryOps.java
index 8289a67..0eff35a 100644
--- a/minecraft/resources/RegistryOps.java
+++ b/minecraft/resources/RegistryOps.java
@@ -25,11 +25,7 @@ public class RegistryOps<T> extends DelegatingOps<T> {
 
             @Override
             public <T> Optional<RegistryOps.RegistryInfo<T>> lookup(ResourceKey<? extends Registry<? extends T>> resourcekey) {
-                Map map = this.lookups;
-                RegistryOps.RegistryInfoLookup registryops_registryinfolookup1 = registryops_registryinfolookup;
-
-                Objects.requireNonNull(registryops_registryinfolookup);
-                return (Optional) map.computeIfAbsent(resourcekey, registryops_registryinfolookup1::lookup);
+                return (Optional<RegistryInfo<T>>) this.lookups.computeIfAbsent(resourcekey, registryops_registryinfolookup::lookup);
             }
         };
     }
