diff --git a/minecraft/util/SpawnUtil.java b/minecraft/util/SpawnUtil.java
index d0ef75e..21612a9 100644
--- a/minecraft/util/SpawnUtil.java
+++ b/minecraft/util/SpawnUtil.java
@@ -29,7 +29,7 @@ public class SpawnUtil {
 
             blockpos_mutableblockpos.setWithOffset(blockpos, i1, k, j1);
             if (serverlevel.getWorldBorder().isWithinBounds((BlockPos) blockpos_mutableblockpos) && moveToPossibleSpawnPosition(serverlevel, k, blockpos_mutableblockpos, spawnutil_strategy)) {
-                T t0 = (Mob) entitytype.create(serverlevel, (CompoundTag) null, (Consumer) null, blockpos_mutableblockpos, mobspawntype, false, false);
+                T t0 = entitytype.create(serverlevel, null, null, blockpos_mutableblockpos, mobspawntype, false, false);
 
                 if (t0 != null) {
                     if (t0.checkSpawnRules(serverlevel, mobspawntype) && t0.checkSpawnObstruction(serverlevel)) {
