diff --git a/minecraft/world/level/storage/loot/predicates/ConditionUserBuilder.java b/minecraft/world/level/storage/loot/predicates/ConditionUserBuilder.java
index 4bd28e7..8366dcb 100644
--- a/minecraft/world/level/storage/loot/predicates/ConditionUserBuilder.java
+++ b/minecraft/world/level/storage/loot/predicates/ConditionUserBuilder.java
@@ -12,7 +12,7 @@ public interface ConditionUserBuilder<T extends ConditionUserBuilder<T>> {
 
         Object object;
 
-        for (Iterator iterator = iterable.iterator(); iterator.hasNext(); t0 = t0.when((LootItemCondition.Builder) function.apply(object))) {
+        for (Iterator iterator = iterable.iterator(); iterator.hasNext(); t0 = t0.when((LootItemCondition.Builder) function.apply((E) object))) {
             object = iterator.next();
         }
 
