diff --git a/minecraft/world/item/CreativeModeTabs.java b/minecraft/world/item/CreativeModeTabs.java
index fe4c737..2091d71 100644
--- a/minecraft/world/item/CreativeModeTabs.java
+++ b/minecraft/world/item/CreativeModeTabs.java
@@ -56,7 +56,7 @@ public class CreativeModeTabs {
     private static final ResourceKey<CreativeModeTab> SPAWN_EGGS = createKey("spawn_eggs");
     private static final ResourceKey<CreativeModeTab> OP_BLOCKS = createKey("op_blocks");
     private static final ResourceKey<CreativeModeTab> INVENTORY = createKey("inventory");
-    private static final Comparator<Holder<PaintingVariant>> PAINTING_COMPARATOR = Comparator.comparing(Holder::value, Comparator.comparingInt((paintingvariant) -> {
+    private static final Comparator<Holder<PaintingVariant>> PAINTING_COMPARATOR = Comparator.comparing(Holder::value, Comparator.<PaintingVariant>comparingInt((paintingvariant) -> {
         return paintingvariant.getHeight() * paintingvariant.getWidth();
     }).thenComparing(PaintingVariant::getWidth));
     @Nullable
