diff --git a/minecraft/world/entity/animal/goat/Goat.java b/minecraft/world/entity/animal/goat/Goat.java
index 290298c..cc7c402 100644
--- a/minecraft/world/entity/animal/goat/Goat.java
+++ b/minecraft/world/entity/animal/goat/Goat.java
@@ -181,7 +181,7 @@ public class Goat extends Animal {
 
     @Override
     public Brain<Goat> getBrain() {
-        return super.getBrain();
+        return (Brain<Goat>) super.getBrain();
     }
 
     @Override
