diff --git a/minecraft/world/level/levelgen/placement/PlacedFeature.java b/minecraft/world/level/levelgen/placement/PlacedFeature.java
index 6fce493..816e9a1 100644
--- a/minecraft/world/level/levelgen/placement/PlacedFeature.java
+++ b/minecraft/world/level/levelgen/placement/PlacedFeature.java
@@ -42,12 +42,10 @@ public record PlacedFeature(Holder<ConfiguredFeature<?, ?>> feature, List<Placem
     private boolean placeWithContext(PlacementContext placementcontext, RandomSource randomsource, BlockPos blockpos) {
         Stream<BlockPos> stream = Stream.of(blockpos);
 
-        PlacementModifier placementmodifier;
-
-        for (Iterator iterator = this.placement.iterator(); iterator.hasNext();stream = stream.flatMap((blockpos1) -> {
-            return placementmodifier.getPositions(placementcontext, randomsource, blockpos1);
-        })) {
-            placementmodifier = (PlacementModifier) iterator.next();
+        for (PlacementModifier placementmodifier : this.placement) {
+            stream = stream.flatMap((blockpos1) -> {
+                return placementmodifier.getPositions(placementcontext, randomsource, blockpos1);
+            });
         }
 
         ConfiguredFeature<?, ?> configuredfeature = (ConfiguredFeature) this.feature.value();
