diff --git a/minecraft/server/players/IpBanList.java b/minecraft/server/players/IpBanList.java
index d3ad457..6b261eb 100644
--- a/minecraft/server/players/IpBanList.java
+++ b/minecraft/server/players/IpBanList.java
@@ -30,7 +30,7 @@ public class IpBanList extends StoredUserList<String, IpBanListEntry> {
     public IpBanListEntry get(SocketAddress socketaddress) {
         String s = this.getIpFromAddress(socketaddress);
 
-        return (IpBanListEntry) this.get((Object) s);
+        return (IpBanListEntry) this.get(s);
     }
 
     private String getIpFromAddress(SocketAddress socketaddress) {
