diff --git a/minecraft/util/datafix/schemas/V99.java b/minecraft/util/datafix/schemas/V99.java
index 470267c..dc40c22 100644
--- a/minecraft/util/datafix/schemas/V99.java
+++ b/minecraft/util/datafix/schemas/V99.java
@@ -55,7 +55,7 @@ public class V99 extends Schema {
     });
     protected static final HookFunction ADD_NAMES = new HookFunction() {
         public <T> T apply(DynamicOps<T> dynamicops, T t0) {
-            return V99.addNames(new Dynamic(dynamicops, t0), V99.ITEM_TO_BLOCKENTITY, "ArmorStand");
+            return V99.addNames(new Dynamic<>(dynamicops, t0), V99.ITEM_TO_BLOCKENTITY, "ArmorStand");
         }
     };
 
