diff --git a/minecraft/world/level/block/grower/AbstractTreeGrower.java b/minecraft/world/level/block/grower/AbstractTreeGrower.java
index 7e79896..400c706 100644
--- a/minecraft/world/level/block/grower/AbstractTreeGrower.java
+++ b/minecraft/world/level/block/grower/AbstractTreeGrower.java
@@ -27,7 +27,7 @@ public abstract class AbstractTreeGrower {
         if (resourcekey == null) {
             return false;
         } else {
-            Holder<ConfiguredFeature<?, ?>> holder = (Holder) serverlevel.registryAccess().registryOrThrow(Registries.CONFIGURED_FEATURE).getHolder(resourcekey).orElse((Object) null);
+            Holder<ConfiguredFeature<?, ?>> holder = serverlevel.registryAccess().registryOrThrow(Registries.CONFIGURED_FEATURE).getHolder(resourcekey).orElse(null);
 
             if (holder == null) {
                 return false;
