diff --git a/minecraft/world/level/block/grower/AbstractMegaTreeGrower.java b/minecraft/world/level/block/grower/AbstractMegaTreeGrower.java
index 6ccc9fb..dadc29b 100644
--- a/minecraft/world/level/block/grower/AbstractMegaTreeGrower.java
+++ b/minecraft/world/level/block/grower/AbstractMegaTreeGrower.java
@@ -40,7 +40,7 @@ public abstract class AbstractMegaTreeGrower extends AbstractTreeGrower {
         if (resourcekey == null) {
             return false;
         } else {
-            Holder<ConfiguredFeature<?, ?>> holder = (Holder) serverlevel.registryAccess().registryOrThrow(Registries.CONFIGURED_FEATURE).getHolder(resourcekey).orElse((Object) null);
+            Holder<ConfiguredFeature<?, ?>> holder = (Holder) serverlevel.registryAccess().registryOrThrow(Registries.CONFIGURED_FEATURE).getHolder(resourcekey).orElse(null);
 
             if (holder == null) {
                 return false;
