diff --git a/minecraft/world/level/block/entity/BannerPattern.java b/minecraft/world/level/block/entity/BannerPattern.java
index 65cc7db..30c5c84 100644
--- a/minecraft/world/level/block/entity/BannerPattern.java
+++ b/minecraft/world/level/block/entity/BannerPattern.java
@@ -35,7 +35,7 @@ public class BannerPattern {
     public static Holder<BannerPattern> byHash(String s) {
         return (Holder) BuiltInRegistries.BANNER_PATTERN.holders().filter((holder_reference) -> {
             return ((BannerPattern) holder_reference.value()).hashname.equals(s);
-        }).findAny().orElse((Object) null);
+        }).findAny().orElse(null);
     }
 
     public static class Builder {
