diff --git a/minecraft/world/entity/monster/breeze/Breeze.java b/minecraft/world/entity/monster/breeze/Breeze.java
index 01d6ac3..b7459d1 100644
--- a/minecraft/world/entity/monster/breeze/Breeze.java
+++ b/minecraft/world/entity/monster/breeze/Breeze.java
@@ -65,7 +65,7 @@ public class Breeze extends Monster {
 
     @Override
     public Brain<Breeze> getBrain() {
-        return super.getBrain();
+        return (Brain<Breeze>) super.getBrain();
     }
 
     @Override
