diff --git a/minecraft/world/entity/monster/breeze/LongJump.java b/minecraft/world/entity/monster/breeze/LongJump.java
index 5c9d70f..6fd84d1 100644
--- a/minecraft/world/entity/monster/breeze/LongJump.java
+++ b/minecraft/world/entity/monster/breeze/LongJump.java
@@ -54,7 +54,7 @@ public class LongJump extends Behavior<Breeze> {
         } else if (breeze.getBrain().checkMemory(MemoryModuleType.BREEZE_JUMP_TARGET, MemoryStatus.VALUE_PRESENT)) {
             return true;
         } else {
-            LivingEntity livingentity = (LivingEntity) breeze.getBrain().getMemory(MemoryModuleType.ATTACK_TARGET).orElse((Object) null);
+            LivingEntity livingentity = (LivingEntity) breeze.getBrain().getMemory(MemoryModuleType.ATTACK_TARGET).orElse(null);
 
             if (livingentity == null) {
                 return false;
@@ -73,7 +73,7 @@ public class LongJump extends Behavior<Breeze> {
                 } else if (!BreezeUtil.hasLineOfSight(breeze, blockpos.getCenter()) && !BreezeUtil.hasLineOfSight(breeze, blockpos.above(4).getCenter())) {
                     return false;
                 } else {
-                    breeze.getBrain().setMemory(MemoryModuleType.BREEZE_JUMP_TARGET, (Object) blockpos);
+                    breeze.getBrain().setMemory(MemoryModuleType.BREEZE_JUMP_TARGET, blockpos);
                     return true;
                 }
             }
@@ -106,7 +106,7 @@ public class LongJump extends Behavior<Breeze> {
         if (isFinishedInhaling(breeze)) {
             Vec3 vec3 = (Vec3) breeze.getBrain().getMemory(MemoryModuleType.BREEZE_JUMP_TARGET).flatMap((blockpos) -> {
                 return calculateOptimalJumpVector(breeze, breeze.getRandom(), Vec3.atBottomCenterOf(blockpos));
-            }).orElse((Object) null);
+            }).orElse(null);
 
             if (vec3 == null) {
                 breeze.setPose(Pose.STANDING);
@@ -114,7 +114,7 @@ public class LongJump extends Behavior<Breeze> {
             }
 
             if (flag) {
-                breeze.getBrain().setMemory(MemoryModuleType.BREEZE_LEAVING_WATER, (Object) Unit.INSTANCE);
+                breeze.getBrain().setMemory(MemoryModuleType.BREEZE_LEAVING_WATER, Unit.INSTANCE);
             }
 
             breeze.playSound(SoundEvents.BREEZE_JUMP, 1.0F, 1.0F);
