diff --git a/minecraft/world/entity/monster/breeze/Slide.java b/minecraft/world/entity/monster/breeze/Slide.java
index 57f5ef4..3def9b2 100644
--- a/minecraft/world/entity/monster/breeze/Slide.java
+++ b/minecraft/world/entity/monster/breeze/Slide.java
@@ -25,7 +25,7 @@ public class Slide extends Behavior<Breeze> {
     }
 
     protected void start(ServerLevel serverlevel, Breeze breeze, long i) {
-        LivingEntity livingentity = (LivingEntity) breeze.getBrain().getMemory(MemoryModuleType.ATTACK_TARGET).orElse((Object) null);
+        LivingEntity livingentity = (LivingEntity) breeze.getBrain().getMemory(MemoryModuleType.ATTACK_TARGET).orElse(null);
 
         if (livingentity != null) {
             boolean flag = breeze.withinInnerCircleRange(livingentity.position());
@@ -43,7 +43,7 @@ public class Slide extends Behavior<Breeze> {
                 vec3 = breeze.getRandom().nextBoolean() ? BreezeUtil.randomPointBehindTarget(livingentity, breeze.getRandom()) : randomPointInMiddleCircle(breeze, livingentity);
             }
 
-            breeze.getBrain().setMemory(MemoryModuleType.WALK_TARGET, (Object) (new WalkTarget(BlockPos.containing(vec3), 0.6F, 1)));
+            breeze.getBrain().setMemory(MemoryModuleType.WALK_TARGET, (new WalkTarget(BlockPos.containing(vec3), 0.6F, 1)));
         }
     }
 
