diff --git a/minecraft/network/chat/numbers/NumberFormatTypes.java b/minecraft/network/chat/numbers/NumberFormatTypes.java
index f6c88eb..96c93ed 100644
--- a/minecraft/network/chat/numbers/NumberFormatTypes.java
+++ b/minecraft/network/chat/numbers/NumberFormatTypes.java
@@ -25,7 +25,7 @@ public class NumberFormatTypes {
     }
 
     public static <T extends NumberFormat> void writeToStream(FriendlyByteBuf friendlybytebuf, T t0) {
-        NumberFormatType<T> numberformattype = t0.type();
+        NumberFormatType<T> numberformattype = (NumberFormatType<T>) t0.type();
 
         friendlybytebuf.writeId(BuiltInRegistries.NUMBER_FORMAT_TYPE, numberformattype);
         numberformattype.writeToStream(friendlybytebuf, t0);
