diff --git a/minecraft/gametest/framework/StructureUtils.java b/minecraft/gametest/framework/StructureUtils.java
index 960ebcd..8eedf92 100644
--- a/minecraft/gametest/framework/StructureUtils.java
+++ b/minecraft/gametest/framework/StructureUtils.java
@@ -205,7 +205,7 @@ public class StructureUtils {
         Collection<BlockPos> collection = findStructureBlocks(blockpos, i, serverlevel);
         Optional<BlockPos> optional = collection.stream().min(comparator);
 
-        return (BlockPos) optional.orElse((Object) null);
+        return (BlockPos) optional.orElse(null);
     }
 
     public static Collection<BlockPos> findStructureBlocks(BlockPos blockpos, int i, ServerLevel serverlevel) {
