diff --git a/minecraft/world/item/BoatItem.java b/minecraft/world/item/BoatItem.java
index 12d7498..28f5518 100644
--- a/minecraft/world/item/BoatItem.java
+++ b/minecraft/world/item/BoatItem.java
@@ -86,7 +86,7 @@ public class BoatItem extends Item {
 
     private Boat getBoat(Level level, HitResult hitresult, ItemStack itemstack, Player player) {
         Vec3 vec3 = hitresult.getLocation();
-        Object object = this.hasChest ? new ChestBoat(level, vec3.x, vec3.y, vec3.z) : new Boat(level, vec3.x, vec3.y, vec3.z);
+        Entity object = this.hasChest ? new ChestBoat(level, vec3.x, vec3.y, vec3.z) : new Boat(level, vec3.x, vec3.y, vec3.z);
 
         if (level instanceof ServerLevel) {
             ServerLevel serverlevel = (ServerLevel) level;
