diff --git a/minecraft/world/item/BundleItem.java b/minecraft/world/item/BundleItem.java
index 1fc9491..9d42c20 100644
--- a/minecraft/world/item/BundleItem.java
+++ b/minecraft/world/item/BundleItem.java
@@ -162,7 +162,7 @@ public class BundleItem extends Item {
         if (itemstack.is(Items.BUNDLE)) {
             return Optional.empty();
         } else {
-            Stream stream = listtag.stream();
+            Stream<Tag> stream = listtag.stream();
 
             Objects.requireNonNull(CompoundTag.class);
             stream = stream.filter(CompoundTag.class::isInstance);
@@ -249,7 +249,7 @@ public class BundleItem extends Item {
             return Stream.empty();
         } else {
             ListTag listtag = compoundtag.getList("Items", 10);
-            Stream stream = listtag.stream();
+            Stream<Tag> stream = listtag.stream();
 
             Objects.requireNonNull(CompoundTag.class);
             return stream.map(CompoundTag.class::cast).map(ItemStack::of);
@@ -259,7 +259,7 @@ public class BundleItem extends Item {
     @Override
     public Optional<TooltipComponent> getTooltipImage(ItemStack itemstack) {
         NonNullList<ItemStack> nonnulllist = NonNullList.create();
-        Stream stream = getContents(itemstack);
+        Stream<ItemStack> stream = getContents(itemstack);
 
         Objects.requireNonNull(nonnulllist);
         stream.forEach(nonnulllist::add);
