diff --git a/minecraft/world/item/CompassItem.java b/minecraft/world/item/CompassItem.java
index 6cde2bc..245acbf 100644
--- a/minecraft/world/item/CompassItem.java
+++ b/minecraft/world/item/CompassItem.java
@@ -10,6 +10,7 @@ import net.minecraft.core.GlobalPos;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.nbt.NbtOps;
 import net.minecraft.nbt.NbtUtils;
+import net.minecraft.nbt.Tag;
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.sounds.SoundEvents;
@@ -132,7 +133,7 @@ public class CompassItem extends Item implements Vanishable {
 
     private void addLodestoneTags(ResourceKey<Level> resourcekey, BlockPos blockpos, CompoundTag compoundtag) {
         compoundtag.put("LodestonePos", NbtUtils.writeBlockPos(blockpos));
-        DataResult dataresult = Level.RESOURCE_KEY_CODEC.encodeStart(NbtOps.INSTANCE, resourcekey);
+        DataResult<Tag> dataresult = Level.RESOURCE_KEY_CODEC.encodeStart(NbtOps.INSTANCE, resourcekey);
         Logger logger = CompassItem.LOGGER;
 
         Objects.requireNonNull(logger);
