diff --git a/minecraft/world/item/SpawnEggItem.java b/minecraft/world/item/SpawnEggItem.java
index 2130f55..a0e8251 100644
--- a/minecraft/world/item/SpawnEggItem.java
+++ b/minecraft/world/item/SpawnEggItem.java
@@ -164,7 +164,7 @@ public class SpawnEggItem extends Item {
         if (!this.spawnsEntity(itemstack.getTag(), entitytype)) {
             return Optional.empty();
         } else {
-            Object object;
+            Mob object;
 
             if (mob instanceof AgeableMob) {
                 object = ((AgeableMob) mob).getBreedOffspring(serverlevel, (AgeableMob) mob);
