diff --git a/minecraft/world/scores/Scoreboard.java b/minecraft/world/scores/Scoreboard.java
index 58d400a..75b0e32 100644
--- a/minecraft/world/scores/Scoreboard.java
+++ b/minecraft/world/scores/Scoreboard.java
@@ -52,7 +52,7 @@ public class Scoreboard {
     }
 
     public final void forAllObjectives(ObjectiveCriteria objectivecriteria, String s, Consumer<Score> consumer) {
-        ((List) this.objectivesByCriteria.getOrDefault(objectivecriteria, Collections.emptyList())).forEach((objective) -> {
+        (this.objectivesByCriteria.getOrDefault(objectivecriteria, Collections.emptyList())).forEach((objective) -> {
             consumer.accept(this.getOrCreatePlayerScore(s, objective));
         });
     }
