diff --git a/minecraft/world/entity/vehicle/AbstractMinecart.java b/minecraft/world/entity/vehicle/AbstractMinecart.java
index 746fc87..7102fcf 100644
--- a/minecraft/world/entity/vehicle/AbstractMinecart.java
+++ b/minecraft/world/entity/vehicle/AbstractMinecart.java
@@ -103,7 +103,7 @@ public abstract class AbstractMinecart extends VehicleEntity {
     }
 
     public static AbstractMinecart createMinecart(ServerLevel serverlevel, double d0, double d1, double d2, AbstractMinecart.Type abstractminecart_type, ItemStack itemstack, @Nullable Player player) {
-        Object object;
+        Entity object;
 
         switch (abstractminecart_type) {
             case CHEST:
@@ -128,7 +128,7 @@ public abstract class AbstractMinecart extends VehicleEntity {
                 object = new Minecart(serverlevel, d0, d1, d2);
         }
 
-        Object object1 = object;
+        Entity object1 = object;
 
         EntityType.createDefaultStackConfig(serverlevel, itemstack, player).accept(object1);
         return (AbstractMinecart) object1;
