diff --git a/minecraft/network/chat/numbers/StyledFormat.java b/minecraft/network/chat/numbers/StyledFormat.java
index bfdcc08..2508390 100644
--- a/minecraft/network/chat/numbers/StyledFormat.java
+++ b/minecraft/network/chat/numbers/StyledFormat.java
@@ -17,7 +17,7 @@ public class StyledFormat implements NumberFormat {
 
         @Override
         public MapCodec<StyledFormat> mapCodec() {
-            return null.CODEC;
+            return CODEC;
         }
 
         public void writeToStream(FriendlyByteBuf friendlybytebuf, StyledFormat styledformat) {
