diff --git a/minecraft/world/level/levelgen/carver/WorldCarver.java b/minecraft/world/level/levelgen/carver/WorldCarver.java
index 9b80c27..fd74ece 100644
--- a/minecraft/world/level/levelgen/carver/WorldCarver.java
+++ b/minecraft/world/level/levelgen/carver/WorldCarver.java
@@ -39,7 +39,7 @@ public abstract class WorldCarver<C extends CarverConfiguration> {
     private final Codec<ConfiguredWorldCarver<C>> configuredCodec;
 
     private static <C extends CarverConfiguration, F extends WorldCarver<C>> F register(String s, F f0) {
-        return (WorldCarver) Registry.register(BuiltInRegistries.CARVER, s, f0);
+        return Registry.register(BuiltInRegistries.CARVER, s, f0);
     }
 
     public WorldCarver(Codec<C> codec) {
