diff --git a/minecraft/data/recipes/ShapelessRecipeBuilder.java b/minecraft/data/recipes/ShapelessRecipeBuilder.java
index 27c4fe8..3c24dbc 100644
--- a/minecraft/data/recipes/ShapelessRecipeBuilder.java
+++ b/minecraft/data/recipes/ShapelessRecipeBuilder.java
@@ -91,7 +91,7 @@ public class ShapelessRecipeBuilder implements RecipeBuilder {
     public void save(RecipeOutput recipeoutput, ResourceLocation resourcelocation) {
         this.ensureValid(resourcelocation);
         Advancement.Builder advancement_builder = recipeoutput.advancement().addCriterion("has_the_recipe", RecipeUnlockedTrigger.unlocked(resourcelocation)).rewards(AdvancementRewards.Builder.recipe(resourcelocation)).requirements(AdvancementRequirements.Strategy.OR);
-        Map map = this.criteria;
+        Map<String, Criterion<?>> map = this.criteria;
 
         Objects.requireNonNull(advancement_builder);
         map.forEach(advancement_builder::addCriterion);
