diff --git a/minecraft/commands/Commands.java b/minecraft/commands/Commands.java
index 1c07856..4e01232 100644
--- a/minecraft/commands/Commands.java
+++ b/minecraft/commands/Commands.java
@@ -363,7 +363,7 @@ public class Commands {
 
                 executioncontext1.close();
             } finally {
-                Commands.CURRENT_EXECUTION_CONTEXT.set((Object) null);
+                Commands.CURRENT_EXECUTION_CONTEXT.set(null);
             }
         } else {
             consumer.accept(executioncontext);
