diff --git a/minecraft/commands/Commands.java b/minecraft/commands/Commands.java
index 6ab0e56..ae6838e 100644
--- a/minecraft/commands/Commands.java
+++ b/minecraft/commands/Commands.java
@@ -387,7 +387,7 @@ public class Commands {
             CommandNode<CommandSourceStack> commandnode2 = (CommandNode) iterator.next();
 
             if (commandnode2.canUse(commandsourcestack)) {
-                ArgumentBuilder<SharedSuggestionProvider, ?> argumentbuilder = commandnode2.createBuilder();
+                ArgumentBuilder<SharedSuggestionProvider, ?> argumentbuilder = (ArgumentBuilder) commandnode2.createBuilder();
 
                 argumentbuilder.requires((sharedsuggestionprovider) -> {
                     return true;
