diff --git a/minecraft/util/datafix/fixes/ItemSpawnEggFix.java b/minecraft/util/datafix/fixes/ItemSpawnEggFix.java
index a34b890..ea627e1 100644
--- a/minecraft/util/datafix/fixes/ItemSpawnEggFix.java
+++ b/minecraft/util/datafix/fixes/ItemSpawnEggFix.java
@@ -123,8 +123,9 @@ public class ItemSpawnEggFix extends DataFix {
                     Typed<?> typed2 = typed.getOrCreateTyped(opticfinder2);
                     Typed<?> typed3 = typed2.getOrCreateTyped(opticfinder3);
                     Typed<?> typed4 = typed3.getOrCreateTyped(opticfinder4);
+                    Dynamic<?> finalDynamic = dynamic;
                     Typed<?> typed5 = (Typed) ((Pair) typed4.write().flatMap((dynamic1) -> {
-                        return type1.readTyped(dynamic1.set("id", dynamic.createString(s)));
+                        return type1.readTyped(dynamic1.set("id", finalDynamic.createString(s)));
                     }).result().orElseThrow(() -> {
                         return new IllegalStateException("Could not parse new entity");
                     })).getFirst();
