diff --git a/minecraft/util/datafix/fixes/NamedEntityWriteReadFix.java b/minecraft/util/datafix/fixes/NamedEntityWriteReadFix.java
index 019ac99..661f3b5 100644
--- a/minecraft/util/datafix/fixes/NamedEntityWriteReadFix.java
+++ b/minecraft/util/datafix/fixes/NamedEntityWriteReadFix.java
@@ -36,7 +36,7 @@ public abstract class NamedEntityWriteReadFix extends DataFix {
 
         return this.fixTypeEverywhereTyped(this.name, type, type2, (typed) -> {
             return typed.updateTyped(opticfinder, type3, (typed1) -> {
-                DataResult dataresult = typed1.write().map(this::fix);
+                DataResult<Dynamic> dataresult = typed1.write().map(this::fix);
 
                 Objects.requireNonNull(type3);
                 return (Typed) ((Pair) Util.getOrThrow(dataresult.flatMap(type3::readTyped), (s) -> {
