diff --git a/minecraft/server/dedicated/Settings.java b/minecraft/server/dedicated/Settings.java
index c228496..3d9ccf4 100644
--- a/minecraft/server/dedicated/Settings.java
+++ b/minecraft/server/dedicated/Settings.java
@@ -125,7 +125,7 @@ public abstract class Settings<T extends Settings<T>> {
     private static <V extends Number> Function<String, V> wrapNumberDeserializer(Function<String, V> function) {
         return (s) -> {
             try {
-                return (Number) function.apply(s);
+                return function.apply(s);
             } catch (NumberFormatException numberformatexception) {
                 return null;
             }
@@ -172,7 +172,7 @@ public abstract class Settings<T extends Settings<T>> {
         V v1 = MoreObjects.firstNonNull(s1 != null ? function.apply(s1) : null, v0);
 
         this.properties.put(s, function1.apply(v1));
-        return new Settings.MutableValue<>(s, v1, function1);
+        return new Settings.MutableValue(s, v1, function1);
     }
 
     protected <V> V get(String s, Function<String, V> function, UnaryOperator<V> unaryoperator, Function<V, String> function1, V v0) {
