diff --git a/minecraft/util/datafix/fixes/EntityMinecartIdentifiersFix.java b/minecraft/util/datafix/fixes/EntityMinecartIdentifiersFix.java
index d22746b..ce5e1d5 100644
--- a/minecraft/util/datafix/fixes/EntityMinecartIdentifiersFix.java
+++ b/minecraft/util/datafix/fixes/EntityMinecartIdentifiersFix.java
@@ -23,8 +23,8 @@ public class EntityMinecartIdentifiersFix extends DataFix {
     }
 
     public TypeRewriteRule makeRule() {
-        TaggedChoiceType<String> taggedchoicetype = this.getInputSchema().findChoiceType(References.ENTITY);
-        TaggedChoiceType<String> taggedchoicetype1 = this.getOutputSchema().findChoiceType(References.ENTITY);
+        TaggedChoiceType<String> taggedchoicetype = (TaggedChoiceType<String>) this.getInputSchema().findChoiceType(References.ENTITY);
+        TaggedChoiceType<String> taggedchoicetype1 = (TaggedChoiceType<String>) this.getOutputSchema().findChoiceType(References.ENTITY);
 
         return this.fixTypeEverywhere("EntityMinecartIdentifiersFix", taggedchoicetype, taggedchoicetype1, (dynamicops) -> {
             return (pair) -> {
