diff --git a/minecraft/stats/ServerStatsCounter.java b/minecraft/stats/ServerStatsCounter.java
index 5bfd509..17259d6 100644
--- a/minecraft/stats/ServerStatsCounter.java
+++ b/minecraft/stats/ServerStatsCounter.java
@@ -158,13 +158,7 @@ public class ServerStatsCounter extends StatsCounter {
     }
 
     private <T> Optional<Stat<T>> getStat(StatType<T> stattype, String s) {
-        Optional optional = Optional.ofNullable(ResourceLocation.tryParse(s));
-        Registry registry = stattype.getRegistry();
-
-        Objects.requireNonNull(registry);
-        optional = optional.flatMap(registry::getOptional);
-        Objects.requireNonNull(stattype);
-        return optional.map(stattype::get);
+        return Optional.ofNullable(ResourceLocation.tryParse(s)).flatMap(stattype.getRegistry()::getOptional).map(stattype::get);
     }
 
     private static CompoundTag fromJson(JsonObject jsonobject) {
