diff --git a/minecraft/world/item/armortrim/ArmorTrim.java b/minecraft/world/item/armortrim/ArmorTrim.java
index 104e9da..9ee709e 100644
--- a/minecraft/world/item/armortrim/ArmorTrim.java
+++ b/minecraft/world/item/armortrim/ArmorTrim.java
@@ -112,7 +112,7 @@ public class ArmorTrim {
                     ArmorTrim.LOGGER.warn(s);
                 }
 
-            }).orElse((Object) null);
+            }).orElse(null);
 
             return Optional.ofNullable(armortrim);
         } else {
