diff --git a/minecraft/world/level/levelgen/feature/FossilFeature.java b/minecraft/world/level/levelgen/feature/FossilFeature.java
index 00cad64..2e6320b 100644
--- a/minecraft/world/level/levelgen/feature/FossilFeature.java
+++ b/minecraft/world/level/levelgen/feature/FossilFeature.java
@@ -16,6 +16,7 @@ import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.levelgen.Heightmap;
 import net.minecraft.world.level.levelgen.structure.BoundingBox;
 import net.minecraft.world.level.levelgen.structure.templatesystem.StructurePlaceSettings;
+import net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessor;
 import net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessorList;
 import net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate;
 import net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplateManager;
@@ -60,7 +61,7 @@ public class FossilFeature extends Feature<FossilFeatureConfiguration> {
             return false;
         } else {
             structureplacesettings.clearProcessors();
-            List list = ((StructureProcessorList) fossilfeatureconfiguration.fossilProcessors.value()).list();
+            List<StructureProcessor> list = ((StructureProcessorList) fossilfeatureconfiguration.fossilProcessors.value()).list();
 
             Objects.requireNonNull(structureplacesettings);
             list.forEach(structureplacesettings::addProcessor);
