diff --git a/minecraft/world/entity/ai/village/poi/PoiSection.java b/minecraft/world/entity/ai/village/poi/PoiSection.java
index 98ed12c..5af538e 100644
--- a/minecraft/world/entity/ai/village/poi/PoiSection.java
+++ b/minecraft/world/entity/ai/village/poi/PoiSection.java
@@ -33,7 +33,7 @@ public class PoiSection {
     private boolean isValid;
 
     public static Codec<PoiSection> codec(Runnable runnable) {
-        Codec codec = RecordCodecBuilder.create((instance) -> {
+        Codec codec = RecordCodecBuilder.<PoiSection>create((instance) -> {
             return instance.group(RecordCodecBuilder.point(runnable), Codec.BOOL.optionalFieldOf("Valid", false).forGetter((poisection) -> {
                 return poisection.isValid;
             }), PoiRecord.codec(runnable).listOf().fieldOf("Records").forGetter((poisection) -> {
