diff --git a/minecraft/world/scores/Scoreboard.java b/minecraft/world/scores/Scoreboard.java
index 4378d94..0b6bcbe 100644
--- a/minecraft/world/scores/Scoreboard.java
+++ b/minecraft/world/scores/Scoreboard.java
@@ -59,7 +59,7 @@ public class Scoreboard {
     }
 
     public final void forAllObjectives(ObjectiveCriteria objectivecriteria, ScoreHolder scoreholder, Consumer<ScoreAccess> consumer) {
-        ((List) this.objectivesByCriteria.getOrDefault(objectivecriteria, Collections.emptyList())).forEach((objective) -> {
+        (this.objectivesByCriteria.getOrDefault(objectivecriteria, Collections.emptyList())).forEach((objective) -> {
             consumer.accept(this.getOrCreatePlayerScore(scoreholder, objective, true));
         });
     }
