diff --git a/src/main/java/net/minecraft/world/level/storage/loot/LootTable.java b/src/main/java/net/minecraft/world/level/storage/loot/LootTable.java
index 8253d23..5bf2617 100644
--- a/src/main/java/net/minecraft/world/level/storage/loot/LootTable.java
+++ b/src/main/java/net/minecraft/world/level/storage/loot/LootTable.java
@@ -154,6 +154,12 @@ public class LootTable {
         List<Integer> list = this.getAvailableSlots(container, randomsource);
 
         this.shuffleAndSplitItems(objectarraylist, list.size(), randomsource);
+        // dish
+        // TODO: 11/15/2023 Make container class and allow the plugin to modify the items in the container
+        net.ouja.api.event.world.level.LootGenerateEvent event = new net.ouja.api.event.world.level.LootGenerateEvent(net.ouja.dish.world.item.DishItemStack.convertFromVanilla(objectarraylist));
+        net.ouja.dish.plugins.RegisteredEvents.callEvent(event);
+        if (event.isCancelled()) return;
+        // dish end
         ObjectListIterator objectlistiterator = objectarraylist.iterator();
 
         while (objectlistiterator.hasNext()) {
