diff --git a/minecraft/world/entity/ai/behavior/AnimalMakeLove.java b/minecraft/world/entity/ai/behavior/AnimalMakeLove.java
index eef9ef6..f5548d3 100644
--- a/minecraft/world/entity/ai/behavior/AnimalMakeLove.java
+++ b/minecraft/world/entity/ai/behavior/AnimalMakeLove.java
@@ -34,8 +34,8 @@ public class AnimalMakeLove extends Behavior<Animal> {
     protected void start(ServerLevel serverlevel, Animal animal, long i) {
         Animal animal1 = (Animal) this.findValidBreedPartner(animal).get();
 
-        animal.getBrain().setMemory(MemoryModuleType.BREED_TARGET, (Object) animal1);
-        animal1.getBrain().setMemory(MemoryModuleType.BREED_TARGET, (Object) animal);
+        animal.getBrain().setMemory(MemoryModuleType.BREED_TARGET, animal1);
+        animal1.getBrain().setMemory(MemoryModuleType.BREED_TARGET, animal);
         BehaviorUtils.lockGazeAndWalkToEachOther(animal, animal1, this.speedModifier);
         int j = 60 + animal.getRandom().nextInt(50);
 
