diff --git a/minecraft/world/entity/ai/behavior/AnimalPanic.java b/minecraft/world/entity/ai/behavior/AnimalPanic.java
index 1b398cc..f693db0 100644
--- a/minecraft/world/entity/ai/behavior/AnimalPanic.java
+++ b/minecraft/world/entity/ai/behavior/AnimalPanic.java
@@ -49,7 +49,7 @@ public class AnimalPanic extends Behavior<PathfinderMob> {
     }
 
     protected void start(ServerLevel serverlevel, PathfinderMob pathfindermob, long i) {
-        pathfindermob.getBrain().setMemory(MemoryModuleType.IS_PANICKING, (Object) true);
+        pathfindermob.getBrain().setMemory(MemoryModuleType.IS_PANICKING, true);
         pathfindermob.getBrain().eraseMemory(MemoryModuleType.WALK_TARGET);
     }
 
@@ -64,7 +64,7 @@ public class AnimalPanic extends Behavior<PathfinderMob> {
             Vec3 vec3 = this.getPanicPos(pathfindermob, serverlevel);
 
             if (vec3 != null) {
-                pathfindermob.getBrain().setMemory(MemoryModuleType.WALK_TARGET, (Object) (new WalkTarget(vec3, this.speedMultiplier, 0)));
+                pathfindermob.getBrain().setMemory(MemoryModuleType.WALK_TARGET, (new WalkTarget(vec3, this.speedMultiplier, 0)));
             }
         }
 
@@ -93,13 +93,13 @@ public class AnimalPanic extends Behavior<PathfinderMob> {
 
             if (Mth.ceil(entity.getBbWidth()) == 2) {
                 predicate = (blockpos1) -> {
-                    return BlockPos.squareOutSouthEast(blockpos1).allMatch((blockpos2) -> {
+                    return BlockPos.squareOutSouthEast((BlockPos) blockpos1).allMatch((blockpos2) -> {
                         return blockgetter.getFluidState(blockpos2).is(FluidTags.WATER);
                     });
                 };
             } else {
                 predicate = (blockpos1) -> {
-                    return blockgetter.getFluidState(blockpos1).is(FluidTags.WATER);
+                    return blockgetter.getFluidState((BlockPos) blockpos1).is(FluidTags.WATER);
                 };
             }
 
