diff --git a/minecraft/world/entity/ai/behavior/DismountOrSkipMounting.java b/minecraft/world/entity/ai/behavior/DismountOrSkipMounting.java
index 73d8e13..608dec5 100644
--- a/minecraft/world/entity/ai/behavior/DismountOrSkipMounting.java
+++ b/minecraft/world/entity/ai/behavior/DismountOrSkipMounting.java
@@ -15,7 +15,7 @@ public class DismountOrSkipMounting {
             return behaviorbuilder_instance.group(behaviorbuilder_instance.registered(MemoryModuleType.RIDE_TARGET)).apply(behaviorbuilder_instance, (memoryaccessor) -> {
                 return (serverlevel, livingentity, j) -> {
                     Entity entity = livingentity.getVehicle();
-                    Entity entity1 = (Entity) behaviorbuilder_instance.tryGet(memoryaccessor).orElse((Object) null);
+                    Entity entity1 = (Entity) behaviorbuilder_instance.tryGet(memoryaccessor).orElse(null);
 
                     if (entity == null && entity1 == null) {
                         return false;
