diff --git a/minecraft/world/entity/ai/behavior/PoiCompetitorScan.java b/minecraft/world/entity/ai/behavior/PoiCompetitorScan.java
index 0eb6f48..64cc557 100644
--- a/minecraft/world/entity/ai/behavior/PoiCompetitorScan.java
+++ b/minecraft/world/entity/ai/behavior/PoiCompetitorScan.java
@@ -22,7 +22,7 @@ public class PoiCompetitorScan {
                     GlobalPos globalpos = (GlobalPos) behaviorbuilder_instance.get(memoryaccessor);
 
                     serverlevel.getPoiManager().getType(globalpos.pos()).ifPresent((holder) -> {
-                        ((List) behaviorbuilder_instance.get(memoryaccessor1)).stream().filter((livingentity) -> {
+                        (behaviorbuilder_instance.get(memoryaccessor1)).stream().filter((livingentity) -> {
                             return livingentity instanceof Villager && livingentity != villager;
                         }).map((livingentity) -> {
                             return (Villager) livingentity;
