diff --git a/minecraft/world/entity/ai/behavior/RandomLookAround.java b/minecraft/world/entity/ai/behavior/RandomLookAround.java
index b6439fd..9d70dc9 100644
--- a/minecraft/world/entity/ai/behavior/RandomLookAround.java
+++ b/minecraft/world/entity/ai/behavior/RandomLookAround.java
@@ -35,7 +35,7 @@ public class RandomLookAround extends Behavior<Mob> {
         float f1 = Mth.wrapDegrees(mob.getYRot() + 2.0F * randomsource.nextFloat() * this.maxYaw - this.maxYaw);
         Vec3 vec3 = Vec3.directionFromRotation(f, f1);
 
-        mob.getBrain().setMemory(MemoryModuleType.LOOK_TARGET, (Object) (new BlockPosTracker(mob.getEyePosition().add(vec3))));
-        mob.getBrain().setMemory(MemoryModuleType.GAZE_COOLDOWN_TICKS, (Object) this.interval.sample(randomsource));
+        mob.getBrain().setMemory(MemoryModuleType.LOOK_TARGET, (new BlockPosTracker(mob.getEyePosition().add(vec3))));
+        mob.getBrain().setMemory(MemoryModuleType.GAZE_COOLDOWN_TICKS, this.interval.sample(randomsource));
     }
 }
