diff --git a/minecraft/world/entity/ai/behavior/UseBonemeal.java b/minecraft/world/entity/ai/behavior/UseBonemeal.java
index ca229af..f442b41 100644
--- a/minecraft/world/entity/ai/behavior/UseBonemeal.java
+++ b/minecraft/world/entity/ai/behavior/UseBonemeal.java
@@ -86,8 +86,8 @@ public class UseBonemeal extends Behavior<Villager> {
         this.cropPos.ifPresent((blockpos) -> {
             BlockPosTracker blockpostracker = new BlockPosTracker(blockpos);
 
-            villager.getBrain().setMemory(MemoryModuleType.LOOK_TARGET, (Object) blockpostracker);
-            villager.getBrain().setMemory(MemoryModuleType.WALK_TARGET, (Object) (new WalkTarget(blockpostracker, 0.5F, 1)));
+            villager.getBrain().setMemory(MemoryModuleType.LOOK_TARGET, blockpostracker);
+            villager.getBrain().setMemory(MemoryModuleType.WALK_TARGET, (new WalkTarget(blockpostracker, 0.5F, 1)));
         });
     }
 
