diff --git a/minecraft/world/entity/ai/behavior/WorkAtPoi.java b/minecraft/world/entity/ai/behavior/WorkAtPoi.java
index 32cb812..dec4732 100644
--- a/minecraft/world/entity/ai/behavior/WorkAtPoi.java
+++ b/minecraft/world/entity/ai/behavior/WorkAtPoi.java
@@ -35,9 +35,9 @@ public class WorkAtPoi extends Behavior<Villager> {
     protected void start(ServerLevel serverlevel, Villager villager, long i) {
         Brain<Villager> brain = villager.getBrain();
 
-        brain.setMemory(MemoryModuleType.LAST_WORKED_AT_POI, (Object) i);
+        brain.setMemory(MemoryModuleType.LAST_WORKED_AT_POI, i);
         brain.getMemory(MemoryModuleType.JOB_SITE).ifPresent((globalpos) -> {
-            brain.setMemory(MemoryModuleType.LOOK_TARGET, (Object) (new BlockPosTracker(globalpos.pos())));
+            brain.setMemory(MemoryModuleType.LOOK_TARGET, (new BlockPosTracker(globalpos.pos())));
         });
         villager.playWorkSound();
         this.useWorkstation(serverlevel, villager);
