diff --git a/minecraft/world/entity/ai/behavior/YieldJobSite.java b/minecraft/world/entity/ai/behavior/YieldJobSite.java
index ccd7c55..34e67e3 100644
--- a/minecraft/world/entity/ai/behavior/YieldJobSite.java
+++ b/minecraft/world/entity/ai/behavior/YieldJobSite.java
@@ -34,7 +34,7 @@ public class YieldJobSite {
                         if (optional.isEmpty()) {
                             return true;
                         } else {
-                            ((List) behaviorbuilder_instance.get(memoryaccessor2)).stream().filter((livingentity) -> {
+                            (behaviorbuilder_instance.get(memoryaccessor2)).stream().filter((livingentity) -> {
                                 return livingentity instanceof Villager && livingentity != villager;
                             }).map((livingentity) -> {
                                 return (Villager) livingentity;
@@ -46,7 +46,7 @@ public class YieldJobSite {
                                 memoryaccessor.erase();
                                 if (villager1.getBrain().getMemory(MemoryModuleType.JOB_SITE).isEmpty()) {
                                     BehaviorUtils.setWalkAndLookTargetMemories(villager1, blockpos, f, 1);
-                                    villager1.getBrain().setMemory(MemoryModuleType.POTENTIAL_JOB_SITE, (Object) GlobalPos.of(serverlevel.dimension(), blockpos));
+                                    villager1.getBrain().setMemory(MemoryModuleType.POTENTIAL_JOB_SITE, GlobalPos.of(serverlevel.dimension(), blockpos));
                                     DebugPackets.sendPoiTicketCountPacket(serverlevel, blockpos);
                                 }
 
