diff --git a/minecraft/data/models/blockstates/Condition.java b/minecraft/data/models/blockstates/Condition.java
index 2bc1f74..ecfec0d 100644
--- a/minecraft/data/models/blockstates/Condition.java
+++ b/minecraft/data/models/blockstates/Condition.java
@@ -119,7 +119,7 @@ public interface Condition extends Supplier<JsonElement> {
 
         public JsonElement get() {
             JsonArray jsonarray = new JsonArray();
-            Stream stream = this.subconditions.stream().map(Supplier::get);
+            Stream<JsonElement> stream = this.subconditions.stream().map(Supplier::get);
 
             Objects.requireNonNull(jsonarray);
             stream.forEach(jsonarray::add);
