diff --git a/minecraft/data/models/blockstates/MultiPartGenerator.java b/minecraft/data/models/blockstates/MultiPartGenerator.java
index cacb2d8..a9b3111 100644
--- a/minecraft/data/models/blockstates/MultiPartGenerator.java
+++ b/minecraft/data/models/blockstates/MultiPartGenerator.java
@@ -37,7 +37,7 @@ public class MultiPartGenerator implements BlockStateGenerator {
     }
 
     public MultiPartGenerator with(Variant variant) {
-        return this.with((List) ImmutableList.of(variant1));
+        return this.with((List) ImmutableList.of(variant));
     }
 
     public MultiPartGenerator with(Condition condition, List<Variant> list) {
@@ -50,7 +50,7 @@ public class MultiPartGenerator implements BlockStateGenerator {
     }
 
     public MultiPartGenerator with(Condition condition, Variant variant) {
-        return this.with(condition, (List) ImmutableList.of(variant1));
+        return this.with(condition, (List) ImmutableList.of(variant));
     }
 
     public JsonElement get() {
@@ -60,7 +60,7 @@ public class MultiPartGenerator implements BlockStateGenerator {
             multipartgenerator_entry.validate(statedefinition);
         });
         JsonArray jsonarray = new JsonArray();
-        Stream stream = this.parts.stream().map(MultiPartGenerator.Entry::get);
+        Stream<JsonElement> stream = this.parts.stream().map(MultiPartGenerator.Entry::get);
 
         Objects.requireNonNull(jsonarray);
         stream.forEach(jsonarray::add);
