diff --git a/minecraft/data/Main.java b/minecraft/data/Main.java
index 6864afe..2354b4f 100644
--- a/minecraft/data/Main.java
+++ b/minecraft/data/Main.java
@@ -93,7 +93,7 @@ public class Main {
 
     private static <T extends DataProvider> DataProvider.Factory<T> bindRegistries(BiFunction<PackOutput, CompletableFuture<HolderLookup.Provider>, T> bifunction, CompletableFuture<HolderLookup.Provider> completablefuture) {
         return (packoutput) -> {
-            return (DataProvider) bifunction.apply(packoutput, completablefuture);
+            return bifunction.apply(packoutput, completablefuture);
         };
     }
 
