diff --git a/minecraft/world/level/LevelAccessor.java b/minecraft/world/level/LevelAccessor.java
index f60e816..c3b87cb 100644
--- a/minecraft/world/level/LevelAccessor.java
+++ b/minecraft/world/level/LevelAccessor.java
@@ -35,11 +35,11 @@ public interface LevelAccessor extends CommonLevelAccessor, LevelTimeAccess {
 
     LevelTickAccess<Block> getBlockTicks();
 
-    private default <T> ScheduledTick<T> createTick(BlockPos blockpos, T t0, int i, TickPriority tickpriority) {
+    private <T> ScheduledTick<T> createTick(BlockPos blockpos, T t0, int i, TickPriority tickpriority) {
         return new ScheduledTick<>(t0, blockpos, this.getLevelData().getGameTime() + (long) i, tickpriority, this.nextSubTickCount());
     }
 
-    private default <T> ScheduledTick<T> createTick(BlockPos blockpos, T t0, int i) {
+    private <T> ScheduledTick<T> createTick(BlockPos blockpos, T t0, int i) {
         return new ScheduledTick<>(t0, blockpos, this.getLevelData().getGameTime() + (long) i, this.nextSubTickCount());
     }
 
