diff --git a/minecraft/server/level/Ticket.java b/minecraft/server/level/Ticket.java
index d771f2f..3033464 100644
--- a/minecraft/server/level/Ticket.java
+++ b/minecraft/server/level/Ticket.java
@@ -23,7 +23,7 @@ public final class Ticket<T> implements Comparable<Ticket<?>> {
         } else {
             int j = Integer.compare(System.identityHashCode(this.type), System.identityHashCode(ticket.type));
 
-            return j != 0 ? j : this.type.getComparator().compare(this.key, ticket.key);
+            return j != 0 ? j : this.type.getComparator().compare(this.key, (T) ticket.key);
         }
     }
 
