diff --git a/minecraft/core/NonNullList.java b/minecraft/core/NonNullList.java
index ba5205c..f27e3db 100644
--- a/minecraft/core/NonNullList.java
+++ b/minecraft/core/NonNullList.java
@@ -15,11 +15,11 @@ public class NonNullList<E> extends AbstractList<E> {
     private final E defaultValue;
 
     public static <E> NonNullList<E> create() {
-        return new NonNullList<>(Lists.newArrayList(), (Object) null);
+        return new NonNullList<>(Lists.newArrayList(), null);
     }
 
     public static <E> NonNullList<E> createWithCapacity(int i) {
-        return new NonNullList<>(Lists.newArrayListWithCapacity(i), (Object) null);
+        return new NonNullList<>(Lists.newArrayListWithCapacity(i), null);
     }
 
     public static <E> NonNullList<E> withSize(int i, E e0) {
@@ -27,7 +27,7 @@ public class NonNullList<E> extends AbstractList<E> {
         Object[] aobject = new Object[i];
 
         Arrays.fill(aobject, e0);
-        return new NonNullList<>(Arrays.asList(aobject), e0);
+        return (NonNullList<E>) new NonNullList<>(Arrays.asList(aobject), e0);
     }
 
     @SafeVarargs
