diff --git a/minecraft/world/level/storage/LevelStorageSource.java b/minecraft/world/level/storage/LevelStorageSource.java
index 77f751f..1a6024c 100644
--- a/minecraft/world/level/storage/LevelStorageSource.java
+++ b/minecraft/world/level/storage/LevelStorageSource.java
@@ -154,7 +154,7 @@ public class LevelStorageSource {
     }
 
     public static WorldDataConfiguration readDataConfig(Dynamic<?> dynamic) {
-        DataResult dataresult = WorldDataConfiguration.CODEC.parse(dynamic);
+        DataResult<WorldDataConfiguration> dataresult = WorldDataConfiguration.CODEC.parse(dynamic);
         Logger logger = LevelStorageSource.LOGGER;
 
         Objects.requireNonNull(logger);
@@ -196,7 +196,7 @@ public class LevelStorageSource {
             throw new LevelStorageException(Component.translatable("selectWorld.load_folder_access"));
         } else {
             try {
-                Stream stream = Files.list(this.baseDir);
+                Stream<Path> stream = Files.list(this.baseDir);
 
                 LevelStorageSource.LevelCandidates levelstoragesource_levelcandidates;
 
@@ -534,7 +534,7 @@ public class LevelStorageSource {
         }
 
         public Path getLevelPath(LevelResource levelresource) {
-            Map map = this.resources;
+            Map<LevelResource, Path> map = this.resources;
             LevelStorageSource.LevelDirectory levelstoragesource_leveldirectory = this.levelDirectory;
 
             Objects.requireNonNull(this.levelDirectory);
@@ -627,12 +627,13 @@ public class LevelStorageSource {
                             object = exception2;
                         }
 
+                        Path finalPath = path1;
                         crashreportcategory.setDetail("Reading Exception", (Object) ((Throwable) object).toString());
                         crashreportcategory.setDetail("Uncompressed", () -> {
                             return Base64.getEncoder().encodeToString(NbtIo.writeToByteArray(compoundtag));
                         });
                         crashreportcategory.setDetail("Compressed saved", () -> {
-                            return Base64.getEncoder().encodeToString(Files.readAllBytes(path1));
+                            return Base64.getEncoder().encodeToString(Files.readAllBytes(finalPath));
                         });
                         crashreportcategory.setDetail("Compressed array", () -> {
                             return Base64.getEncoder().encodeToString(NbtIo.writeToByteArrayCompressed(compoundtag));
@@ -642,7 +643,7 @@ public class LevelStorageSource {
                         crashreportcategory.setDetail("Corrupted file", () -> {
                             Path path4 = this.levelDirectory.corruptedDataFile(localdatetime);
 
-                            Files.move(path1, path4);
+                            Files.move(finalPath, path4);
                             return path4.getFileName().toString();
                         });
                         crashreportcategory.setDetail("Raw file", () -> {
