diff --git a/minecraft/world/entity/ai/behavior/GateBehavior.java b/minecraft/world/entity/ai/behavior/GateBehavior.java
index ac38b14..05d8b08 100644
--- a/minecraft/world/entity/ai/behavior/GateBehavior.java
+++ b/minecraft/world/entity/ai/behavior/GateBehavior.java
@@ -96,7 +96,7 @@ public class GateBehavior<E extends LivingEntity> implements BehaviorControl<E>
         }).forEach((behaviorcontrol) -> {
             behaviorcontrol.doStop(serverlevel, e0, i);
         });
-        Set set = this.exitErasedMemories;
+        Set<MemoryModuleType<?>> set = this.exitErasedMemories;
         Brain brain = e0.getBrain();
 
         Objects.requireNonNull(brain);
