diff --git a/minecraft/world/entity/ai/behavior/HarvestFarmland.java b/minecraft/world/entity/ai/behavior/HarvestFarmland.java
index 67af348..1fa8d52 100644
--- a/minecraft/world/entity/ai/behavior/HarvestFarmland.java
+++ b/minecraft/world/entity/ai/behavior/HarvestFarmland.java
@@ -81,8 +81,8 @@ public class HarvestFarmland extends Behavior<Villager> {
 
     protected void start(ServerLevel serverlevel, Villager villager, long i) {
         if (i > this.nextOkStartTime && this.aboveFarmlandPos != null) {
-            villager.getBrain().setMemory(MemoryModuleType.LOOK_TARGET, (Object) (new BlockPosTracker(this.aboveFarmlandPos)));
-            villager.getBrain().setMemory(MemoryModuleType.WALK_TARGET, (Object) (new WalkTarget(new BlockPosTracker(this.aboveFarmlandPos), 0.5F, 1)));
+            villager.getBrain().setMemory(MemoryModuleType.LOOK_TARGET, (new BlockPosTracker(this.aboveFarmlandPos)));
+            villager.getBrain().setMemory(MemoryModuleType.WALK_TARGET, (new WalkTarget(new BlockPosTracker(this.aboveFarmlandPos), 0.5F, 1)));
         }
 
     }
@@ -141,8 +141,8 @@ public class HarvestFarmland extends Behavior<Villager> {
                     this.aboveFarmlandPos = this.getValidFarmland(serverlevel);
                     if (this.aboveFarmlandPos != null) {
                         this.nextOkStartTime = i + 20L;
-                        villager.getBrain().setMemory(MemoryModuleType.WALK_TARGET, (Object) (new WalkTarget(new BlockPosTracker(this.aboveFarmlandPos), 0.5F, 1)));
-                        villager.getBrain().setMemory(MemoryModuleType.LOOK_TARGET, (Object) (new BlockPosTracker(this.aboveFarmlandPos)));
+                        villager.getBrain().setMemory(MemoryModuleType.WALK_TARGET, (new WalkTarget(new BlockPosTracker(this.aboveFarmlandPos), 0.5F, 1)));
+                        villager.getBrain().setMemory(MemoryModuleType.LOOK_TARGET, (new BlockPosTracker(this.aboveFarmlandPos)));
                     }
                 }
             }
