diff --git a/minecraft/world/entity/ai/behavior/LongJumpMidJump.java b/minecraft/world/entity/ai/behavior/LongJumpMidJump.java
index 2902784..478ba73 100644
--- a/minecraft/world/entity/ai/behavior/LongJumpMidJump.java
+++ b/minecraft/world/entity/ai/behavior/LongJumpMidJump.java
@@ -42,6 +42,6 @@ public class LongJumpMidJump extends Behavior<Mob> {
         mob.setDiscardFriction(false);
         mob.setPose(Pose.STANDING);
         mob.getBrain().eraseMemory(MemoryModuleType.LONG_JUMP_MID_JUMP);
-        mob.getBrain().setMemory(MemoryModuleType.LONG_JUMP_COOLDOWN_TICKS, (Object) this.timeBetweenLongJumps.sample(serverlevel.random));
+        mob.getBrain().setMemory(MemoryModuleType.LONG_JUMP_COOLDOWN_TICKS, this.timeBetweenLongJumps.sample(serverlevel.random));
     }
 }
