diff --git a/minecraft/world/entity/ai/behavior/RamTarget.java b/minecraft/world/entity/ai/behavior/RamTarget.java
index 32ad198..1827ddb 100644
--- a/minecraft/world/entity/ai/behavior/RamTarget.java
+++ b/minecraft/world/entity/ai/behavior/RamTarget.java
@@ -63,7 +63,7 @@ public class RamTarget extends Behavior<Goat> {
         Vec3 vec3 = (Vec3) brain.getMemory(MemoryModuleType.RAM_TARGET).get();
 
         this.ramDirection = (new Vec3((double) blockpos.getX() - vec3.x(), 0.0D, (double) blockpos.getZ() - vec3.z())).normalize();
-        brain.setMemory(MemoryModuleType.WALK_TARGET, (Object) (new WalkTarget(vec3, this.speed, 0)));
+        brain.setMemory(MemoryModuleType.WALK_TARGET, (new WalkTarget(vec3, this.speed, 0)));
     }
 
     protected void tick(ServerLevel serverlevel, Goat goat, long i) {
@@ -113,7 +113,7 @@ public class RamTarget extends Behavior<Goat> {
 
     protected void finishRam(ServerLevel serverlevel, Goat goat) {
         serverlevel.broadcastEntityEvent(goat, (byte) 59);
-        goat.getBrain().setMemory(MemoryModuleType.RAM_COOLDOWN_TICKS, (Object) ((UniformInt) this.getTimeBetweenRams.apply(goat)).sample(serverlevel.random));
+        goat.getBrain().setMemory(MemoryModuleType.RAM_COOLDOWN_TICKS, ((UniformInt) this.getTimeBetweenRams.apply(goat)).sample(serverlevel.random));
         goat.getBrain().eraseMemory(MemoryModuleType.RAM_TARGET);
     }
 }
