diff --git a/minecraft/world/entity/ai/behavior/VillagerMakeLove.java b/minecraft/world/entity/ai/behavior/VillagerMakeLove.java
index 6aa4044..4462895 100644
--- a/minecraft/world/entity/ai/behavior/VillagerMakeLove.java
+++ b/minecraft/world/entity/ai/behavior/VillagerMakeLove.java
@@ -129,6 +129,6 @@ public class VillagerMakeLove extends Behavior<Villager> {
     private void giveBedToChild(ServerLevel serverlevel, Villager villager, BlockPos blockpos) {
         GlobalPos globalpos = GlobalPos.of(serverlevel.dimension(), blockpos);
 
-        villager.getBrain().setMemory(MemoryModuleType.HOME, (Object) globalpos);
+        villager.getBrain().setMemory(MemoryModuleType.HOME, globalpos);
     }
 }
