diff --git a/minecraft/world/level/gameevent/vibrations/VibrationInfo.java b/minecraft/world/level/gameevent/vibrations/VibrationInfo.java
index 678500f..19af3ba 100644
--- a/minecraft/world/level/gameevent/vibrations/VibrationInfo.java
+++ b/minecraft/world/level/gameevent/vibrations/VibrationInfo.java
@@ -23,7 +23,7 @@ public record VibrationInfo(Holder<GameEvent> gameEvent, float distance, Vec3 po
         }), UUIDUtil.CODEC.optionalFieldOf("projectile_owner").forGetter((vibrationinfo) -> {
             return Optional.ofNullable(vibrationinfo.projectileOwnerUuid());
         })).apply(instance, (holder, ofloat, vec3, optional, optional1) -> {
-            return new VibrationInfo(holder, ofloat, vec3, (UUID) optional.orElse((Object) null), (UUID) optional1.orElse((Object) null));
+            return new VibrationInfo(holder, ofloat, vec3, (UUID) optional.orElse(null), (UUID) optional1.orElse(null));
         });
     });
 
@@ -50,10 +50,7 @@ public record VibrationInfo(Holder<GameEvent> gameEvent, float distance, Vec3 po
 
     public Optional<Entity> getEntity(ServerLevel serverlevel) {
         return Optional.ofNullable(this.entity).or(() -> {
-            Optional optional = Optional.ofNullable(this.uuid);
-
-            Objects.requireNonNull(serverlevel);
-            return optional.map(serverlevel::getEntity);
+            return Optional.ofNullable(this.uuid).map(serverlevel::getEntity);
         });
     }
 
@@ -63,7 +60,7 @@ public record VibrationInfo(Holder<GameEvent> gameEvent, float distance, Vec3 po
         }).map((entity) -> {
             return (Projectile) entity;
         }).map(Projectile::getOwner).or(() -> {
-            Optional optional = Optional.ofNullable(this.projectileOwnerUuid);
+            Optional<UUID> optional = Optional.ofNullable(this.projectileOwnerUuid);
 
             Objects.requireNonNull(serverlevel);
             return optional.map(serverlevel::getEntity);
