diff --git a/minecraft/world/level/storage/loot/entries/AlternativesEntry.java b/minecraft/world/level/storage/loot/entries/AlternativesEntry.java
index 5cb2c82..6a317c0 100644
--- a/minecraft/world/level/storage/loot/entries/AlternativesEntry.java
+++ b/minecraft/world/level/storage/loot/entries/AlternativesEntry.java
@@ -76,7 +76,7 @@ public class AlternativesEntry extends CompositeEntryBase {
     }
 
     public static <E> AlternativesEntry.Builder alternatives(Collection<E> collection, Function<E, LootPoolEntryContainer.Builder<?>> function) {
-        Stream stream = collection.stream();
+        Stream<E> stream = collection.stream();
 
         Objects.requireNonNull(function);
         return new AlternativesEntry.Builder((LootPoolEntryContainer.Builder[]) stream.map(function::apply).toArray((i) -> {
