diff --git a/minecraft/util/thread/BlockableEventLoop.java b/minecraft/util/thread/BlockableEventLoop.java
index 0011a0b..a23b252 100644
--- a/minecraft/util/thread/BlockableEventLoop.java
+++ b/minecraft/util/thread/BlockableEventLoop.java
@@ -67,7 +67,7 @@ public abstract class BlockableEventLoop<R extends Runnable> implements Profiler
             return this.submitAsync(runnable);
         } else {
             runnable.run();
-            return CompletableFuture.completedFuture((Object) null);
+            return CompletableFuture.completedFuture(null);
         }
     }
 
@@ -110,14 +110,14 @@ public abstract class BlockableEventLoop<R extends Runnable> implements Profiler
     }
 
     public boolean pollTask() {
-        R r0 = (Runnable) this.pendingRunnables.peek();
+        R r0 = this.pendingRunnables.peek();
 
         if (r0 == null) {
             return false;
         } else if (this.blockingCount == 0 && !this.shouldRun(r0)) {
             return false;
         } else {
-            this.doRunTask((Runnable) this.pendingRunnables.remove());
+            this.doRunTask(this.pendingRunnables.remove());
             return true;
         }
     }
