diff --git a/minecraft/world/level/chunk/storage/SectionStorage.java b/minecraft/world/level/chunk/storage/SectionStorage.java
index 294c925..c34f02d 100644
--- a/minecraft/world/level/chunk/storage/SectionStorage.java
+++ b/minecraft/world/level/chunk/storage/SectionStorage.java
@@ -127,7 +127,7 @@ public class SectionStorage<R> implements AutoCloseable {
         Optional<CompoundTag> optional = (Optional) this.tryRead(chunkpos).join();
         RegistryOps<Tag> registryops = RegistryOps.create(NbtOps.INSTANCE, (HolderLookup.Provider) this.registryAccess);
 
-        this.readColumn(chunkpos, registryops, (Tag) optional.orElse((Object) null));
+        this.readColumn(chunkpos, registryops, (Tag) optional.orElse(null));
     }
 
     private CompletableFuture<Optional<CompoundTag>> tryRead(ChunkPos chunkpos) {
@@ -159,7 +159,7 @@ public class SectionStorage<R> implements AutoCloseable {
             for (int l = this.levelHeightAccessor.getMinSection(); l < this.levelHeightAccessor.getMaxSection(); ++l) {
                 long i1 = getKey(chunkpos, l);
                 Optional<R> optional = optionaldynamic.get(Integer.toString(l)).result().flatMap((dynamic2) -> {
-                    DataResult dataresult = ((Codec) this.codec.apply(() -> {
+                    DataResult<R> dataresult = ((Codec) this.codec.apply(() -> {
                         this.setDirty(i1);
                     })).parse(dynamic2);
                     Logger logger = SectionStorage.LOGGER;
