diff --git a/minecraft/network/Connection.java b/minecraft/network/Connection.java
index 676fca9..09255aa 100644
--- a/minecraft/network/Connection.java
+++ b/minecraft/network/Connection.java
@@ -194,7 +194,7 @@ public class Connection extends SimpleChannelInboundHandler<Packet<?>> {
     }
 
     private static <T extends PacketListener> void genericsFtw(Packet<T> packet, PacketListener packetlistener) {
-        packet.handle(packetlistener);
+        packet.handle((T) packetlistener);
     }
 
     public void suspendInboundAfterProtocolChange() {
