diff --git a/minecraft/network/chat/ComponentUtils.java b/minecraft/network/chat/ComponentUtils.java
index b46b7a2..e606594 100644
--- a/minecraft/network/chat/ComponentUtils.java
+++ b/minecraft/network/chat/ComponentUtils.java
@@ -81,7 +81,7 @@ public class ComponentUtils {
         if (collection.isEmpty()) {
             return CommonComponents.EMPTY;
         } else if (collection.size() == 1) {
-            return (Component) function.apply((Comparable) collection.iterator().next());
+            return (Component) function.apply(collection.iterator().next());
         } else {
             List<T> list = Lists.newArrayList(collection);
 
@@ -112,7 +112,7 @@ public class ComponentUtils {
             boolean flag = true;
 
             for (Iterator iterator = collection.iterator(); iterator.hasNext(); flag = false) {
-                T t0 = iterator.next();
+                T t0 = (T) iterator.next();
 
                 if (!flag) {
                     mutablecomponent.append(component);
