diff --git a/minecraft/server/players/GameProfileCache.java b/minecraft/server/players/GameProfileCache.java
index 67d9e5c..42b97b4 100644
--- a/minecraft/server/players/GameProfileCache.java
+++ b/minecraft/server/players/GameProfileCache.java
@@ -85,7 +85,7 @@ public class GameProfileCache {
                 }
 
                 public void onProfileLookupFailed(String s1, Exception exception) {
-                    atomicreference.set((Object) null);
+                    atomicreference.set(null);
                 }
             };
 
