diff --git a/src/main/java/net/minecraft/commands/Commands.java b/src/main/java/net/minecraft/commands/Commands.java
index 4c5920e..88db0bc 100644
--- a/src/main/java/net/minecraft/commands/Commands.java
+++ b/src/main/java/net/minecraft/commands/Commands.java
@@ -131,7 +131,7 @@ public class Commands {
     public static final int LEVEL_GAMEMASTERS = 2;
     public static final int LEVEL_ADMINS = 3;
     public static final int LEVEL_OWNERS = 4;
-    private final CommandDispatcher<CommandSourceStack> dispatcher = new CommandDispatcher();
+    private static final CommandDispatcher<CommandSourceStack> dispatcher = new CommandDispatcher(); // dish -- make this static
 
     public Commands(Commands.CommandSelection commands_commandselection, CommandBuildContext commandbuildcontext) {
         AdvancementCommands.register(this.dispatcher);
@@ -388,8 +388,8 @@ public class Commands {
         };
     }
 
-    public CommandDispatcher<CommandSourceStack> getDispatcher() {
-        return this.dispatcher;
+    public static CommandDispatcher<CommandSourceStack> getDispatcher() { // dish -- make this static so that plugins can use commands
+        return dispatcher;
     }
 
     @Nullable
