diff --git a/minecraft/server/rcon/thread/QueryThreadGs4.java b/minecraft/server/rcon/thread/QueryThreadGs4.java
index 476949a..f982051 100644
--- a/minecraft/server/rcon/thread/QueryThreadGs4.java
+++ b/minecraft/server/rcon/thread/QueryThreadGs4.java
@@ -168,11 +168,11 @@ public class QueryThreadGs4 extends GenericThread {
             this.rulesResponse.writeString("map");
             this.rulesResponse.writeString(this.worldName);
             this.rulesResponse.writeString("numplayers");
-            this.rulesResponse.writeString(this.serverInterface.getPlayerCount().makeConcatWithConstants < invokedynamic > (this.serverInterface.getPlayerCount()));
+            this.rulesResponse.writeString("" + this.serverInterface.getPlayerCount());
             this.rulesResponse.writeString("maxplayers");
-            this.rulesResponse.writeString(this.maxPlayers.makeConcatWithConstants < invokedynamic > (this.maxPlayers));
+            this.rulesResponse.writeString("" + this.maxPlayers);
             this.rulesResponse.writeString("hostport");
-            this.rulesResponse.writeString(this.serverPort.makeConcatWithConstants < invokedynamic > (this.serverPort));
+            this.rulesResponse.writeString("" + this.serverPort);
             this.rulesResponse.writeString("hostip");
             this.rulesResponse.writeString(this.hostIp);
             this.rulesResponse.write(0);
