diff --git a/minecraft/data/recipes/SmithingTransformRecipeBuilder.java b/minecraft/data/recipes/SmithingTransformRecipeBuilder.java
index 5506665..386d78f 100644
--- a/minecraft/data/recipes/SmithingTransformRecipeBuilder.java
+++ b/minecraft/data/recipes/SmithingTransformRecipeBuilder.java
@@ -47,7 +47,7 @@ public class SmithingTransformRecipeBuilder {
     public void save(RecipeOutput recipeoutput, ResourceLocation resourcelocation) {
         this.ensureValid(resourcelocation);
         Advancement.Builder advancement_builder = recipeoutput.advancement().addCriterion("has_the_recipe", RecipeUnlockedTrigger.unlocked(resourcelocation)).rewards(AdvancementRewards.Builder.recipe(resourcelocation)).requirements(AdvancementRequirements.Strategy.OR);
-        Map map = this.criteria;
+        Map<String, Criterion<?>> map = this.criteria;
 
         Objects.requireNonNull(advancement_builder);
         map.forEach(advancement_builder::addCriterion);
