diff --git a/minecraft/core/dispenser/BoatDispenseItemBehavior.java b/minecraft/core/dispenser/BoatDispenseItemBehavior.java
index af39323..44d0afe 100644
--- a/minecraft/core/dispenser/BoatDispenseItemBehavior.java
+++ b/minecraft/core/dispenser/BoatDispenseItemBehavior.java
@@ -51,7 +51,7 @@ public class BoatDispenseItemBehavior extends DefaultDispenseItemBehavior {
             d4 = 0.0D;
         }
 
-        Object object = this.isChestBoat ? new ChestBoat(serverlevel, d1, d2 + d4, d3) : new Boat(serverlevel, d1, d2 + d4, d3);
+        Entity object = this.isChestBoat ? new ChestBoat(serverlevel, d1, d2 + d4, d3) : new Boat(serverlevel, d1, d2 + d4, d3);
 
         EntityType.createDefaultStackConfig(serverlevel, itemstack, (Player) null).accept(object);
         ((Boat) object).setVariant(this.type);
