diff --git a/minecraft/util/profiling/metrics/MetricsRegistry.java b/minecraft/util/profiling/metrics/MetricsRegistry.java
index d9494b2..3a7540c 100644
--- a/minecraft/util/profiling/metrics/MetricsRegistry.java
+++ b/minecraft/util/profiling/metrics/MetricsRegistry.java
@@ -16,7 +16,7 @@ public class MetricsRegistry {
     private MetricsRegistry() {}
 
     public void add(ProfilerMeasured profilermeasured) {
-        this.measuredInstances.put(profilermeasured, (Object) null);
+        this.measuredInstances.put(profilermeasured, null);
     }
 
     public List<MetricSampler> getRegisteredSamplers() {
