diff --git a/minecraft/nbt/visitors/CollectToTag.java b/minecraft/nbt/visitors/CollectToTag.java
index 4e582e3..19ca5a4 100644
--- a/minecraft/nbt/visitors/CollectToTag.java
+++ b/minecraft/nbt/visitors/CollectToTag.java
@@ -138,7 +138,7 @@ public class CollectToTag implements StreamTagVisitor {
             ListTag listtag = new ListTag();
 
             this.appendEntry(listtag);
-            Deque deque = this.consumerStack;
+            Deque<Consumer<Tag>> deque = this.consumerStack;
 
             Objects.requireNonNull(listtag);
             deque.addLast(listtag::add);
@@ -165,7 +165,7 @@ public class CollectToTag implements StreamTagVisitor {
             ListTag listtag = new ListTag();
 
             this.rootTag = listtag;
-            Deque deque = this.consumerStack;
+            Deque<Consumer<Tag>> deque = this.consumerStack;
 
             Objects.requireNonNull(listtag);
             deque.addLast(listtag::add);
