diff --git a/src/main/java/net/minecraft/world/entity/item/ItemEntity.java b/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
index d39c530..862e474 100644
--- a/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
+++ b/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
@@ -365,6 +365,19 @@ public class ItemEntity extends Entity implements TraceableEntity {
             ItemStack itemstack = this.getItem();
             Item item = itemstack.getItem();
             int i = itemstack.getCount();
+            // dish
+            // TODO: 12/11/2023 Check performance
+            net.ouja.dish.entity.DishEntity cachedThrower = new net.ouja.dish.entity.DishEntity((net.minecraft.world.entity.LivingEntity)this.cachedThrower);
+            net.ouja.api.entity.ItemEntity itemEntity = new net.ouja.api.entity.ItemEntity(
+                    itemstack.toDish(),
+                    this.pickupDelay,
+                    this.age,
+                    cachedThrower,
+                    this.thrower
+            );
+            net.ouja.api.event.player.PlayerAttemptPickUpItem event = new net.ouja.api.event.player.PlayerAttemptPickUpItem(new net.ouja.dish.entity.DishPlayer((net.minecraft.server.level.ServerPlayer)player), itemEntity);
+            net.ouja.dish.plugins.RegisteredEvents.callEvent(event);
+            // dish end
 
             if (this.pickupDelay == 0 && (this.target == null || this.target.equals(player.getUUID())) && player.getInventory().add(itemstack)) {
                 player.take(this, i);
