diff --git a/minecraft/util/datafix/fixes/LevelDataGeneratorOptionsFix.java b/minecraft/util/datafix/fixes/LevelDataGeneratorOptionsFix.java
index 5ff5146..7fc6fa2 100644
--- a/minecraft/util/datafix/fixes/LevelDataGeneratorOptionsFix.java
+++ b/minecraft/util/datafix/fixes/LevelDataGeneratorOptionsFix.java
@@ -112,9 +112,9 @@ public class LevelDataGeneratorOptionsFix extends DataFix {
         Type<?> type = this.getOutputSchema().getType(References.LEVEL);
 
         return this.fixTypeEverywhereTyped("LevelDataGeneratorOptionsFix", this.getInputSchema().getType(References.LEVEL), type, (typed) -> {
-            return (Typed) typed.write().flatMap((dynamic) -> {
+            return typed.write().flatMap((dynamic) -> {
                 Optional<String> optional = dynamic.get("generatorOptions").asString().result();
-                Dynamic dynamic1;
+                Dynamic<?> dynamic1;
 
                 if ("flat".equalsIgnoreCase(dynamic.get("generatorName").asString(""))) {
                     String s = (String) optional.orElse("");
@@ -139,7 +139,7 @@ public class LevelDataGeneratorOptionsFix extends DataFix {
         Iterator<String> iterator = Splitter.on(';').split(s).iterator();
         String s1 = "minecraft:plains";
         Map<String, Map<String, String>> map = Maps.newHashMap();
-        Object object;
+        List<Pair<Integer, String>> object;
 
         if (!s.isEmpty() && iterator.hasNext()) {
             object = getLayersInfoFromString((String) iterator.next());
@@ -187,11 +187,11 @@ public class LevelDataGeneratorOptionsFix extends DataFix {
             map.put("village", Maps.newHashMap());
         }
 
-        T t0 = dynamicops.createList(((List) object).stream().map((pair) -> {
+        T t0 = dynamicops.createList(object.stream().map((pair) -> {
             return dynamicops.createMap(ImmutableMap.of(dynamicops.createString("height"), dynamicops.createInt((Integer) pair.getFirst()), dynamicops.createString("block"), dynamicops.createString((String) pair.getSecond())));
         }));
-        T t1 = dynamicops.createMap((Map) map.entrySet().stream().map((entry) -> {
-            return Pair.of(dynamicops.createString(((String) entry.getKey()).toLowerCase(Locale.ROOT)), dynamicops.createMap((Map) ((Map) entry.getValue()).entrySet().stream().map((entry1) -> {
+        T t1 = dynamicops.createMap(map.entrySet().stream().map((entry) -> {
+            return Pair.of(dynamicops.createString(((String) entry.getKey()).toLowerCase(Locale.ROOT)), dynamicops.createMap((entry.getValue()).entrySet().stream().map((entry1) -> {
                 return Pair.of(dynamicops.createString((String) entry1.getKey()), dynamicops.createString((String) entry1.getValue()));
             }).collect(Collectors.toMap(Pair::getFirst, Pair::getSecond))));
         }).collect(Collectors.toMap(Pair::getFirst, Pair::getSecond)));
