diff --git a/minecraft/network/chat/numbers/FixedFormat.java b/minecraft/network/chat/numbers/FixedFormat.java
index 6b015aa..500f411 100644
--- a/minecraft/network/chat/numbers/FixedFormat.java
+++ b/minecraft/network/chat/numbers/FixedFormat.java
@@ -15,7 +15,7 @@ public class FixedFormat implements NumberFormat {
 
         @Override
         public MapCodec<FixedFormat> mapCodec() {
-            return null.CODEC;
+            return CODEC;
         }
 
         public void writeToStream(FriendlyByteBuf friendlybytebuf, FixedFormat fixedformat) {
