diff --git a/minecraft/world/entity/animal/sniffer/SnifferAi.java b/minecraft/world/entity/animal/sniffer/SnifferAi.java
index afb1dbf..4e5759d 100644
--- a/minecraft/world/entity/animal/sniffer/SnifferAi.java
+++ b/minecraft/world/entity/animal/sniffer/SnifferAi.java
@@ -143,7 +143,7 @@ public class SnifferAi {
 
         protected void stop(ServerLevel serverlevel, Sniffer sniffer, long i) {
             if (sniffer.canDig() && sniffer.canSniff()) {
-                sniffer.getBrain().setMemory(MemoryModuleType.SNIFFER_DIGGING, (Object) true);
+                sniffer.getBrain().setMemory(MemoryModuleType.SNIFFER_DIGGING, true);
             }
 
             sniffer.getBrain().eraseMemory(MemoryModuleType.WALK_TARGET);
@@ -204,7 +204,7 @@ public class SnifferAi {
 
             sniffer.transitionTo(Sniffer.State.IDLING).onDiggingComplete(flag);
             sniffer.getBrain().eraseMemory(MemoryModuleType.SNIFFER_DIGGING);
-            sniffer.getBrain().setMemory(MemoryModuleType.SNIFFER_HAPPY, (Object) true);
+            sniffer.getBrain().setMemory(MemoryModuleType.SNIFFER_HAPPY, true);
         }
     }
 
@@ -275,8 +275,8 @@ public class SnifferAi {
             sniffer.transitionTo(Sniffer.State.IDLING);
             if (flag) {
                 sniffer.calculateDigPosition().ifPresent((blockpos) -> {
-                    sniffer.getBrain().setMemory(MemoryModuleType.SNIFFER_SNIFFING_TARGET, (Object) blockpos);
-                    sniffer.getBrain().setMemory(MemoryModuleType.WALK_TARGET, (Object) (new WalkTarget(blockpos, 1.25F, 0)));
+                    sniffer.getBrain().setMemory(MemoryModuleType.SNIFFER_SNIFFING_TARGET, blockpos);
+                    sniffer.getBrain().setMemory(MemoryModuleType.WALK_TARGET, (new WalkTarget(blockpos, 1.25F, 0)));
                 });
             }
 
