diff --git a/minecraft/world/level/biome/Climate.java b/minecraft/world/level/biome/Climate.java
index 007360b..6f8c033 100644
--- a/minecraft/world/level/biome/Climate.java
+++ b/minecraft/world/level/biome/Climate.java
@@ -231,7 +231,7 @@ public class Climate {
         private final Climate.RTree<T> index;
 
         public static <T> Codec<Climate.ParameterList<T>> codec(MapCodec<T> mapcodec) {
-            return ExtraCodecs.nonEmptyList(RecordCodecBuilder.create((instance) -> {
+            return ExtraCodecs.nonEmptyList(RecordCodecBuilder.<Pair<Climate.ParameterPoint, T>>create((instance) -> {
                 return instance.group(Climate.ParameterPoint.CODEC.fieldOf("parameters").forGetter(Pair::getFirst), mapcodec.forGetter(Pair::getSecond)).apply(instance, Pair::of);
             }).listOf()).xmap(Climate.ParameterList::new, Climate.ParameterList::values);
         }
@@ -254,7 +254,7 @@ public class Climate {
             Iterator<Pair<Climate.ParameterPoint, T>> iterator = this.values().iterator();
             Pair<Climate.ParameterPoint, T> pair = (Pair) iterator.next();
             long i = ((Climate.ParameterPoint) pair.getFirst()).fitness(climate_targetpoint);
-            Object object = pair.getSecond();
+            T object = pair.getSecond();
 
             while (iterator.hasNext()) {
                 Pair<Climate.ParameterPoint, T> pair1 = (Pair) iterator.next();
@@ -418,7 +418,7 @@ public class Climate {
                 List<Climate.Parameter> list1 = Lists.newArrayList();
 
                 for (int i = 0; i < 7; ++i) {
-                    list1.add((Object) null);
+                    list1.add(null);
                 }
 
                 Iterator iterator = list.iterator();
