diff --git a/minecraft/util/datafix/fixes/EntityIdFix.java b/minecraft/util/datafix/fixes/EntityIdFix.java
index 63afbe7..1264cab 100644
--- a/minecraft/util/datafix/fixes/EntityIdFix.java
+++ b/minecraft/util/datafix/fixes/EntityIdFix.java
@@ -94,8 +94,8 @@ public class EntityIdFix extends DataFix {
     }
 
     public TypeRewriteRule makeRule() {
-        TaggedChoiceType<String> taggedchoicetype = this.getInputSchema().findChoiceType(References.ENTITY);
-        TaggedChoiceType<String> taggedchoicetype1 = this.getOutputSchema().findChoiceType(References.ENTITY);
+        TaggedChoiceType<String> taggedchoicetype = (TaggedChoiceType<String>) this.getInputSchema().findChoiceType(References.ENTITY);
+        TaggedChoiceType<String> taggedchoicetype1 = (TaggedChoiceType<String>) this.getOutputSchema().findChoiceType(References.ENTITY);
         Type<?> type = this.getInputSchema().getType(References.ITEM_STACK);
         Type<?> type1 = this.getOutputSchema().getType(References.ITEM_STACK);
 
