diff --git a/minecraft/server/network/PlayerChunkSender.java b/minecraft/server/network/PlayerChunkSender.java
index 676e0f6..8b5ae7a 100644
--- a/minecraft/server/network/PlayerChunkSender.java
+++ b/minecraft/server/network/PlayerChunkSender.java
@@ -98,10 +98,10 @@ public class PlayerChunkSender {
         List list;
 
         if (!this.memoryConnection && this.pendingChunks.size() > i) {
-            Stream stream = this.pendingChunks.stream();
+            Stream<Long> stream = this.pendingChunks.stream();
 
             Objects.requireNonNull(chunkpos);
-            longstream = ((List) stream.collect(Comparators.least(i, Comparator.comparingInt(chunkpos::distanceSquared)))).stream().mapToLong(Long::longValue);
+            longstream = (stream.collect(Comparators.least(i, Comparator.comparingInt(chunkpos::distanceSquared)))).stream().mapToLong(Long::longValue);
             Objects.requireNonNull(chunkmap);
             list = longstream.mapToObj(chunkmap::getChunkToSend).filter(Objects::nonNull).toList();
         } else {
