diff --git a/minecraft/world/entity/animal/goat/GoatAi.java b/minecraft/world/entity/animal/goat/GoatAi.java
index 2f63561..0aa6518 100644
--- a/minecraft/world/entity/animal/goat/GoatAi.java
+++ b/minecraft/world/entity/animal/goat/GoatAi.java
@@ -60,8 +60,8 @@ public class GoatAi {
     public GoatAi() {}
 
     protected static void initMemories(Goat goat, RandomSource randomsource) {
-        goat.getBrain().setMemory(MemoryModuleType.LONG_JUMP_COOLDOWN_TICKS, (Object) GoatAi.TIME_BETWEEN_LONG_JUMPS.sample(randomsource));
-        goat.getBrain().setMemory(MemoryModuleType.RAM_COOLDOWN_TICKS, (Object) GoatAi.TIME_BETWEEN_RAMS.sample(randomsource));
+        goat.getBrain().setMemory(MemoryModuleType.LONG_JUMP_COOLDOWN_TICKS, GoatAi.TIME_BETWEEN_LONG_JUMPS.sample(randomsource));
+        goat.getBrain().setMemory(MemoryModuleType.RAM_COOLDOWN_TICKS, GoatAi.TIME_BETWEEN_RAMS.sample(randomsource));
     }
 
     protected static Brain<?> makeBrain(Brain<Goat> brain) {
