diff --git a/minecraft/world/item/BlockItem.java b/minecraft/world/item/BlockItem.java
index ed2e7f2..4c4d617 100644
--- a/minecraft/world/item/BlockItem.java
+++ b/minecraft/world/item/BlockItem.java
@@ -10,6 +10,7 @@ import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.core.BlockPos;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.nbt.ListTag;
+import net.minecraft.nbt.Tag;
 import net.minecraft.network.chat.Component;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.server.level.ServerPlayer;
@@ -239,7 +240,7 @@ public class BlockItem extends Item {
 
             if (compoundtag != null && compoundtag.contains("Items", 9)) {
                 ListTag listtag = compoundtag.getList("Items", 10);
-                Stream stream = listtag.stream();
+                Stream<Tag> stream = listtag.stream();
 
                 Objects.requireNonNull(CompoundTag.class);
                 ItemUtils.onContainerDestroyed(itementity, stream.map(CompoundTag.class::cast).map(ItemStack::of));
