diff --git a/src/main/java/net/minecraft/world/entity/animal/Squid.java b/src/main/java/net/minecraft/world/entity/animal/Squid.java
index 120889e..e22c9fc 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Squid.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Squid.java
@@ -176,7 +176,7 @@ public class Squid extends WaterAnimal {
         return vec3.yRot(-this.yBodyRotO * (float) (Math.PI / 180.0));
     }
 
-    private void spawnInk() {
+    public void spawnInk() { // dish -- make this public
         this.makeSound(this.getSquirtSound());
         Vec3 vec3 = this.rotateVector(new Vec3(0.0, -1.0, 0.0)).add(this.getX(), this.getY(), this.getZ());
 
