diff --git a/minecraft/world/level/Level.java b/minecraft/world/level/Level.java
index 098e22f..ce4c898 100644
--- a/minecraft/world/level/Level.java
+++ b/minecraft/world/level/Level.java
@@ -663,7 +663,7 @@ public abstract class Level implements LevelAccessor, AutoCloseable {
 
                 for (int k = 0; k < j; ++k) {
                     EnderDragonPart enderdragonpart = aenderdragonpart[k];
-                    T t0 = (Entity) entitytypetest.tryCast(enderdragonpart);
+                    T t0 = entitytypetest.tryCast(enderdragonpart);
 
                     if (t0 != null && predicate.test(t0)) {
                         list.add(t0);
