diff --git a/minecraft/world/level/storage/loot/functions/CopyBlockState.java b/minecraft/world/level/storage/loot/functions/CopyBlockState.java
index c441146..649516a 100644
--- a/minecraft/world/level/storage/loot/functions/CopyBlockState.java
+++ b/minecraft/world/level/storage/loot/functions/CopyBlockState.java
@@ -41,11 +41,7 @@ public class CopyBlockState extends LootItemConditionalFunction {
     }
 
     private CopyBlockState(List<LootItemCondition> list, Holder<Block> holder, List<String> list1) {
-        Stream stream = list1.stream();
-        StateDefinition statedefinition = ((Block) holder.value()).getStateDefinition();
-
-        Objects.requireNonNull(statedefinition);
-        this(list, holder, (Set) stream.map(statedefinition::getProperty).filter(Objects::nonNull).collect(Collectors.toSet()));
+        this(list, holder, (Set) list1.stream().map(((Block) holder.value()).getStateDefinition()::getProperty).filter(Objects::nonNull).collect(Collectors.toSet()));
     }
 
     @Override
