diff --git a/minecraft/world/level/storage/loot/functions/SetBannerPatternFunction.java b/minecraft/world/level/storage/loot/functions/SetBannerPatternFunction.java
index c4c16eb..ed475e2 100644
--- a/minecraft/world/level/storage/loot/functions/SetBannerPatternFunction.java
+++ b/minecraft/world/level/storage/loot/functions/SetBannerPatternFunction.java
@@ -47,7 +47,7 @@ public class SetBannerPatternFunction extends LootItemConditionalFunction {
         }
 
         BannerPattern.Builder bannerpattern_builder = new BannerPattern.Builder();
-        List list = this.patterns;
+        List<Pair<Holder<BannerPattern>, DyeColor>> list = this.patterns;
 
         Objects.requireNonNull(bannerpattern_builder);
         list.forEach(bannerpattern_builder::addPattern);
