diff --git a/minecraft/world/level/levelgen/flat/FlatLevelGeneratorSettings.java b/minecraft/world/level/levelgen/flat/FlatLevelGeneratorSettings.java
index 53677c6..6886ef9 100644
--- a/minecraft/world/level/levelgen/flat/FlatLevelGeneratorSettings.java
+++ b/minecraft/world/level/levelgen/flat/FlatLevelGeneratorSettings.java
@@ -35,7 +35,7 @@ import org.slf4j.Logger;
 public class FlatLevelGeneratorSettings {
 
     private static final Logger LOGGER = LogUtils.getLogger();
-    public static final Codec<FlatLevelGeneratorSettings> CODEC = RecordCodecBuilder.create((instance) -> {
+    public static final Codec<FlatLevelGeneratorSettings> CODEC = RecordCodecBuilder.<FlatLevelGeneratorSettings>create((instance) -> {
         return instance.group(RegistryCodecs.homogeneousList(Registries.STRUCTURE_SET).optionalFieldOf("structure_overrides").forGetter((flatlevelgeneratorsettings) -> {
             return flatlevelgeneratorsettings.structureOverrides;
         }), FlatLayerInfo.CODEC.listOf().fieldOf("layers").forGetter(FlatLevelGeneratorSettings::getLayersInfo), Codec.BOOL.fieldOf("lakes").orElse(false).forGetter((flatlevelgeneratorsettings) -> {
