diff --git a/minecraft/data/loot/LootTableProvider.java b/minecraft/data/loot/LootTableProvider.java
index ef2265d..42c4f41 100644
--- a/minecraft/data/loot/LootTableProvider.java
+++ b/minecraft/data/loot/LootTableProvider.java
@@ -67,7 +67,7 @@ public class LootTableProvider implements DataProvider {
             @Nullable
             @Override
             public <T> T getElement(LootDataId<T> lootdataid) {
-                return lootdataid.type() == LootDataType.TABLE ? map.get(lootdataid.location()) : null;
+                return lootdataid.type() == LootDataType.TABLE ? (T) map.get(lootdataid.location()) : null;
             }
         });
         Set<ResourceLocation> set = Sets.difference(this.requiredTables, map.keySet());
