diff --git a/minecraft/server/commands/FunctionCommand.java b/minecraft/server/commands/FunctionCommand.java
index 7017ec8..2a89f8f 100644
--- a/minecraft/server/commands/FunctionCommand.java
+++ b/minecraft/server/commands/FunctionCommand.java
@@ -189,12 +189,12 @@ public class FunctionCommand {
                     instantiateAndQueueFunctions(compoundtag, executioncontrol, commanddispatcher, t2, commandfunction1, resourcelocation1, commandresultcallback2, false);
                 }
             } else {
-                class 1Accumulator {
+                class Accumulator {
 
                     boolean anyResult;
                     int sum;
 
-                    _Accumulator/* $FF was: 1Accumulator*/() {
+                    Accumulator/* $FF was: 1Accumulator*/() {
 }
 
                     public void add(int i) {
@@ -203,9 +203,9 @@ public class FunctionCommand {
                     }
                 }
 
-                1Accumulator 1accumulator = new 1Accumulator();
+                Accumulator accumulator = new Accumulator();
                 CommandResultCallback commandresultcallback3 = (flag, i) -> {
-                    1accumulator.add(i);
+                    accumulator.add(i);
                 };
                 Iterator iterator1 = collection.iterator();
 
@@ -218,8 +218,8 @@ public class FunctionCommand {
                 }
 
                 executioncontrol.queueNext((executioncontext, frame) -> {
-                    if (1accumulator.anyResult) {
-                        commandresultcallback.onSuccess(1accumulator.sum);
+                    if (accumulator.anyResult) {
+                        commandresultcallback.onSuccess(accumulator.sum);
                     }
 
                 });
