diff --git a/minecraft/server/commands/LocateCommand.java b/minecraft/server/commands/LocateCommand.java
index 4c04edc..79be23d 100644
--- a/minecraft/server/commands/LocateCommand.java
+++ b/minecraft/server/commands/LocateCommand.java
@@ -74,15 +74,11 @@ public class LocateCommand {
     }
 
     private static Optional<? extends HolderSet.ListBacked<Structure>> getHolders(ResourceOrTagKeyArgument.Result<Structure> resourceortagkeyargument_result, Registry<Structure> registry) {
-        Either either = resourceortagkeyargument_result.unwrap();
-        Function function = (resourcekey) -> {
+        return resourceortagkeyargument_result.unwrap().map((resourcekey) -> {
             return registry.getHolder(resourcekey).map((holder) -> {
                 return HolderSet.direct(holder);
             });
-        };
-
-        Objects.requireNonNull(registry);
-        return (Optional) either.map(function, registry::getTag);
+        }, registry::getTag);
     }
 
     private static int locateStructure(CommandSourceStack commandsourcestack, ResourceOrTagKeyArgument.Result<Structure> resourceortagkeyargument_result) throws CommandSyntaxException {
