diff --git a/minecraft/server/commands/PardonIpCommand.java b/minecraft/server/commands/PardonIpCommand.java
index 1ad72ad..380a819 100644
--- a/minecraft/server/commands/PardonIpCommand.java
+++ b/minecraft/server/commands/PardonIpCommand.java
@@ -38,7 +38,7 @@ public class PardonIpCommand {
             if (!ipbanlist.isBanned(s)) {
                 throw PardonIpCommand.ERROR_NOT_BANNED.create();
             } else {
-                ipbanlist.remove((Object) s);
+                ipbanlist.remove(s);
                 commandsourcestack.sendSuccess(() -> {
                     return Component.translatable("commands.pardonip.success", s);
                 }, true);
