diff --git a/minecraft/server/commands/SpreadPlayersCommand.java b/minecraft/server/commands/SpreadPlayersCommand.java
index 444ffed..747184b 100644
--- a/minecraft/server/commands/SpreadPlayersCommand.java
+++ b/minecraft/server/commands/SpreadPlayersCommand.java
@@ -93,7 +93,7 @@ public class SpreadPlayersCommand {
             if (entity instanceof Player) {
                 set.add(entity.getTeam());
             } else {
-                set.add((Object) null);
+                set.add(null);
             }
         }
 
