diff --git a/src/main/java/net/minecraft/world/entity/animal/sniffer/Sniffer.java b/src/main/java/net/minecraft/world/entity/animal/sniffer/Sniffer.java
index 834f3ae..95691f3 100644
--- a/src/main/java/net/minecraft/world/entity/animal/sniffer/Sniffer.java
+++ b/src/main/java/net/minecraft/world/entity/animal/sniffer/Sniffer.java
@@ -141,7 +141,7 @@ public class Sniffer extends Animal {
         return this.position().add(this.getForward().scale(2.25));
     }
 
-    private Sniffer.State getState() {
+    public Sniffer.State getState() { // dish -- make this public
         return this.entityData.get(DATA_STATE);
     }
 
