diff --git a/minecraft/core/cauldron/CauldronInteraction.java b/minecraft/core/cauldron/CauldronInteraction.java
index a71f4db..e3d5a25 100644
--- a/minecraft/core/cauldron/CauldronInteraction.java
+++ b/minecraft/core/cauldron/CauldronInteraction.java
@@ -115,7 +115,7 @@ public interface CauldronInteraction {
     };
 
     static Object2ObjectOpenHashMap<Item, CauldronInteraction> newInteractionMap() {
-        return (Object2ObjectOpenHashMap) Util.make(new Object2ObjectOpenHashMap(), (object2objectopenhashmap) -> {
+        return Util.make(new Object2ObjectOpenHashMap<>(), (object2objectopenhashmap) -> {
             object2objectopenhashmap.defaultReturnValue((blockstate, level, blockpos, player, interactionhand, itemstack) -> {
                 return InteractionResult.PASS;
             });
