diff --git a/minecraft/world/level/chunk/LevelChunk.java b/minecraft/world/level/chunk/LevelChunk.java
index 35275ec..ec036d5 100644
--- a/minecraft/world/level/chunk/LevelChunk.java
+++ b/minecraft/world/level/chunk/LevelChunk.java
@@ -697,7 +697,7 @@ public class LevelChunk extends ChunkAccess {
 
     private <T extends BlockEntity> void updateBlockEntityTicker(T t0) {
         BlockState blockstate = t0.getBlockState();
-        BlockEntityTicker<T> blockentityticker = blockstate.getTicker(this.level, t0.getType());
+        BlockEntityTicker<T> blockentityticker = (BlockEntityTicker<T>) blockstate.getTicker(this.level, t0.getType());
 
         if (blockentityticker == null) {
             this.removeBlockEntityTicker(t0.getBlockPos());
