diff --git a/minecraft/world/level/levelgen/placement/PlacementModifierType.java b/minecraft/world/level/levelgen/placement/PlacementModifierType.java
index 28f31b5..9aae340 100644
--- a/minecraft/world/level/levelgen/placement/PlacementModifierType.java
+++ b/minecraft/world/level/levelgen/placement/PlacementModifierType.java
@@ -25,7 +25,7 @@ public interface PlacementModifierType<P extends PlacementModifier> {
     Codec<P> codec();
 
     private static <P extends PlacementModifier> PlacementModifierType<P> register(String s, Codec<P> codec) {
-        return (PlacementModifierType) Registry.register(BuiltInRegistries.PLACEMENT_MODIFIER_TYPE, s, () -> {
+        return Registry.register(BuiltInRegistries.PLACEMENT_MODIFIER_TYPE, s, () -> {
             return codec;
         });
     }
