diff --git a/minecraft/world/level/levelgen/structure/structures/RuinedPortalPiece.java b/minecraft/world/level/levelgen/structure/structures/RuinedPortalPiece.java
index d8cf630..c7f3997 100644
--- a/minecraft/world/level/levelgen/structure/structures/RuinedPortalPiece.java
+++ b/minecraft/world/level/levelgen/structure/structures/RuinedPortalPiece.java
@@ -12,6 +12,7 @@ import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.nbt.NbtOps;
+import net.minecraft.nbt.Tag;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.tags.BlockTags;
 import net.minecraft.util.RandomSource;
@@ -70,7 +71,7 @@ public class RuinedPortalPiece extends TemplateStructurePiece {
             return makeSettings(structuretemplatemanager, compoundtag, resourcelocation);
         });
         this.verticalPlacement = RuinedPortalPiece.VerticalPlacement.byName(compoundtag.getString("VerticalPlacement"));
-        DataResult dataresult = RuinedPortalPiece.Properties.CODEC.parse(new Dynamic(NbtOps.INSTANCE, compoundtag.get("Properties")));
+        DataResult<RuinedPortalPiece.Properties> dataresult = RuinedPortalPiece.Properties.CODEC.parse(new Dynamic<>(NbtOps.INSTANCE, compoundtag.get("Properties")));
         Logger logger = RuinedPortalPiece.LOGGER;
 
         Objects.requireNonNull(logger);
@@ -83,7 +84,7 @@ public class RuinedPortalPiece extends TemplateStructurePiece {
         compoundtag.putString("Rotation", this.placeSettings.getRotation().name());
         compoundtag.putString("Mirror", this.placeSettings.getMirror().name());
         compoundtag.putString("VerticalPlacement", this.verticalPlacement.getName());
-        DataResult dataresult = RuinedPortalPiece.Properties.CODEC.encodeStart(NbtOps.INSTANCE, this.properties);
+        DataResult<Tag> dataresult = RuinedPortalPiece.Properties.CODEC.encodeStart(NbtOps.INSTANCE, this.properties);
         Logger logger = RuinedPortalPiece.LOGGER;
 
         Objects.requireNonNull(logger);
@@ -98,7 +99,7 @@ public class RuinedPortalPiece extends TemplateStructurePiece {
         Mirror mirror = Mirror.valueOf(compoundtag.getString("Mirror"));
         Rotation rotation = Rotation.valueOf(compoundtag.getString("Rotation"));
         RuinedPortalPiece.VerticalPlacement ruinedportalpiece_verticalplacement = RuinedPortalPiece.VerticalPlacement.byName(compoundtag.getString("VerticalPlacement"));
-        DataResult dataresult = RuinedPortalPiece.Properties.CODEC.parse(new Dynamic(NbtOps.INSTANCE, compoundtag.get("Properties")));
+        DataResult<RuinedPortalPiece.Properties> dataresult = RuinedPortalPiece.Properties.CODEC.parse(new Dynamic<>(NbtOps.INSTANCE, compoundtag.get("Properties")));
         Logger logger = RuinedPortalPiece.LOGGER;
 
         Objects.requireNonNull(logger);
