diff --git a/minecraft/data/info/RegistryDumpReport.java b/minecraft/data/info/RegistryDumpReport.java
index 8144a03..0e4baed 100644
--- a/minecraft/data/info/RegistryDumpReport.java
+++ b/minecraft/data/info/RegistryDumpReport.java
@@ -41,7 +41,7 @@ public class RegistryDumpReport implements DataProvider {
             jsonobject.addProperty("default", resourcelocation.toString());
         }
 
-        int i = BuiltInRegistries.REGISTRY.getId(registry);
+        int i = ((DefaultedRegistry)BuiltInRegistries.REGISTRY).getId(registry);
 
         jsonobject.addProperty("protocol_id", i);
         JsonObject jsonobject1 = new JsonObject();
