diff --git a/minecraft/server/ServerFunctionLibrary.java b/minecraft/server/ServerFunctionLibrary.java
index a16658a..b1191c7 100644
--- a/minecraft/server/ServerFunctionLibrary.java
+++ b/minecraft/server/ServerFunctionLibrary.java
@@ -96,10 +96,8 @@ public class ServerFunctionLibrary implements PreparableReloadListener {
                 return map1;
             });
         });
-        CompletableFuture completablefuture2 = completablefuture.thenCombine(completablefuture1, Pair::of);
 
-        Objects.requireNonNull(preparablereloadlistener_preparationbarrier);
-        return completablefuture2.thenCompose(preparablereloadlistener_preparationbarrier::wait).thenAcceptAsync((pair) -> {
+        return completablefuture.thenCombine(completablefuture1, Pair::of).thenCompose(preparablereloadlistener_preparationbarrier::wait).thenAcceptAsync((pair) -> {
             Map<ResourceLocation, CompletableFuture<CommandFunction>> map = (Map) pair.getSecond();
             Builder<ResourceLocation, CommandFunction> builder = ImmutableMap.builder();
 
