diff --git a/minecraft/world/level/levelgen/structure/templatesystem/rule/blockentity/RuleBlockEntityModifierType.java b/minecraft/world/level/levelgen/structure/templatesystem/rule/blockentity/RuleBlockEntityModifierType.java
index be37b9f..c25d116 100644
--- a/minecraft/world/level/levelgen/structure/templatesystem/rule/blockentity/RuleBlockEntityModifierType.java
+++ b/minecraft/world/level/levelgen/structure/templatesystem/rule/blockentity/RuleBlockEntityModifierType.java
@@ -14,7 +14,7 @@ public interface RuleBlockEntityModifierType<P extends RuleBlockEntityModifier>
     Codec<P> codec();
 
     private static <P extends RuleBlockEntityModifier> RuleBlockEntityModifierType<P> register(String s, Codec<P> codec) {
-        return (RuleBlockEntityModifierType) Registry.register(BuiltInRegistries.RULE_BLOCK_ENTITY_MODIFIER, s, () -> {
+        return Registry.register(BuiltInRegistries.RULE_BLOCK_ENTITY_MODIFIER, s, () -> {
             return codec;
         });
     }
