diff --git a/minecraft/world/level/chunk/HashMapPalette.java b/minecraft/world/level/chunk/HashMapPalette.java
index 983d2be..06b7717 100644
--- a/minecraft/world/level/chunk/HashMapPalette.java
+++ b/minecraft/world/level/chunk/HashMapPalette.java
@@ -112,7 +112,7 @@ public class HashMapPalette<T> implements Palette<T> {
 
     public List<T> getEntries() {
         ArrayList<T> arraylist = new ArrayList();
-        Iterator iterator = this.values.iterator();
+        Iterator<T> iterator = this.values.iterator();
 
         Objects.requireNonNull(arraylist);
         iterator.forEachRemaining(arraylist::add);
