diff --git a/minecraft/util/datafix/fixes/SavedDataUUIDFix.java b/minecraft/util/datafix/fixes/SavedDataUUIDFix.java
index 896f596..ed1152e 100644
--- a/minecraft/util/datafix/fixes/SavedDataUUIDFix.java
+++ b/minecraft/util/datafix/fixes/SavedDataUUIDFix.java
@@ -23,7 +23,7 @@ public class SavedDataUUIDFix extends AbstractUUIDFix {
                         return dynamic2.createList(dynamic2.asStream().map((dynamic3) -> {
                             return dynamic3.update("HeroesOfTheVillage", (dynamic4) -> {
                                 return dynamic4.createList(dynamic4.asStream().map((dynamic5) -> {
-                                    return (Dynamic) createUUIDFromLongs(dynamic5, "UUIDMost", "UUIDLeast").orElseGet(() -> {
+                                    return createUUIDFromLongs(dynamic5, "UUIDMost", "UUIDLeast").orElseGet(() -> {
                                         SavedDataUUIDFix.LOGGER.warn("HeroesOfTheVillage contained invalid UUIDs.");
                                         return dynamic5;
                                     });
