diff --git a/minecraft/world/level/storage/loot/LootParams.java b/minecraft/world/level/storage/loot/LootParams.java
index a00b823..c34905e 100644
--- a/minecraft/world/level/storage/loot/LootParams.java
+++ b/minecraft/world/level/storage/loot/LootParams.java
@@ -36,7 +36,7 @@ public class LootParams {
     }
 
     public <T> T getParameter(LootContextParam<T> lootcontextparam) {
-        T t0 = this.params.get(lootcontextparam);
+        T t0 = (T) this.params.get(lootcontextparam);
 
         if (t0 == null) {
             throw new NoSuchElementException(lootcontextparam.getName().toString());
@@ -47,12 +47,12 @@ public class LootParams {
 
     @Nullable
     public <T> T getOptionalParameter(LootContextParam<T> lootcontextparam) {
-        return this.params.get(lootcontextparam);
+        return (T) this.params.get(lootcontextparam);
     }
 
     @Nullable
     public <T> T getParamOrNull(LootContextParam<T> lootcontextparam) {
-        return this.params.get(lootcontextparam);
+        return (T) this.params.get(lootcontextparam);
     }
 
     public void addDynamicDrops(ResourceLocation resourcelocation, Consumer<ItemStack> consumer) {
@@ -105,7 +105,7 @@ public class LootParams {
         }
 
         public <T> T getParameter(LootContextParam<T> lootcontextparam) {
-            T t0 = this.params.get(lootcontextparam);
+            T t0 = (T) this.params.get(lootcontextparam);
 
             if (t0 == null) {
                 throw new NoSuchElementException(lootcontextparam.getName().toString());
@@ -116,7 +116,7 @@ public class LootParams {
 
         @Nullable
         public <T> T getOptionalParameter(LootContextParam<T> lootcontextparam) {
-            return this.params.get(lootcontextparam);
+            return (T) this.params.get(lootcontextparam);
         }
 
         public LootParams.Builder withDynamicDrop(ResourceLocation resourcelocation, LootParams.DynamicDrop lootparams_dynamicdrop) {
