diff --git a/minecraft/world/level/entity/EntityLookup.java b/minecraft/world/level/entity/EntityLookup.java
index b9f4263..9dda51d 100644
--- a/minecraft/world/level/entity/EntityLookup.java
+++ b/minecraft/world/level/entity/EntityLookup.java
@@ -30,10 +30,10 @@ public class EntityLookup<T extends EntityAccess> {
                 return;
             }
 
-            T t0 = (EntityAccess) objectiterator.next();
+            T t0 = (T) objectiterator.next();
 
             entityaccess = (EntityAccess) entitytypetest.tryCast(t0);
-        } while (entityaccess == null || !abortableiterationconsumer.accept(entityaccess).shouldAbort());
+        } while (entityaccess == null || !abortableiterationconsumer.accept((U) entityaccess).shouldAbort());
 
     }
 
@@ -59,12 +59,12 @@ public class EntityLookup<T extends EntityAccess> {
 
     @Nullable
     public T getEntity(int i) {
-        return (EntityAccess) this.byId.get(i);
+        return this.byId.get(i);
     }
 
     @Nullable
     public T getEntity(UUID uuid) {
-        return (EntityAccess) this.byUuid.get(uuid);
+        return this.byUuid.get(uuid);
     }
 
     public int count() {
