diff --git a/src/main/java/net/minecraft/world/level/block/Block.java b/src/main/java/net/minecraft/world/level/block/Block.java
index 656e27d..48b3678 100644
--- a/src/main/java/net/minecraft/world/level/block/Block.java
+++ b/src/main/java/net/minecraft/world/level/block/Block.java
@@ -522,7 +522,7 @@ public class Block extends BlockBehaviour implements ItemLike {
         }
 
     }
-
+    public net.ouja.api.world.level.block.Block toDish(BlockPos blockPos, Level level) { return new net.ouja.dish.world.block.DishBlockInfo(this, blockPos, level).getBlock(); } // dish
     public static final class BlockStatePairKey {
 
         private final BlockState first;
