diff --git a/minecraft/world/level/levelgen/structure/pools/SinglePoolElement.java b/minecraft/world/level/levelgen/structure/pools/SinglePoolElement.java
index 40649f6..52a9dda 100644
--- a/minecraft/world/level/levelgen/structure/pools/SinglePoolElement.java
+++ b/minecraft/world/level/levelgen/structure/pools/SinglePoolElement.java
@@ -78,7 +78,7 @@ public class SinglePoolElement extends StructurePoolElement {
     }
 
     private StructureTemplate getTemplate(StructureTemplateManager structuretemplatemanager) {
-        Either either = this.template;
+        Either<ResourceLocation, StructureTemplate> either = this.template;
 
         Objects.requireNonNull(structuretemplatemanager);
         return (StructureTemplate) either.map(structuretemplatemanager::getOrCreate, Function.identity());
@@ -156,11 +156,11 @@ public class SinglePoolElement extends StructurePoolElement {
             structureplacesettings.addProcessor(JigsawReplacementProcessor.INSTANCE);
         }
 
-        List list = ((StructureProcessorList) this.processors.value()).list();
+        List<net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessor> list = ((StructureProcessorList) this.processors.value()).list();
 
         Objects.requireNonNull(structureplacesettings);
         list.forEach(structureplacesettings::addProcessor);
-        ImmutableList immutablelist = this.getProjection().getProcessors();
+        ImmutableList<net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessor> immutablelist = this.getProjection().getProcessors();
 
         Objects.requireNonNull(structureplacesettings);
         immutablelist.forEach(structureplacesettings::addProcessor);
