diff --git a/minecraft/util/datafix/fixes/GossipUUIDFix.java b/minecraft/util/datafix/fixes/GossipUUIDFix.java
index 40b0751..6ef0c04 100644
--- a/minecraft/util/datafix/fixes/GossipUUIDFix.java
+++ b/minecraft/util/datafix/fixes/GossipUUIDFix.java
@@ -7,6 +7,7 @@ import com.mojang.datafixers.schemas.Schema;
 import com.mojang.serialization.Dynamic;
 import java.util.Objects;
 import java.util.Optional;
+import java.util.stream.Stream;
 
 public class GossipUUIDFix extends NamedEntityFix {
 
@@ -18,14 +19,14 @@ public class GossipUUIDFix extends NamedEntityFix {
     protected Typed<?> fix(Typed<?> typed) {
         return typed.update(DSL.remainderFinder(), (dynamic) -> {
             return dynamic.update("Gossips", (dynamic1) -> {
-                Optional optional = dynamic1.asStreamOpt().result().map((stream) -> {
+                Optional<Stream> optional = dynamic1.asStreamOpt().result().map((stream) -> {
                     return stream.map((dynamic2) -> {
-                        return (Dynamic) AbstractUUIDFix.replaceUUIDLeastMost(dynamic2, "Target", "Target").orElse(dynamic2);
+                        return AbstractUUIDFix.replaceUUIDLeastMost(dynamic2, "Target", "Target").orElse(dynamic2);
                     });
                 });
 
                 Objects.requireNonNull(dynamic1);
-                return (Dynamic) DataFixUtils.orElse(optional.map(dynamic1::createList), dynamic1);
+                return DataFixUtils.orElse(optional.map(dynamic1::createList), dynamic1);
             });
         });
     }
