diff --git a/minecraft/util/datafix/fixes/ItemLoreFix.java b/minecraft/util/datafix/fixes/ItemLoreFix.java
index c52f095..6bfbbbf 100644
--- a/minecraft/util/datafix/fixes/ItemLoreFix.java
+++ b/minecraft/util/datafix/fixes/ItemLoreFix.java
@@ -28,10 +28,10 @@ public class ItemLoreFix extends DataFix {
                 return typed1.update(DSL.remainderFinder(), (dynamic) -> {
                     return dynamic.update("display", (dynamic1) -> {
                         return dynamic1.update("Lore", (dynamic2) -> {
-                            DataResult dataresult = dynamic2.asStreamOpt().map(ItemLoreFix::fixLoreList);
+                            DataResult<Stream> dataresult = dynamic2.asStreamOpt().map(ItemLoreFix::fixLoreList);
 
                             Objects.requireNonNull(dynamic2);
-                            return (Dynamic) DataFixUtils.orElse(dataresult.map(dynamic2::createList).result(), dynamic2);
+                            return DataFixUtils.orElse(dataresult.map(dynamic2::createList).result(), dynamic2);
                         });
                     });
                 });
