diff --git a/src/main/java/net/minecraft/world/item/BlockItem.java b/src/main/java/net/minecraft/world/item/BlockItem.java
index 7e979a9..faefa31 100644
--- a/src/main/java/net/minecraft/world/item/BlockItem.java
+++ b/src/main/java/net/minecraft/world/item/BlockItem.java
@@ -29,6 +29,14 @@ import net.minecraft.world.level.gameevent.GameEvent;
 import net.minecraft.world.level.storage.TagValueOutput;
 import net.minecraft.world.phys.shapes.CollisionContext;
 
+// dish
+import net.minecraft.server.level.ServerLevel;
+import net.ouja.dish.entity.DishPlayer;
+import net.ouja.dish.world.DishLevel;
+import net.ouja.api.event.player.PlayerPlaceBlockEvent;
+import net.ouja.dish.plugins.RegisteredEvents;
+// dish end
+
 public class BlockItem extends Item {
     @Deprecated
     private final Block block;
@@ -53,6 +61,15 @@ public class BlockItem extends Item {
             return InteractionResult.FAIL;
         } else {
             BlockPlaceContext blockPlaceContext = this.updatePlacementContext(context);
+            // dish
+            DishPlayer dishPlayer = new DishPlayer((ServerPlayer)blockPlaceContext.getPlayer());
+            net.ouja.api.world.level.block.Block dishBlock = this.getBlock().toDish(blockPlaceContext.getClickedPos(), blockPlaceContext.getLevel());
+            DishLevel dishLevel = new DishLevel((ServerLevel)blockPlaceContext.getLevel());
+            net.ouja.api.world.level.block.BlockPos dishBlockPos = new net.ouja.api.world.level.block.BlockPos(blockPlaceContext.getClickedPos().getX(), blockPlaceContext.getClickedPos().getY(), blockPlaceContext.getClickedPos().getZ(), dishLevel);
+            PlayerPlaceBlockEvent event = new PlayerPlaceBlockEvent(dishPlayer, dishBlock, dishBlockPos);
+            RegisteredEvents.callEvent(event);
+            if (event.isCancelled()) return InteractionResult.FAIL;
+            // dish end
             if (blockPlaceContext == null) {
                 return InteractionResult.FAIL;
             } else {
