diff --git a/src/main/java/net/minecraft/world/entity/monster/Vindicator.java b/src/main/java/net/minecraft/world/entity/monster/Vindicator.java
index 90574f2..20bf974 100644
--- a/src/main/java/net/minecraft/world/entity/monster/Vindicator.java
+++ b/src/main/java/net/minecraft/world/entity/monster/Vindicator.java
@@ -192,6 +192,12 @@ public class Vindicator extends AbstractIllager {
         this.setItemSlot(EquipmentSlot.MAINHAND, itemstack);
     }
 
+    // dish -- make isJohny a method
+    public boolean isJohnny() {
+        return isJohnny;
+    }
+    // dish end
+
     private static class VindicatorBreakDoorGoal extends BreakDoorGoal {
 
         public VindicatorBreakDoorGoal(Mob mob) {
