diff --git a/src/main/java/net/minecraft/world/entity/animal/armadillo/Armadillo.java b/src/main/java/net/minecraft/world/entity/animal/armadillo/Armadillo.java
index d782bc7..c6249b1 100644
--- a/src/main/java/net/minecraft/world/entity/animal/armadillo/Armadillo.java
+++ b/src/main/java/net/minecraft/world/entity/animal/armadillo/Armadillo.java
@@ -99,7 +99,7 @@ public class Armadillo extends Animal {
         this.entityData.set(Armadillo.ARMADILLO_STATE, armadillo_armadillostate);
     }
 
-    private void setScared(boolean flag) {
+    public void setScared(boolean flag) { // dish -- make this public
         this.switchToState(flag ? Armadillo.ArmadilloState.ROLLING : Armadillo.ArmadilloState.IDLE);
     }
 
