diff --git a/minecraft/world/inventory/BeaconMenu.java b/minecraft/world/inventory/BeaconMenu.java
index 6b9633c..64053ba 100644
--- a/minecraft/world/inventory/BeaconMenu.java
+++ b/minecraft/world/inventory/BeaconMenu.java
@@ -163,8 +163,8 @@ public class BeaconMenu extends AbstractContainerMenu {
 
     public void updateEffects(Optional<MobEffect> optional, Optional<MobEffect> optional1) {
         if (this.paymentSlot.hasItem()) {
-            this.beaconData.set(1, encodeEffect((MobEffect) optional.orElse((Object) null)));
-            this.beaconData.set(2, encodeEffect((MobEffect) optional1.orElse((Object) null)));
+            this.beaconData.set(1, encodeEffect((MobEffect) optional.orElse(null)));
+            this.beaconData.set(2, encodeEffect((MobEffect) optional1.orElse(null)));
             this.paymentSlot.remove(1);
             this.access.execute(Level::blockEntityChanged);
         }
