diff --git a/minecraft/world/scores/ScoreboardSaveData.java b/minecraft/world/scores/ScoreboardSaveData.java
index c0ce898..1e7dc5a 100644
--- a/minecraft/world/scores/ScoreboardSaveData.java
+++ b/minecraft/world/scores/ScoreboardSaveData.java
@@ -144,7 +144,7 @@ public class ScoreboardSaveData extends SavedData {
             MutableComponent mutablecomponent = Component.Serializer.fromJson(compoundtag.getString("DisplayName"));
             ObjectiveCriteria.RenderType objectivecriteria_rendertype = ObjectiveCriteria.RenderType.byId(compoundtag.getString("RenderType"));
             boolean flag = compoundtag.getBoolean("display_auto_update");
-            NumberFormat numberformat = (NumberFormat) NumberFormatTypes.CODEC.parse(NbtOps.INSTANCE, compoundtag.get("format")).result().orElse((Object) null);
+            NumberFormat numberformat = (NumberFormat) NumberFormatTypes.CODEC.parse(NbtOps.INSTANCE, compoundtag.get("format")).result().orElse(null);
 
             this.scoreboard.addObjective(s1, objectivecriteria, mutablecomponent, objectivecriteria_rendertype, flag, numberformat);
         }
@@ -236,8 +236,9 @@ public class ScoreboardSaveData extends SavedData {
             NumberFormat numberformat = objective.numberFormat();
 
             if (numberformat != null) {
+                CompoundTag finalCompoundtag = compoundtag;
                 NumberFormatTypes.CODEC.encodeStart(NbtOps.INSTANCE, numberformat).result().ifPresent((tag) -> {
-                    compoundtag.put("format", tag);
+                    finalCompoundtag.put("format", tag);
                 });
             }
         }
