diff --git a/minecraft/server/commands/FunctionCommand.java b/minecraft/server/commands/FunctionCommand.java
index 745d3be..40fc45c 100644
--- a/minecraft/server/commands/FunctionCommand.java
+++ b/minecraft/server/commands/FunctionCommand.java
@@ -97,14 +97,15 @@ public class FunctionCommand {
         }
 
         if (flag1) {
+            int finalI = i;
             if (collection.size() == 1) {
                 if (flag) {
                     commandsourcestack.sendSuccess(() -> {
-                        return Component.translatable("commands.function.success.single.result", i, ((CommandFunction) collection.iterator().next()).getId());
+                        return Component.translatable("commands.function.success.single.result", finalI, ((CommandFunction) collection.iterator().next()).getId());
                     }, true);
                 } else {
                     commandsourcestack.sendSuccess(() -> {
-                        return Component.translatable("commands.function.success.single", i, ((CommandFunction) collection.iterator().next()).getId());
+                        return Component.translatable("commands.function.success.single", finalI, ((CommandFunction) collection.iterator().next()).getId());
                     }, true);
                 }
             } else if (flag) {
@@ -113,7 +114,7 @@ public class FunctionCommand {
                 }, true);
             } else {
                 commandsourcestack.sendSuccess(() -> {
-                    return Component.translatable("commands.function.success.multiple", i, collection.size());
+                    return Component.translatable("commands.function.success.multiple", finalI, collection.size());
                 }, true);
             }
         }
