diff --git a/minecraft/server/commands/LocateCommand.java b/minecraft/server/commands/LocateCommand.java
index a067f91..e4df100 100644
--- a/minecraft/server/commands/LocateCommand.java
+++ b/minecraft/server/commands/LocateCommand.java
@@ -24,13 +24,16 @@ import net.minecraft.core.Holder;
 import net.minecraft.core.HolderSet;
 import net.minecraft.core.Registry;
 import net.minecraft.core.registries.Registries;
+import net.minecraft.nbt.Tag;
 import net.minecraft.network.chat.ClickEvent;
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.ComponentUtils;
 import net.minecraft.network.chat.HoverEvent;
 import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.resources.ResourceKey;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.level.ServerLevel;
+import net.minecraft.tags.TagKey;
 import net.minecraft.util.Mth;
 import net.minecraft.world.entity.ai.village.poi.PoiManager;
 import net.minecraft.world.entity.ai.village.poi.PoiType;
@@ -74,15 +77,15 @@ public class LocateCommand {
     }
 
     private static Optional<? extends HolderSet.ListBacked<Structure>> getHolders(ResourceOrTagKeyArgument.Result<Structure> resourceortagkeyargument_result, Registry<Structure> registry) {
-        Either either = resourceortagkeyargument_result.unwrap();
+        Either<ResourceKey<Structure>, TagKey<Structure>> either = resourceortagkeyargument_result.unwrap();
         Function function = (resourcekey) -> {
-            return registry.getHolder(resourcekey).map((holder) -> {
+            return registry.getHolder((ResourceKey<Structure>) resourcekey).map((holder) -> {
                 return HolderSet.direct(holder);
             });
         };
 
         Objects.requireNonNull(registry);
-        return (Optional) either.map(function, registry::getTag);
+        return either.map(function, registry::getTag);
     }
 
     private static int locateStructure(CommandSourceStack commandsourcestack, ResourceOrTagKeyArgument.Result<Structure> resourceortagkeyargument_result) throws CommandSyntaxException {
@@ -131,7 +134,7 @@ public class LocateCommand {
     }
 
     private static String getElementName(Pair<BlockPos, ? extends Holder<?>> pair) {
-        return (String) ((Holder) pair.getSecond()).unwrapKey().map((resourcekey) -> {
+        return (String) (pair.getSecond()).unwrapKey().map((resourcekey) -> {
             return resourcekey.location().toString();
         }).orElse("[unregistered]");
     }
