diff --git a/minecraft/server/commands/PardonCommand.java b/minecraft/server/commands/PardonCommand.java
index 2dffd85..ac2e08c 100644
--- a/minecraft/server/commands/PardonCommand.java
+++ b/minecraft/server/commands/PardonCommand.java
@@ -39,7 +39,7 @@ public class PardonCommand {
             GameProfile gameprofile = (GameProfile) iterator.next();
 
             if (userbanlist.isBanned(gameprofile)) {
-                userbanlist.remove((Object) gameprofile);
+                userbanlist.remove(gameprofile);
                 ++i;
                 commandsourcestack.sendSuccess(() -> {
                     return Component.translatable("commands.pardon.success", Component.literal(gameprofile.getName()));
