diff --git a/minecraft/server/commands/SpawnArmorTrimsCommand.java b/minecraft/server/commands/SpawnArmorTrimsCommand.java
index e212c92..c8d80ae 100644
--- a/minecraft/server/commands/SpawnArmorTrimsCommand.java
+++ b/minecraft/server/commands/SpawnArmorTrimsCommand.java
@@ -81,10 +81,10 @@ public class SpawnArmorTrimsCommand {
         Registry<TrimMaterial> registry1 = level.registryAccess().registryOrThrow(Registries.TRIM_MATERIAL);
 
         registry.stream().sorted(Comparator.comparing((trimpattern) -> {
-            return SpawnArmorTrimsCommand.TRIM_PATTERN_ORDER.applyAsInt((ResourceKey) registry.getResourceKey(trimpattern).orElse((Object) null));
+            return SpawnArmorTrimsCommand.TRIM_PATTERN_ORDER.applyAsInt((ResourceKey) registry.getResourceKey(trimpattern).orElse(null));
         })).forEachOrdered((trimpattern) -> {
             registry1.stream().sorted(Comparator.comparing((trimmaterial) -> {
-                return SpawnArmorTrimsCommand.TRIM_MATERIAL_ORDER.applyAsInt((ResourceKey) registry1.getResourceKey(trimmaterial).orElse((Object) null));
+                return SpawnArmorTrimsCommand.TRIM_MATERIAL_ORDER.applyAsInt((ResourceKey) registry1.getResourceKey(trimmaterial).orElse(null));
             })).forEachOrdered((trimmaterial) -> {
                 nonnulllist.add(new ArmorTrim(registry1.wrapAsHolder(trimmaterial), registry.wrapAsHolder(trimpattern)));
             });
