diff --git a/src/main/java/net/ouja/dish/entity/monster/DishMagmaCube.java b/src/main/java/net/ouja/dish/entity/monster/DishMagmaCube.java
new file mode 100644
index 0000000..ac07db3
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/monster/DishMagmaCube.java
@@ -0,0 +1,14 @@
+package net.ouja.dish.entity.monster;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.monster.MagmaCube;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishMagmaCube extends DishEntity implements net.ouja.api.entity.monster.MagmaCube {
+    private MagmaCube magmaCube;
+
+    public DishMagmaCube(Entity entity) {
+        super(entity);
+        this.magmaCube = (MagmaCube)entity;
+    }
+}
