diff --git a/src/main/java/net/ouja/dish/entity/monster/DishZombieHorse.java b/src/main/java/net/ouja/dish/entity/monster/DishZombieHorse.java
new file mode 100644
index 0000000..1f37762
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/monster/DishZombieHorse.java
@@ -0,0 +1,14 @@
+package net.ouja.dish.entity.monster;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.animal.horse.ZombieHorse;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishZombieHorse extends DishEntity implements net.ouja.api.entity.monster.ZombieHorse {
+    private ZombieHorse zombieHorse;
+
+    public DishZombieHorse(Entity entity) {
+        super(entity);
+        this.zombieHorse = (ZombieHorse)entity;
+    }
+}
