diff --git a/src/main/java/net/ouja/dish/entity/monster/DishWarden.java b/src/main/java/net/ouja/dish/entity/monster/DishWarden.java
new file mode 100644
index 0000000..3ad3187
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/monster/DishWarden.java
@@ -0,0 +1,14 @@
+package net.ouja.dish.entity.monster;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.monster.warden.Warden;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishWarden extends DishEntity implements net.ouja.api.entity.monster.Warden {
+    private Warden warden;
+
+    public DishWarden(Entity entity) {
+        super(entity);
+        this.warden = (Warden)entity;
+    }
+}
