diff --git a/src/main/java/net/ouja/dish/entity/monster/DishPhantom.java b/src/main/java/net/ouja/dish/entity/monster/DishPhantom.java
new file mode 100644
index 0000000..85d9744
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/monster/DishPhantom.java
@@ -0,0 +1,14 @@
+package net.ouja.dish.entity.monster;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.monster.Phantom;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishPhantom extends DishEntity implements net.ouja.api.entity.monster.Phantom {
+    private Phantom phantom;
+
+    public DishPhantom(Entity entity) {
+        super(entity);
+        this.phantom = (Phantom)entity;
+    }
+}
