diff --git a/src/main/java/net/ouja/dish/entity/monster/DishStray.java b/src/main/java/net/ouja/dish/entity/monster/DishStray.java
new file mode 100644
index 0000000..67ae4d2
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/monster/DishStray.java
@@ -0,0 +1,14 @@
+package net.ouja.dish.entity.monster;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.monster.Stray;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishStray extends DishEntity implements net.ouja.api.entity.monster.Stray {
+    private Stray stray;
+
+    public DishStray(Entity entity) {
+        super(entity);
+        this.stray = (Stray)entity;
+    }
+}
