diff --git a/src/main/java/net/ouja/dish/entity/monster/DishBlaze.java b/src/main/java/net/ouja/dish/entity/monster/DishBlaze.java
new file mode 100644
index 0000000..f28aacd
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/monster/DishBlaze.java
@@ -0,0 +1,19 @@
+package net.ouja.dish.entity.monster;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.monster.Blaze;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishBlaze extends DishEntity implements net.ouja.api.entity.monster.Blaze {
+    private Blaze blaze;
+
+    public DishBlaze(Entity entity) {
+        super(entity);
+        this.blaze = (Blaze)entity;
+    }
+
+    @Override
+    public boolean isOnFire() {
+        return this.blaze.isOnFire();
+    }
+}
