diff --git a/src/main/java/net/ouja/dish/entity/monster/DishSlime.java b/src/main/java/net/ouja/dish/entity/monster/DishSlime.java
new file mode 100644
index 0000000..95202ce
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/monster/DishSlime.java
@@ -0,0 +1,19 @@
+package net.ouja.dish.entity.monster;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.monster.Slime;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishSlime extends DishEntity implements net.ouja.api.entity.monster.Slime {
+    private Slime slime;
+
+    public DishSlime(Entity entity) {
+        super(entity);
+        this.slime = (Slime)entity;
+    }
+
+    @Override
+    public int getSize() {
+        return this.slime.getSize();
+    }
+}
