diff --git a/src/main/java/net/ouja/dish/entity/monster/DishSilverfish.java b/src/main/java/net/ouja/dish/entity/monster/DishSilverfish.java
new file mode 100644
index 0000000..f3c6970
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/monster/DishSilverfish.java
@@ -0,0 +1,14 @@
+package net.ouja.dish.entity.monster;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.monster.Silverfish;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishSilverfish extends DishEntity implements net.ouja.api.entity.monster.Silverfish {
+    private Silverfish silverfish;
+
+    public DishSilverfish(Entity entity) {
+        super(entity);
+        this.silverfish = (Silverfish)entity;
+    }
+}
