diff --git a/src/main/java/net/ouja/dish/entity/monster/DishElderGuardian.java b/src/main/java/net/ouja/dish/entity/monster/DishElderGuardian.java
new file mode 100644
index 0000000..83ac428
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/monster/DishElderGuardian.java
@@ -0,0 +1,14 @@
+package net.ouja.dish.entity.monster;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.monster.ElderGuardian;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishElderGuardian extends DishEntity implements net.ouja.api.entity.monster.ElderGuardian {
+    private ElderGuardian elderGuardian;
+
+    public DishElderGuardian(Entity entity) {
+        super(entity);
+        this.elderGuardian = (ElderGuardian)entity;
+    }
+}
