diff --git a/src/main/java/net/ouja/dish/entity/monster/DishWitch.java b/src/main/java/net/ouja/dish/entity/monster/DishWitch.java
new file mode 100644
index 0000000..9cb168a
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/monster/DishWitch.java
@@ -0,0 +1,14 @@
+package net.ouja.dish.entity.monster;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.monster.Witch;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishWitch extends DishEntity implements net.ouja.api.entity.monster.Witch {
+    private Witch witch;
+
+    public DishWitch(Entity entity) {
+        super(entity);
+        this.witch = (Witch)entity;
+    }
+}
