diff --git a/src/main/java/net/minecraft/server/Main.java b/src/main/java/net/minecraft/server/Main.java
index e8a5576..e21f361 100644
--- a/src/main/java/net/minecraft/server/Main.java
+++ b/src/main/java/net/minecraft/server/Main.java
@@ -120,6 +120,12 @@ public class Main {
                 return;
             }
 
+            // dish -- create a plugins directory
+            if (!new File("plugins").exists()) {
+                Files.createDirectories(Path.of("plugins"));
+            }
+            // dish end
+
             if (!eula.hasAgreedToEULA()) {
                 Main.LOGGER.info("You need to agree to the EULA in order to run the server. Go to eula.txt for more info.");
                 return;
