diff --git a/minecraft/network/protocol/BundlerInfo.java b/minecraft/network/protocol/BundlerInfo.java
index 5a6b22a..dbea1a3 100644
--- a/minecraft/network/protocol/BundlerInfo.java
+++ b/minecraft/network/protocol/BundlerInfo.java
@@ -28,7 +28,7 @@ public interface BundlerInfo {
             @Override
             public void unbundlePacket(Packet<?> packet, Consumer<Packet<?>> consumer) {
                 if (packet.getClass() == oclass) {
-                    P p0 = (BundlePacket) packet;
+                    P p0 = (P) packet;
 
                     consumer.accept(bundledelimiterpacket);
                     p0.subPackets().forEach(consumer);
@@ -53,7 +53,7 @@ public interface BundlerInfo {
                         } else if (this.bundlePackets.size() >= 4096) {
                             throw new IllegalStateException("Too many packets in a bundle");
                         } else {
-                            this.bundlePackets.add(packet1);
+                            this.bundlePackets.add((Packet<T>) packet1);
                             return null;
                         }
                     }
