diff --git a/minecraft/world/level/levelgen/structure/templatesystem/PosRuleTestType.java b/minecraft/world/level/levelgen/structure/templatesystem/PosRuleTestType.java
index a5bace2..1ff916c 100644
--- a/minecraft/world/level/levelgen/structure/templatesystem/PosRuleTestType.java
+++ b/minecraft/world/level/levelgen/structure/templatesystem/PosRuleTestType.java
@@ -13,7 +13,7 @@ public interface PosRuleTestType<P extends PosRuleTest> {
     Codec<P> codec();
 
     static <P extends PosRuleTest> PosRuleTestType<P> register(String s, Codec<P> codec) {
-        return (PosRuleTestType) Registry.register(BuiltInRegistries.POS_RULE_TEST, s, () -> {
+        return Registry.register(BuiltInRegistries.POS_RULE_TEST, s, () -> {
             return codec;
         });
     }
