diff --git a/minecraft/util/datafix/DataFixTypes.java b/minecraft/util/datafix/DataFixTypes.java
index 90e11b4..d815fc7 100644
--- a/minecraft/util/datafix/DataFixTypes.java
+++ b/minecraft/util/datafix/DataFixTypes.java
@@ -38,7 +38,7 @@ public enum DataFixTypes {
             }
 
             public <T> DataResult<Pair<A, T>> decode(DynamicOps<T> dynamicops, T t0) {
-                DataResult dataresult = dynamicops.get(t0, "DataVersion");
+                DataResult<T> dataresult = dynamicops.get(t0, "DataVersion");
 
                 Objects.requireNonNull(dynamicops);
                 int j = (Integer) dataresult.flatMap(dynamicops::getNumberValue).map(Number::intValue).result().orElse(i);
