diff --git a/minecraft/world/level/levelgen/structure/structures/NetherFortressStructure.java b/minecraft/world/level/levelgen/structure/structures/NetherFortressStructure.java
index 70b83c4..79d6ce1 100644
--- a/minecraft/world/level/levelgen/structure/structures/NetherFortressStructure.java
+++ b/minecraft/world/level/levelgen/structure/structures/NetherFortressStructure.java
@@ -16,7 +16,7 @@ import net.minecraft.world.level.levelgen.structure.pieces.StructurePiecesBuilde
 
 public class NetherFortressStructure extends Structure {
 
-    public static final WeightedRandomList<MobSpawnSettings.SpawnerData> FORTRESS_ENEMIES = WeightedRandomList.create((WeightedEntry[])(new MobSpawnSettings.SpawnerData(EntityType.BLAZE, 10, 2, 3), new MobSpawnSettings.SpawnerData(EntityType.ZOMBIFIED_PIGLIN, 5, 4, 4), new MobSpawnSettings.SpawnerData(EntityType.WITHER_SKELETON, 8, 5, 5), new MobSpawnSettings.SpawnerData(EntityType.SKELETON, 2, 5, 5), new MobSpawnSettings.SpawnerData(EntityType.MAGMA_CUBE, 3, 4, 4)));
+    public static final WeightedRandomList<MobSpawnSettings.SpawnerData> FORTRESS_ENEMIES = WeightedRandomList.create(new MobSpawnSettings.SpawnerData(EntityType.BLAZE, 10, 2, 3), new MobSpawnSettings.SpawnerData(EntityType.ZOMBIFIED_PIGLIN, 5, 4, 4), new MobSpawnSettings.SpawnerData(EntityType.WITHER_SKELETON, 8, 5, 5), new MobSpawnSettings.SpawnerData(EntityType.SKELETON, 2, 5, 5), new MobSpawnSettings.SpawnerData(EntityType.MAGMA_CUBE, 3, 4, 4));
     public static final Codec<NetherFortressStructure> CODEC = simpleCodec(NetherFortressStructure::new);
 
     public NetherFortressStructure(Structure.StructureSettings structure_structuresettings) {
