diff --git a/minecraft/world/level/block/state/properties/WoodType.java b/minecraft/world/level/block/state/properties/WoodType.java
index 6a1bb09..e1f437b 100644
--- a/minecraft/world/level/block/state/properties/WoodType.java
+++ b/minecraft/world/level/block/state/properties/WoodType.java
@@ -41,11 +41,7 @@ public record WoodType(String name, BlockSetType setType, SoundType soundType, S
     }
 
     static {
-        Function function = WoodType::name;
-        Map map = WoodType.TYPES;
-
-        Objects.requireNonNull(map);
-        CODEC = ExtraCodecs.stringResolverCodec(function, map::get);
+        CODEC = ExtraCodecs.stringResolverCodec(WoodType::name, WoodType.TYPES::get);
         OAK = register(new WoodType("oak", BlockSetType.OAK));
         SPRUCE = register(new WoodType("spruce", BlockSetType.SPRUCE));
         BIRCH = register(new WoodType("birch", BlockSetType.BIRCH));
