diff --git a/minecraft/util/valueproviders/ClampedNormalInt.java b/minecraft/util/valueproviders/ClampedNormalInt.java
index ef3f821..3d91dde 100644
--- a/minecraft/util/valueproviders/ClampedNormalInt.java
+++ b/minecraft/util/valueproviders/ClampedNormalInt.java
@@ -9,7 +9,7 @@ import net.minecraft.util.RandomSource;
 
 public class ClampedNormalInt extends IntProvider {
 
-    public static final Codec<ClampedNormalInt> CODEC = RecordCodecBuilder.create((instance) -> {
+    public static final Codec<ClampedNormalInt> CODEC = RecordCodecBuilder.<ClampedNormalInt>create((instance) -> {
         return instance.group(Codec.FLOAT.fieldOf("mean").forGetter((clampednormalint) -> {
             return clampednormalint.mean;
         }), Codec.FLOAT.fieldOf("deviation").forGetter((clampednormalint) -> {
