diff --git a/minecraft/util/valueproviders/FloatProviderType.java b/minecraft/util/valueproviders/FloatProviderType.java
index ddf4a1f..ee97c8a 100644
--- a/minecraft/util/valueproviders/FloatProviderType.java
+++ b/minecraft/util/valueproviders/FloatProviderType.java
@@ -14,7 +14,7 @@ public interface FloatProviderType<P extends FloatProvider> {
     Codec<P> codec();
 
     static <P extends FloatProvider> FloatProviderType<P> register(String s, Codec<P> codec) {
-        return (FloatProviderType) Registry.register(BuiltInRegistries.FLOAT_PROVIDER_TYPE, s, () -> {
+        return Registry.register(BuiltInRegistries.FLOAT_PROVIDER_TYPE, s, () -> {
             return codec;
         });
     }
