diff --git a/minecraft/util/valueproviders/IntProviderType.java b/minecraft/util/valueproviders/IntProviderType.java
index 6bf9f2d..3ccca45 100644
--- a/minecraft/util/valueproviders/IntProviderType.java
+++ b/minecraft/util/valueproviders/IntProviderType.java
@@ -16,7 +16,7 @@ public interface IntProviderType<P extends IntProvider> {
     Codec<P> codec();
 
     static <P extends IntProvider> IntProviderType<P> register(String s, Codec<P> codec) {
-        return (IntProviderType) Registry.register(BuiltInRegistries.INT_PROVIDER_TYPE, s, () -> {
+        return Registry.register(BuiltInRegistries.INT_PROVIDER_TYPE, s, () -> {
             return codec;
         });
     }
