diff --git a/minecraft/tags/TagLoader.java b/minecraft/tags/TagLoader.java
index a15d5f5..18a1bba 100644
--- a/minecraft/tags/TagLoader.java
+++ b/minecraft/tags/TagLoader.java
@@ -63,7 +63,7 @@ public class TagLoader<T> {
                         List<TagLoader.EntryWithSource> list = (List) map.computeIfAbsent(resourcelocation1, (resourcelocation2) -> {
                             return new ArrayList();
                         });
-                        DataResult dataresult = TagFile.CODEC.parse(new Dynamic(JsonOps.INSTANCE, jsonelement));
+                        DataResult<TagFile> dataresult = TagFile.CODEC.parse(new Dynamic(JsonOps.INSTANCE, jsonelement));
                         Logger logger = TagLoader.LOGGER;
 
                         Objects.requireNonNull(logger);
@@ -126,7 +126,7 @@ public class TagLoader<T> {
             @Nullable
             @Override
             public T element(ResourceLocation resourcelocation) {
-                return ((Optional) TagLoader.this.idToValue.apply(resourcelocation)).orElse((Object) null);
+                return (TagLoader.this.idToValue.apply(resourcelocation)).orElse(null);
             }
 
             @Nullable
