diff --git a/minecraft/commands/Commands.java b/minecraft/commands/Commands.java
index 25ec79f..15938e3 100644
--- a/minecraft/commands/Commands.java
+++ b/minecraft/commands/Commands.java
@@ -363,7 +363,7 @@ public class Commands {
 
                 executioncontext1.close();
             } finally {
-                Commands.CURRENT_EXECUTION_CONTEXT.set((Object) null);
+                Commands.CURRENT_EXECUTION_CONTEXT.set(null);
             }
         } else {
             consumer.accept(executioncontext);
@@ -387,7 +387,7 @@ public class Commands {
             CommandNode<CommandSourceStack> commandnode2 = (CommandNode) iterator.next();
 
             if (commandnode2.canUse(commandsourcestack)) {
-                ArgumentBuilder<SharedSuggestionProvider, ?> argumentbuilder = commandnode2.createBuilder();
+                ArgumentBuilder<SharedSuggestionProvider, ?> argumentbuilder = (ArgumentBuilder) commandnode2.createBuilder();
 
                 argumentbuilder.requires((sharedsuggestionprovider) -> {
                     return true;
