diff --git a/src/main/java/net/minecraft/world/entity/monster/hoglin/Hoglin.java b/src/main/java/net/minecraft/world/entity/monster/hoglin/Hoglin.java
index 0bf65e0..c8fa34c 100644
--- a/src/main/java/net/minecraft/world/entity/monster/hoglin/Hoglin.java
+++ b/src/main/java/net/minecraft/world/entity/monster/hoglin/Hoglin.java
@@ -304,7 +304,7 @@ public class Hoglin extends Animal implements Enemy, HoglinBase {
         this.getEntityData().set(DATA_IMMUNE_TO_ZOMBIFICATION, immuneToZombification);
     }
 
-    private boolean isImmuneToZombification() {
+    public boolean isImmuneToZombification() { // dish -- make this public
         return this.getEntityData().get(DATA_IMMUNE_TO_ZOMBIFICATION);
     }
 
