diff --git a/minecraft/util/profiling/jfr/JfrProfiler.java b/minecraft/util/profiling/jfr/JfrProfiler.java
index 3bbbb10..b7b5df3 100644
--- a/minecraft/util/profiling/jfr/JfrProfiler.java
+++ b/minecraft/util/profiling/jfr/JfrProfiler.java
@@ -146,7 +146,7 @@ public class JfrProfiler implements JvmProfiler {
                 String s = JfrProfiler.DATE_TIME_FORMATTER.format(Instant.now());
 
                 this.recording = (Recording) Util.make(new Recording(configuration), (recording) -> {
-                    List list = JfrProfiler.CUSTOM_EVENTS;
+                    List<Class<? extends Event>> list = JfrProfiler.CUSTOM_EVENTS;
 
                     Objects.requireNonNull(recording);
                     list.forEach(recording::enable);
