diff --git a/minecraft/world/inventory/BeaconMenu.java b/minecraft/world/inventory/BeaconMenu.java
index 0c8f878..067734e 100644
--- a/minecraft/world/inventory/BeaconMenu.java
+++ b/minecraft/world/inventory/BeaconMenu.java
@@ -164,8 +164,8 @@ public class BeaconMenu extends AbstractContainerMenu {
 
     public void updateEffects(Optional<Holder<MobEffect>> optional, Optional<Holder<MobEffect>> optional1) {
         if (this.paymentSlot.hasItem()) {
-            this.beaconData.set(1, encodeEffect((Holder) optional.orElse((Object) null)));
-            this.beaconData.set(2, encodeEffect((Holder) optional1.orElse((Object) null)));
+            this.beaconData.set(1, encodeEffect((Holder) optional.orElse(null)));
+            this.beaconData.set(2, encodeEffect((Holder) optional1.orElse( null)));
             this.paymentSlot.remove(1);
             this.access.execute(Level::blockEntityChanged);
         }
