diff --git a/src/main/java/net/minecraft/world/entity/player/Player.java b/src/main/java/net/minecraft/world/entity/player/Player.java
index cdd4043..0866480 100644
--- a/src/main/java/net/minecraft/world/entity/player/Player.java
+++ b/src/main/java/net/minecraft/world/entity/player/Player.java
@@ -672,6 +672,11 @@ public abstract class Player extends LivingEntity {
 
     @Nullable
     public ItemEntity drop(ItemStack itemstack, boolean flag, boolean flag1) {
+        // dish
+        net.ouja.api.event.player.PlayDropItemEvent event = new net.ouja.api.event.player.PlayDropItemEvent(new net.ouja.dish.entity.DishPlayer((ServerPlayer)this), itemstack.toDish());
+        net.ouja.dish.plugins.RegisteredEvents.callEvent(event);
+        if (event.isCancelled()) return null;
+        // dish end
         if (itemstack.isEmpty()) {
             return null;
         } else {
@@ -682,7 +687,7 @@ public abstract class Player extends LivingEntity {
             double d0 = this.getEyeY() - 0.30000001192092896D;
             ItemEntity itementity = new ItemEntity(this.level(), this.getX(), d0, this.getZ(), itemstack);
 
-            itementity.setPickUpDelay(40);
+            itementity.setPickUpDelay(event.getPickUpDelay()); // dish -- allow plugins to modify pick up delay. Default is 40
             if (flag1) {
                 itementity.setThrower(this);
             }
