diff --git a/minecraft/world/entity/monster/breeze/Breeze.java b/minecraft/world/entity/monster/breeze/Breeze.java
index b07dd0e..8fff15b 100644
--- a/minecraft/world/entity/monster/breeze/Breeze.java
+++ b/minecraft/world/entity/monster/breeze/Breeze.java
@@ -60,7 +60,7 @@ public class Breeze extends Monster {
 
     @Override
     public Brain<Breeze> getBrain() {
-        return super.getBrain();
+        return (Brain<Breeze>) super.getBrain();
     }
 
     @Override
