diff --git a/minecraft/world/entity/monster/breeze/LongJump.java b/minecraft/world/entity/monster/breeze/LongJump.java
index bcab8fb..f7d5bf6 100644
--- a/minecraft/world/entity/monster/breeze/LongJump.java
+++ b/minecraft/world/entity/monster/breeze/LongJump.java
@@ -50,7 +50,7 @@ public class LongJump extends Behavior<Breeze> {
         } else if (breeze.getBrain().checkMemory(MemoryModuleType.BREEZE_JUMP_TARGET, MemoryStatus.VALUE_PRESENT)) {
             return true;
         } else {
-            LivingEntity livingentity = (LivingEntity) breeze.getBrain().getMemory(MemoryModuleType.ATTACK_TARGET).orElse((Object) null);
+            LivingEntity livingentity = (LivingEntity) breeze.getBrain().getMemory(MemoryModuleType.ATTACK_TARGET).orElse(null);
 
             if (livingentity == null) {
                 return false;
@@ -69,7 +69,7 @@ public class LongJump extends Behavior<Breeze> {
                 } else if (!hasLineOfSight(breeze, blockpos.getCenter()) && !hasLineOfSight(breeze, blockpos.above(4).getCenter())) {
                     return false;
                 } else {
-                    breeze.getBrain().setMemory(MemoryModuleType.BREEZE_JUMP_TARGET, (Object) blockpos);
+                    breeze.getBrain().setMemory(MemoryModuleType.BREEZE_JUMP_TARGET, blockpos);
                     return true;
                 }
             }
@@ -95,7 +95,7 @@ public class LongJump extends Behavior<Breeze> {
         if (finishedInhaling(breeze)) {
             Vec3 vec3 = (Vec3) breeze.getBrain().getMemory(MemoryModuleType.BREEZE_JUMP_TARGET).flatMap((blockpos) -> {
                 return calculateOptimalJumpVector(breeze, breeze.getRandom(), Vec3.atBottomCenterOf(blockpos));
-            }).orElse((Object) null);
+            }).orElse(null);
 
             if (vec3 == null) {
                 breeze.setPose(Pose.STANDING);
