diff --git a/minecraft/world/level/levelgen/structure/structures/OceanMonumentPieces.java b/minecraft/world/level/levelgen/structure/structures/OceanMonumentPieces.java
index fffbbad..300571d 100644
--- a/minecraft/world/level/levelgen/structure/structures/OceanMonumentPieces.java
+++ b/minecraft/world/level/levelgen/structure/structures/OceanMonumentPieces.java
@@ -1842,7 +1842,7 @@ public class OceanMonumentPieces {
         protected static final BlockState LAMP_BLOCK = Blocks.SEA_LANTERN.defaultBlockState();
         protected static final boolean DO_FILL = true;
         protected static final BlockState FILL_BLOCK = Blocks.WATER.defaultBlockState();
-        protected static final Set<Block> FILL_KEEP = ImmutableSet.builder().add(Blocks.ICE).add(Blocks.PACKED_ICE).add(Blocks.BLUE_ICE).add(OceanMonumentPieces.OceanMonumentPiece.FILL_BLOCK.getBlock()).build();
+        protected static final Set<Block> FILL_KEEP = ImmutableSet.<Block>builder().add(Blocks.ICE).add(Blocks.PACKED_ICE).add(Blocks.BLUE_ICE).add(OceanMonumentPieces.OceanMonumentPiece.FILL_BLOCK.getBlock()).build();
         protected static final int GRIDROOM_WIDTH = 8;
         protected static final int GRIDROOM_DEPTH = 8;
         protected static final int GRIDROOM_HEIGHT = 4;
