diff --git a/src/main/java/net/minecraft/world/level/levelgen/BitRandomSource.java b/src/main/java/net/minecraft/world/level/levelgen/BitRandomSource.java
index e2e46f4..a7f7f74 100644
--- a/src/main/java/net/minecraft/world/level/levelgen/BitRandomSource.java
+++ b/src/main/java/net/minecraft/world/level/levelgen/BitRandomSource.java
@@ -18,7 +18,7 @@ public interface BitRandomSource extends RandomSource {
         if (bound <= 0) {
             throw new IllegalArgumentException("Bound must be positive");
         } else if ((bound & bound - 1) == 0) {
-            return (int)(bound * this.next(31) >> 31);
+            return (int)((long)bound * (long)this.next(31) >> 31); // fix decompiler error
         } else {
             int i;
             int i1;
