diff --git a/minecraft/world/level/levelgen/OreVeinifier.java b/minecraft/world/level/levelgen/OreVeinifier.java
index 3793bfd..404f762 100644
--- a/minecraft/world/level/levelgen/OreVeinifier.java
+++ b/minecraft/world/level/levelgen/OreVeinifier.java
@@ -20,7 +20,7 @@ public final class OreVeinifier {
     private OreVeinifier() {}
 
     protected static NoiseChunk.BlockStateFiller create(DensityFunction densityfunction, DensityFunction densityfunction1, DensityFunction densityfunction2, PositionalRandomFactory positionalrandomfactory) {
-        Object object = null;
+        BlockState object = null;
 
         return (densityfunction_functioncontext) -> {
             double d0 = densityfunction.compute(densityfunction_functioncontext);
