diff --git a/minecraft/world/level/block/entity/BlockEntityType.java b/minecraft/world/level/block/entity/BlockEntityType.java
index 5d65af8..1f06b89 100644
--- a/minecraft/world/level/block/entity/BlockEntityType.java
+++ b/minecraft/world/level/block/entity/BlockEntityType.java
@@ -109,7 +109,7 @@ public class BlockEntityType<T extends BlockEntity> {
     public T getBlockEntity(BlockGetter blockgetter, BlockPos blockpos) {
         BlockEntity blockentity = blockgetter.getBlockEntity(blockpos);
 
-        return blockentity != null && blockentity.getType() == this ? blockentity : null;
+        return blockentity != null && blockentity.getType() == this ? (T) blockentity : null;
     }
 
     public static final class Builder<T extends BlockEntity> {
