diff --git a/minecraft/world/level/block/entity/SignBlockEntity.java b/minecraft/world/level/block/entity/SignBlockEntity.java
index 3d61924..c87a102 100644
--- a/minecraft/world/level/block/entity/SignBlockEntity.java
+++ b/minecraft/world/level/block/entity/SignBlockEntity.java
@@ -13,6 +13,7 @@ import net.minecraft.commands.CommandSourceStack;
 import net.minecraft.core.BlockPos;
 import net.minecraft.nbt.CompoundTag;
 import net.minecraft.nbt.NbtOps;
+import net.minecraft.nbt.Tag;
 import net.minecraft.network.chat.ClickEvent;
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.ComponentUtils;
@@ -98,7 +99,7 @@ public class SignBlockEntity extends BlockEntity {
     @Override
     protected void saveAdditional(CompoundTag compoundtag) {
         super.saveAdditional(compoundtag);
-        DataResult dataresult = SignText.DIRECT_CODEC.encodeStart(NbtOps.INSTANCE, this.frontText);
+        DataResult<Tag> dataresult = SignText.DIRECT_CODEC.encodeStart(NbtOps.INSTANCE, this.frontText);
         Logger logger = SignBlockEntity.LOGGER;
 
         Objects.requireNonNull(logger);
@@ -117,7 +118,7 @@ public class SignBlockEntity extends BlockEntity {
     @Override
     public void load(CompoundTag compoundtag) {
         super.load(compoundtag);
-        DataResult dataresult;
+        DataResult<SignText> dataresult;
         Logger logger;
 
         if (compoundtag.contains("front_text")) {
