diff --git a/minecraft/world/effect/MobEffectInstance.java b/minecraft/world/effect/MobEffectInstance.java
index e902a51..5c2f069 100644
--- a/minecraft/world/effect/MobEffectInstance.java
+++ b/minecraft/world/effect/MobEffectInstance.java
@@ -289,7 +289,7 @@ public class MobEffectInstance implements Comparable<MobEffectInstance> {
 
         return resourcelocation == null ? null : (MobEffectInstance) BuiltInRegistries.MOB_EFFECT.getHolder(resourcelocation).map((holder_reference) -> {
             return loadSpecifiedEffect(holder_reference, compoundtag);
-        }).orElse((Object) null);
+        }).orElse(null);
     }
 
     private static MobEffectInstance loadSpecifiedEffect(Holder<MobEffect> holder, CompoundTag compoundtag) {
