diff --git a/minecraft/network/protocol/game/ClientboundUpdateMobEffectPacket.java b/minecraft/network/protocol/game/ClientboundUpdateMobEffectPacket.java
index 8b4fa07..091feca 100644
--- a/minecraft/network/protocol/game/ClientboundUpdateMobEffectPacket.java
+++ b/minecraft/network/protocol/game/ClientboundUpdateMobEffectPacket.java
@@ -42,7 +42,7 @@ public class ClientboundUpdateMobEffectPacket implements Packet<ClientGamePacket
         }
 
         this.flags = b0;
-        this.factorData = (MobEffectInstance.FactorData) mobeffectinstance.getFactorData().orElse((Object) null);
+        this.factorData = (MobEffectInstance.FactorData) mobeffectinstance.getFactorData().orElse(null);
     }
 
     public ClientboundUpdateMobEffectPacket(FriendlyByteBuf friendlybytebuf) {
