diff --git a/minecraft/network/protocol/game/ClientboundCommandsPacket.java b/minecraft/network/protocol/game/ClientboundCommandsPacket.java
index b4a8298..9b0f6f6 100644
--- a/minecraft/network/protocol/game/ClientboundCommandsPacket.java
+++ b/minecraft/network/protocol/game/ClientboundCommandsPacket.java
@@ -323,7 +323,7 @@ public class ClientboundCommandsPacket implements Packet<ClientGamePacketListene
 
         private static <A extends ArgumentType<?>, T extends ArgumentTypeInfo.Template<A>> void serializeCap(FriendlyByteBuf friendlybytebuf, ArgumentTypeInfo<A, T> argumenttypeinfo, ArgumentTypeInfo.Template<A> argumenttypeinfo_template) {
             friendlybytebuf.writeVarInt(BuiltInRegistries.COMMAND_ARGUMENT_TYPE.getId(argumenttypeinfo));
-            argumenttypeinfo.serializeToNetwork(argumenttypeinfo_template, friendlybytebuf);
+            argumenttypeinfo.serializeToNetwork((T) argumenttypeinfo_template, friendlybytebuf);
         }
     }
 
@@ -368,7 +368,7 @@ public class ClientboundCommandsPacket implements Packet<ClientGamePacketListene
                 return commandnode;
             } else {
                 ClientboundCommandsPacket.Entry clientboundcommandspacket_entry = (ClientboundCommandsPacket.Entry) this.entries.get(i);
-                Object object;
+                CommandNode<SharedSuggestionProvider> object;
 
                 if (clientboundcommandspacket_entry.stub == null) {
                     object = new RootCommandNode();
