diff --git a/minecraft/world/level/levelgen/FlatLevelSource.java b/minecraft/world/level/levelgen/FlatLevelSource.java
index 76adc0d..419bfef 100644
--- a/minecraft/world/level/levelgen/FlatLevelSource.java
+++ b/minecraft/world/level/levelgen/FlatLevelSource.java
@@ -35,10 +35,7 @@ public class FlatLevelSource extends ChunkGenerator {
     private final FlatLevelGeneratorSettings settings;
 
     public FlatLevelSource(FlatLevelGeneratorSettings flatlevelgeneratorsettings) {
-        FixedBiomeSource fixedbiomesource = new FixedBiomeSource(flatlevelgeneratorsettings.getBiome());
-
-        Objects.requireNonNull(flatlevelgeneratorsettings);
-        super(fixedbiomesource, Util.memoize(flatlevelgeneratorsettings::adjustGenerationSettings));
+        super(new FixedBiomeSource(flatlevelgeneratorsettings.getBiome()), Util.memoize(flatlevelgeneratorsettings::adjustGenerationSettings));
         this.settings = flatlevelgeneratorsettings;
     }
 
