diff --git a/minecraft/world/level/levelgen/RandomState.java b/minecraft/world/level/levelgen/RandomState.java
index 31640d6..b479c1b 100644
--- a/minecraft/world/level/levelgen/RandomState.java
+++ b/minecraft/world/level/levelgen/RandomState.java
@@ -43,11 +43,11 @@ public final class RandomState {
         this.surfaceSystem = new SurfaceSystem(this, noisegeneratorsettings.defaultBlock(), noisegeneratorsettings.seaLevel(), this.random);
         final boolean flag = noisegeneratorsettings.useLegacyRandomSource();
 
-        class 1NoiseWiringHelper implements DensityFunction.Visitor {
+        class NoiseWiringHelper implements DensityFunction.Visitor {
 
             private final Map<DensityFunction, DensityFunction> wrapped = new HashMap();
 
-            _NoiseWiringHelper/* $FF was: 1NoiseWiringHelper*/() {
+            NoiseWiringHelper/* $FF was: 1NoiseWiringHelper*/() {
 }
 
             private RandomSource newLegacyInstance(long j) {
@@ -98,7 +98,7 @@ public final class RandomState {
         }
 
 
-        this.router = noisegeneratorsettings.noiseRouter().mapAll(new 1NoiseWiringHelper());
+        this.router = noisegeneratorsettings.noiseRouter().mapAll(new NoiseWiringHelper());
         DensityFunction.Visitor densityfunction_visitor = new DensityFunction.Visitor() {
             private final Map<DensityFunction, DensityFunction> wrapped = new HashMap();
 
