diff --git a/minecraft/world/entity/ai/sensing/PiglinBruteSpecificSensor.java b/minecraft/world/entity/ai/sensing/PiglinBruteSpecificSensor.java
index c3d06c1..3a865bb 100644
--- a/minecraft/world/entity/ai/sensing/PiglinBruteSpecificSensor.java
+++ b/minecraft/world/entity/ai/sensing/PiglinBruteSpecificSensor.java
@@ -50,6 +50,6 @@ public class PiglinBruteSpecificSensor extends Sensor<LivingEntity> {
         }
 
         brain.setMemory(MemoryModuleType.NEAREST_VISIBLE_NEMESIS, optional1);
-        brain.setMemory(MemoryModuleType.NEARBY_ADULT_PIGLINS, (Object) list);
+        brain.setMemory(MemoryModuleType.NEARBY_ADULT_PIGLINS, list);
     }
 }
