diff --git a/minecraft/world/entity/ai/sensing/SecondaryPoiSensor.java b/minecraft/world/entity/ai/sensing/SecondaryPoiSensor.java
index 1e05b6e..169a440 100644
--- a/minecraft/world/entity/ai/sensing/SecondaryPoiSensor.java
+++ b/minecraft/world/entity/ai/sensing/SecondaryPoiSensor.java
@@ -42,7 +42,7 @@ public class SecondaryPoiSensor extends Sensor<Villager> {
         Brain<?> brain = villager.getBrain();
 
         if (!list.isEmpty()) {
-            brain.setMemory(MemoryModuleType.SECONDARY_JOB_SITE, (Object) list);
+            brain.setMemory(MemoryModuleType.SECONDARY_JOB_SITE, list);
         } else {
             brain.eraseMemory(MemoryModuleType.SECONDARY_JOB_SITE);
         }
