diff --git a/minecraft/util/valueproviders/ClampedInt.java b/minecraft/util/valueproviders/ClampedInt.java
index 81cbd8b..7418071 100644
--- a/minecraft/util/valueproviders/ClampedInt.java
+++ b/minecraft/util/valueproviders/ClampedInt.java
@@ -9,7 +9,7 @@ import net.minecraft.util.RandomSource;
 
 public class ClampedInt extends IntProvider {
 
-    public static final Codec<ClampedInt> CODEC = RecordCodecBuilder.create((instance) -> {
+    public static final Codec<ClampedInt> CODEC = RecordCodecBuilder.<ClampedInt>create((instance) -> {
         return instance.group(IntProvider.CODEC.fieldOf("source").forGetter((clampedint) -> {
             return clampedint.source;
         }), Codec.INT.fieldOf("min_inclusive").forGetter((clampedint) -> {
