diff --git a/minecraft/commands/arguments/item/ItemInput.java b/minecraft/commands/arguments/item/ItemInput.java
index 3dd656d..638fe4a 100644
--- a/minecraft/commands/arguments/item/ItemInput.java
+++ b/minecraft/commands/arguments/item/ItemInput.java
@@ -9,6 +9,7 @@ import net.minecraft.nbt.CompoundTag;
 import net.minecraft.nbt.NbtUtils;
 import net.minecraft.network.chat.Component;
 import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
 
@@ -60,7 +61,7 @@ public class ItemInput implements Predicate<ItemStack> {
 
     private String getItemName() {
         return this.item.unwrapKey().map(ResourceKey::location).orElseGet(() -> {
-            return "unknown[" + this.item + "]";
+            return ResourceLocation.tryParse("unknown[" + this.item + "]");
         }).toString();
     }
 }
