diff --git a/minecraft/server/commands/DebugCommand.java b/minecraft/server/commands/DebugCommand.java
index f7af574..25a0264 100644
--- a/minecraft/server/commands/DebugCommand.java
+++ b/minecraft/server/commands/DebugCommand.java
@@ -139,14 +139,15 @@ public class DebugCommand {
                     commandsourcestack.sendFailure(Component.translatable("commands.debug.function.traceFailed"));
                 }
 
+                int finalI = i;
                 executioncontrol.queueNext((executioncontext, j) -> {
                     if (collection.size() == 1) {
                         commandsourcestack.sendSuccess(() -> {
-                            return Component.translatable("commands.debug.function.success.single", i, Component.translationArg(((CommandFunction) collection.iterator().next()).id()), s);
+                            return Component.translatable("commands.debug.function.success.single", finalI, Component.translationArg(((CommandFunction) collection.iterator().next()).id()), s);
                         }, true);
                     } else {
                         commandsourcestack.sendSuccess(() -> {
-                            return Component.translatable("commands.debug.function.success.multiple", i, collection.size(), s);
+                            return Component.translatable("commands.debug.function.success.multiple", finalI, collection.size(), s);
                         }, true);
                     }
 
