diff --git a/minecraft/util/datafix/schemas/V1451_6.java b/minecraft/util/datafix/schemas/V1451_6.java
index 8dcdcfa..20775a6 100644
--- a/minecraft/util/datafix/schemas/V1451_6.java
+++ b/minecraft/util/datafix/schemas/V1451_6.java
@@ -23,7 +23,7 @@ public class V1451_6 extends NamespacedSchema {
         public <T> T apply(DynamicOps<T> dynamicops, T t0) {
             Dynamic<T> dynamic = new Dynamic(dynamicops, t0);
 
-            return ((Dynamic) DataFixUtils.orElse(dynamic.get("CriteriaName").asString().get().left().map((s) -> {
+            return (DataFixUtils.orElse(dynamic.get("CriteriaName").asString().get().left().map((s) -> {
                 int i = s.indexOf(58);
 
                 if (i < 0) {
@@ -65,7 +65,7 @@ public class V1451_6 extends NamespacedSchema {
                 }
             });
 
-            return ((Dynamic) DataFixUtils.orElse(optional.map((dynamic1) -> {
+            return (DataFixUtils.orElse(optional.map((dynamic1) -> {
                 return dynamic.set("CriteriaName", dynamic1).remove("CriteriaType");
             }), dynamic)).getValue();
         }
