diff --git a/src/main/java/net/minecraft/world/level/LevelReader.java b/src/main/java/net/minecraft/world/level/LevelReader.java
index 2c10803..68e2236 100644
--- a/src/main/java/net/minecraft/world/level/LevelReader.java
+++ b/src/main/java/net/minecraft/world/level/LevelReader.java
@@ -41,6 +41,13 @@ public interface LevelReader extends BlockAndTintGetter, CollisionGetter, Signal
         return this.getBiomeManager().getBiome(blockpos);
     }
 
+    // dish
+    public default BlockState getBlockStateIfLoaded(BlockPos blockpos) {
+        ChunkAccess chunk = this.getChunk(blockpos.getX() >> 4, blockpos.getZ() >> 4, ChunkStatus.FULL, false);
+        return chunk == null ? null : chunk.getBlockState(blockpos);
+    }
+    // dish end
+
     default Stream<BlockState> getBlockStatesIfLoaded(AABB aabb) {
         int i = Mth.floor(aabb.minX);
         int j = Mth.floor(aabb.maxX);
