diff --git a/minecraft/data/tags/BiomeTagsProvider.java b/minecraft/data/tags/BiomeTagsProvider.java
index 4028e74..656fd55 100644
--- a/minecraft/data/tags/BiomeTagsProvider.java
+++ b/minecraft/data/tags/BiomeTagsProvider.java
@@ -6,6 +6,7 @@ import java.util.stream.Stream;
 import net.minecraft.core.HolderLookup;
 import net.minecraft.core.registries.Registries;
 import net.minecraft.data.PackOutput;
+import net.minecraft.resources.ResourceKey;
 import net.minecraft.tags.BiomeTags;
 import net.minecraft.world.level.biome.Biome;
 import net.minecraft.world.level.biome.Biomes;
@@ -31,7 +32,7 @@ public class BiomeTagsProvider extends TagsProvider<Biome> {
         this.tag(BiomeTags.IS_FOREST).add(Biomes.FOREST).add(Biomes.FLOWER_FOREST).add(Biomes.BIRCH_FOREST).add(Biomes.OLD_GROWTH_BIRCH_FOREST).add(Biomes.DARK_FOREST).add(Biomes.GROVE);
         this.tag(BiomeTags.IS_SAVANNA).add(Biomes.SAVANNA).add(Biomes.SAVANNA_PLATEAU).add(Biomes.WINDSWEPT_SAVANNA);
         TagsProvider.TagAppender<Biome> tagsprovider_tagappender = this.tag(BiomeTags.IS_NETHER);
-        Stream stream = MultiNoiseBiomeSourceParameterList.Preset.NETHER.usedBiomes();
+        Stream<ResourceKey<Biome>> stream = MultiNoiseBiomeSourceParameterList.Preset.NETHER.usedBiomes();
 
         Objects.requireNonNull(tagsprovider_tagappender);
         stream.forEach(tagsprovider_tagappender::add);
