diff --git a/minecraft/world/level/block/PointedDripstoneBlock.java b/minecraft/world/level/block/PointedDripstoneBlock.java
index 163d960..bba35d1 100644
--- a/minecraft/world/level/block/PointedDripstoneBlock.java
+++ b/minecraft/world/level/block/PointedDripstoneBlock.java
@@ -443,7 +443,7 @@ public class PointedDripstoneBlock extends Block implements Fallable, SimpleWate
 
             return (BlockPos) findBlockVertical(levelaccessor, blockpos, direction.getAxisDirection(), bipredicate, (blockstate1) -> {
                 return isTip(blockstate1, flag);
-            }, i).orElse((Object) null);
+            }, i).orElse(null);
         }
     }
 
@@ -559,7 +559,7 @@ public class PointedDripstoneBlock extends Block implements Fallable, SimpleWate
             return canDripThrough(level, blockpos1, blockstate);
         };
 
-        return (BlockPos) findBlockVertical(level, blockpos, Direction.DOWN.getAxisDirection(), bipredicate, predicate, 11).orElse((Object) null);
+        return (BlockPos) findBlockVertical(level, blockpos, Direction.DOWN.getAxisDirection(), bipredicate, predicate, 11).orElse( null);
     }
 
     @Nullable
@@ -568,7 +568,7 @@ public class PointedDripstoneBlock extends Block implements Fallable, SimpleWate
             return canDripThrough(level, blockpos1, blockstate);
         };
 
-        return (BlockPos) findBlockVertical(level, blockpos, Direction.UP.getAxisDirection(), bipredicate, PointedDripstoneBlock::canDrip, 11).orElse((Object) null);
+        return (BlockPos) findBlockVertical(level, blockpos, Direction.UP.getAxisDirection(), bipredicate, PointedDripstoneBlock::canDrip, 11).orElse(null);
     }
 
     public static Fluid getCauldronFillFluidType(ServerLevel serverlevel, BlockPos blockpos) {
