diff --git a/minecraft/world/level/levelgen/structure/PoolElementStructurePiece.java b/minecraft/world/level/levelgen/structure/PoolElementStructurePiece.java
index a62f272..389a663 100644
--- a/minecraft/world/level/levelgen/structure/PoolElementStructurePiece.java
+++ b/minecraft/world/level/levelgen/structure/PoolElementStructurePiece.java
@@ -54,7 +54,7 @@ public class PoolElementStructurePiece extends StructurePiece {
         this.position = new BlockPos(compoundtag.getInt("PosX"), compoundtag.getInt("PosY"), compoundtag.getInt("PosZ"));
         this.groundLevelDelta = compoundtag.getInt("ground_level_delta");
         DynamicOps<Tag> dynamicops = RegistryOps.create(NbtOps.INSTANCE, (HolderLookup.Provider) structurepieceserializationcontext.registryAccess());
-        DataResult dataresult = StructurePoolElement.CODEC.parse(dynamicops, compoundtag.getCompound("pool_element"));
+        DataResult<StructurePoolElement> dataresult = StructurePoolElement.CODEC.parse(dynamicops, compoundtag.getCompound("pool_element"));
         Logger logger = PoolElementStructurePiece.LOGGER;
 
         Objects.requireNonNull(logger);
@@ -78,7 +78,7 @@ public class PoolElementStructurePiece extends StructurePiece {
         compoundtag.putInt("PosZ", this.position.getZ());
         compoundtag.putInt("ground_level_delta", this.groundLevelDelta);
         DynamicOps<Tag> dynamicops = RegistryOps.create(NbtOps.INSTANCE, (HolderLookup.Provider) structurepieceserializationcontext.registryAccess());
-        DataResult dataresult = StructurePoolElement.CODEC.encodeStart(dynamicops, this.element);
+        DataResult<Tag> dataresult = StructurePoolElement.CODEC.encodeStart(dynamicops, this.element);
         Logger logger = PoolElementStructurePiece.LOGGER;
 
         Objects.requireNonNull(logger);
