diff --git a/minecraft/world/level/levelgen/structure/StructureType.java b/minecraft/world/level/levelgen/structure/StructureType.java
index 9f629e2..1e53295 100644
--- a/minecraft/world/level/levelgen/structure/StructureType.java
+++ b/minecraft/world/level/levelgen/structure/StructureType.java
@@ -42,7 +42,7 @@ public interface StructureType<S extends Structure> {
     Codec<S> codec();
 
     private static <S extends Structure> StructureType<S> register(String s, Codec<S> codec) {
-        return (StructureType) Registry.register(BuiltInRegistries.STRUCTURE_TYPE, s, () -> {
+        return Registry.register(BuiltInRegistries.STRUCTURE_TYPE, s, () -> {
             return codec;
         });
     }
