diff --git a/minecraft/world/level/levelgen/blending/Blender.java b/minecraft/world/level/levelgen/blending/Blender.java
index bb4a8e4..f3e79bc 100644
--- a/minecraft/world/level/levelgen/blending/Blender.java
+++ b/minecraft/world/level/levelgen/blending/Blender.java
@@ -358,7 +358,7 @@ public class Blender {
 
                 return blender_distancegetter.getDistance(d0, d1, d2) < 4.0D;
             };
-            Stream stream = Stream.of(GenerationStep.Carving.values());
+            Stream<GenerationStep.Carving> stream = Stream.of(GenerationStep.Carving.values());
 
             Objects.requireNonNull(protochunk);
             stream.map(protochunk::getOrCreateCarvingMask).forEach((carvingmask) -> {
@@ -409,8 +409,10 @@ public class Blender {
         double d2 = (double) blendingdata.getAreaWithOldGeneration().getHeight() / 2.0D;
         double d3 = (double) blendingdata.getAreaWithOldGeneration().getMinBuildHeight() + d2;
 
+        double finalD = d0;
+        double finalD1 = d1;
         return (d4, d5, d6) -> {
-            return distanceToCube(d4 - 8.0D - d0, d5 - d3, d6 - 8.0D - d1, 8.0D, d2, 8.0D);
+            return distanceToCube(d4 - 8.0D - finalD, d5 - d3, d6 - 8.0D - finalD1, 8.0D, d2, 8.0D);
         };
     }
 
