diff --git a/minecraft/util/profiling/metrics/storage/MetricsPersister.java b/minecraft/util/profiling/metrics/storage/MetricsPersister.java
index 7fccb03..20bd870 100644
--- a/minecraft/util/profiling/metrics/storage/MetricsPersister.java
+++ b/minecraft/util/profiling/metrics/storage/MetricsPersister.java
@@ -103,8 +103,9 @@ public class MetricsPersister {
             int j = list1.stream().mapToInt(MetricSampler.SamplerResult::getLastTick).summaryStatistics().getMax();
 
             for (int k = i; k <= j; ++k) {
+                int finalK = k;
                 Stream<String> stream = list1.stream().map((metricsampler_samplerresult) -> {
-                    return String.valueOf(metricsampler_samplerresult.valueAtTick(k));
+                    return String.valueOf(metricsampler_samplerresult.valueAtTick(finalK));
                 });
                 Object[] aobject = Stream.concat(Stream.of(String.valueOf(k)), stream).toArray((l) -> {
                     return new String[l];
