diff --git a/minecraft/world/level/chunk/storage/EntityStorage.java b/minecraft/world/level/chunk/storage/EntityStorage.java
index c0bd234..a1e8246 100644
--- a/minecraft/world/level/chunk/storage/EntityStorage.java
+++ b/minecraft/world/level/chunk/storage/EntityStorage.java
@@ -9,6 +9,7 @@ import java.io.IOException;
 import java.nio.file.Path;
 import java.util.List;
 import java.util.Objects;
+import java.util.Optional;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.Executor;
 import java.util.function.Function;
@@ -50,7 +51,7 @@ public class EntityStorage implements EntityPersistentStorage<Entity> {
             return CompletableFuture.completedFuture(emptyChunk(chunkpos));
         } else {
             CompletableFuture completablefuture = this.worker.loadAsync(chunkpos);
-            Function function = (optional) -> {
+            Function<Optional, ChunkEntities> function = (optional) -> {
                 if (optional.isEmpty()) {
                     this.emptyChunks.add(chunkpos.toLong());
                     return emptyChunk(chunkpos);
