diff --git a/minecraft/tags/TagKey.java b/minecraft/tags/TagKey.java
index 7d5271c..a3a6c04 100644
--- a/minecraft/tags/TagKey.java
+++ b/minecraft/tags/TagKey.java
@@ -40,7 +40,7 @@ public record TagKey<T> (ResourceKey<? extends Registry<T>> registry, ResourceLo
     }
 
     public <E> Optional<TagKey<E>> cast(ResourceKey<? extends Registry<E>> resourcekey) {
-        return this.isFor(resourcekey) ? Optional.of(this) : Optional.empty();
+        return this.isFor(resourcekey) ? Optional.of((TagKey<E>) this) : Optional.empty();
     }
 
     public String toString() {
