diff --git a/minecraft/core/MappedRegistry.java b/minecraft/core/MappedRegistry.java
index c2ac522..009fa14 100644
--- a/minecraft/core/MappedRegistry.java
+++ b/minecraft/core/MappedRegistry.java
@@ -207,13 +207,13 @@ public class MappedRegistry<T> implements WritableRegistry<T> {
     @Nullable
     @Override
     public T get(@Nullable ResourceKey<T> resourcekey) {
-        return getValueFromNullable((Holder.Reference) this.byKey.get(resourcekey));
+        return getValueFromNullable(this.byKey.get(resourcekey));
     }
 
     @Nullable
     @Override
     public T byId(int i) {
-        return i >= 0 && i < this.byId.size() ? getValueFromNullable((Holder.Reference) this.byId.get(i)) : null;
+        return i >= 0 && i < this.byId.size() ? getValueFromNullable(this.byId.get(i)) : null;
     }
 
     @Override
