diff --git a/minecraft/util/valueproviders/BiasedToBottomInt.java b/minecraft/util/valueproviders/BiasedToBottomInt.java
index fd7af1e..5702ede 100644
--- a/minecraft/util/valueproviders/BiasedToBottomInt.java
+++ b/minecraft/util/valueproviders/BiasedToBottomInt.java
@@ -8,7 +8,7 @@ import net.minecraft.util.RandomSource;
 
 public class BiasedToBottomInt extends IntProvider {
 
-    public static final Codec<BiasedToBottomInt> CODEC = RecordCodecBuilder.create((instance) -> {
+    public static final Codec<BiasedToBottomInt> CODEC = RecordCodecBuilder.<BiasedToBottomInt>create((instance) -> {
         return instance.group(Codec.INT.fieldOf("min_inclusive").forGetter((biasedtobottomint) -> {
             return biasedtobottomint.minInclusive;
         }), Codec.INT.fieldOf("max_inclusive").forGetter((biasedtobottomint) -> {
