diff --git a/minecraft/util/valueproviders/TrapezoidFloat.java b/minecraft/util/valueproviders/TrapezoidFloat.java
index a021eb9..17ffe1c 100644
--- a/minecraft/util/valueproviders/TrapezoidFloat.java
+++ b/minecraft/util/valueproviders/TrapezoidFloat.java
@@ -8,7 +8,7 @@ import net.minecraft.util.RandomSource;
 
 public class TrapezoidFloat extends FloatProvider {
 
-    public static final Codec<TrapezoidFloat> CODEC = RecordCodecBuilder.create((instance) -> {
+    public static final Codec<TrapezoidFloat> CODEC = RecordCodecBuilder.<TrapezoidFloat>create((instance) -> {
         return instance.group(Codec.FLOAT.fieldOf("min").forGetter((trapezoidfloat) -> {
             return trapezoidfloat.min;
         }), Codec.FLOAT.fieldOf("max").forGetter((trapezoidfloat) -> {
