diff --git a/minecraft/util/valueproviders/UniformFloat.java b/minecraft/util/valueproviders/UniformFloat.java
index 62b8047..07eba45 100644
--- a/minecraft/util/valueproviders/UniformFloat.java
+++ b/minecraft/util/valueproviders/UniformFloat.java
@@ -9,7 +9,7 @@ import net.minecraft.util.RandomSource;
 
 public class UniformFloat extends FloatProvider {
 
-    public static final Codec<UniformFloat> CODEC = RecordCodecBuilder.create((instance) -> {
+    public static final Codec<UniformFloat> CODEC = RecordCodecBuilder.<UniformFloat>create((instance) -> {
         return instance.group(Codec.FLOAT.fieldOf("min_inclusive").forGetter((uniformfloat) -> {
             return uniformfloat.minInclusive;
         }), Codec.FLOAT.fieldOf("max_exclusive").forGetter((uniformfloat) -> {
