diff --git a/minecraft/world/entity/ai/behavior/LongJumpToRandomPos.java b/minecraft/world/entity/ai/behavior/LongJumpToRandomPos.java
index 6c0fdd3..c089e8e 100644
--- a/minecraft/world/entity/ai/behavior/LongJumpToRandomPos.java
+++ b/minecraft/world/entity/ai/behavior/LongJumpToRandomPos.java
@@ -78,7 +78,7 @@ public class LongJumpToRandomPos<E extends Mob> extends Behavior<E> {
         boolean flag = mob.onGround() && !mob.isInWater() && !mob.isInLava() && !serverlevel.getBlockState(mob.blockPosition()).is(Blocks.HONEY_BLOCK);
 
         if (!flag) {
-            mob.getBrain().setMemory(MemoryModuleType.LONG_JUMP_COOLDOWN_TICKS, (Object) (this.timeBetweenLongJumps.sample(serverlevel.random) / 2));
+            mob.getBrain().setMemory(MemoryModuleType.LONG_JUMP_COOLDOWN_TICKS, (this.timeBetweenLongJumps.sample(serverlevel.random) / 2));
         }
 
         return flag;
@@ -88,7 +88,7 @@ public class LongJumpToRandomPos<E extends Mob> extends Behavior<E> {
         boolean flag = this.initialPosition.isPresent() && ((Vec3) this.initialPosition.get()).equals(mob.position()) && this.findJumpTries > 0 && !mob.isInWaterOrBubble() && (this.chosenJump != null || !this.jumpCandidates.isEmpty());
 
         if (!flag && mob.getBrain().getMemory(MemoryModuleType.LONG_JUMP_MID_JUMP).isEmpty()) {
-            mob.getBrain().setMemory(MemoryModuleType.LONG_JUMP_COOLDOWN_TICKS, (Object) (this.timeBetweenLongJumps.sample(serverlevel.random) / 2));
+            mob.getBrain().setMemory(MemoryModuleType.LONG_JUMP_COOLDOWN_TICKS, (this.timeBetweenLongJumps.sample(serverlevel.random) / 2));
             mob.getBrain().eraseMemory(MemoryModuleType.LOOK_TARGET);
         }
 
@@ -120,7 +120,7 @@ public class LongJumpToRandomPos<E extends Mob> extends Behavior<E> {
                 double d1 = d0 + (double) e0.getJumpBoostPower();
 
                 e0.setDeltaMovement(this.chosenJump.scale(d1 / d0));
-                e0.getBrain().setMemory(MemoryModuleType.LONG_JUMP_MID_JUMP, (Object) true);
+                e0.getBrain().setMemory(MemoryModuleType.LONG_JUMP_MID_JUMP, true);
                 serverlevel.playSound((Player) null, (Entity) e0, (SoundEvent) this.getJumpSound.apply(e0), SoundSource.NEUTRAL, 1.0F, 1.0F);
             }
         } else {
@@ -153,7 +153,7 @@ public class LongJumpToRandomPos<E extends Mob> extends Behavior<E> {
                     continue;
                 }
 
-                e0.getBrain().setMemory(MemoryModuleType.LOOK_TARGET, (Object) (new BlockPosTracker(blockpos)));
+                e0.getBrain().setMemory(MemoryModuleType.LOOK_TARGET, (new BlockPosTracker(blockpos)));
                 PathNavigation pathnavigation = e0.getNavigation();
                 Path path = pathnavigation.createPath(blockpos, 0, 8);
 
