diff --git a/minecraft/server/commands/DebugCommand.java b/minecraft/server/commands/DebugCommand.java
index 3def5ca..f058ce0 100644
--- a/minecraft/server/commands/DebugCommand.java
+++ b/minecraft/server/commands/DebugCommand.java
@@ -130,14 +130,14 @@ public class DebugCommand {
             DebugCommand.LOGGER.warn("Tracing failed", uncheckedioexception);
             commandsourcestack.sendFailure(Component.translatable("commands.debug.function.traceFailed"));
         }
-
+        int finalI = i;
         if (collection.size() == 1) {
             commandsourcestack.sendSuccess(() -> {
-                return Component.translatable("commands.debug.function.success.single", i, Component.translationArg(((CommandFunction) collection.iterator().next()).getId()), s);
+                return Component.translatable("commands.debug.function.success.single", finalI, Component.translationArg(((CommandFunction) collection.iterator().next()).getId()), s);
             }, true);
         } else {
             commandsourcestack.sendSuccess(() -> {
-                return Component.translatable("commands.debug.function.success.multiple", i, collection.size(), s);
+                return Component.translatable("commands.debug.function.success.multiple", finalI, collection.size(), s);
             }, true);
         }
 
