diff --git a/minecraft/world/entity/monster/Zoglin.java b/minecraft/world/entity/monster/Zoglin.java
index c2b3b4c..62476c0 100644
--- a/minecraft/world/entity/monster/Zoglin.java
+++ b/minecraft/world/entity/monster/Zoglin.java
@@ -189,14 +189,14 @@ public class Zoglin extends Monster implements Enemy, HoglinBase {
 
     @Override
     public Brain<Zoglin> getBrain() {
-        return super.getBrain();
+        return (Brain<Zoglin>) super.getBrain();
     }
 
     protected void updateActivity() {
-        Activity activity = (Activity) this.brain.getActiveNonCoreActivity().orElse((Object) null);
+        Activity activity = (Activity) this.brain.getActiveNonCoreActivity().orElse(null);
 
         this.brain.setActiveActivityToFirstValid(ImmutableList.of(Activity.FIGHT, Activity.IDLE));
-        Activity activity1 = (Activity) this.brain.getActiveNonCoreActivity().orElse((Object) null);
+        Activity activity1 = (Activity) this.brain.getActiveNonCoreActivity().orElse( null);
 
         if (activity1 == Activity.FIGHT && activity != Activity.FIGHT) {
             this.playAngrySound();
