diff --git a/minecraft/commands/arguments/selector/EntitySelectorParser.java b/minecraft/commands/arguments/selector/EntitySelectorParser.java
index dd47d24..94e8291 100644
--- a/minecraft/commands/arguments/selector/EntitySelectorParser.java
+++ b/minecraft/commands/arguments/selector/EntitySelectorParser.java
@@ -147,7 +147,7 @@ public class EntitySelectorParser {
             aabb = this.createAabb(this.deltaX == null ? 0.0D : this.deltaX, this.deltaY == null ? 0.0D : this.deltaY, this.deltaZ == null ? 0.0D : this.deltaZ);
         }
 
-        Function function;
+        Function<Vec3, Vec3> function;
 
         if (this.x == null && this.y == null && this.z == null) {
             function = (vec3) -> {
