diff --git a/minecraft/world/level/entity/EntityTypeTest.java b/minecraft/world/level/entity/EntityTypeTest.java
index c937fae..b19b443 100644
--- a/minecraft/world/level/entity/EntityTypeTest.java
+++ b/minecraft/world/level/entity/EntityTypeTest.java
@@ -9,7 +9,7 @@ public interface EntityTypeTest<B, T extends B> {
             @Nullable
             @Override
             public T tryCast(B b0) {
-                return oclass.isInstance(b0) ? b0 : null;
+                return oclass.isInstance(b0) ? (T) b0 : null;
             }
 
             @Override
@@ -24,7 +24,7 @@ public interface EntityTypeTest<B, T extends B> {
             @Nullable
             @Override
             public T tryCast(B b0) {
-                return oclass.equals(b0.getClass()) ? b0 : null;
+                return oclass.equals(b0.getClass()) ? (T) b0 : null;
             }
 
             @Override
