diff --git a/minecraft/world/level/entity/EntitySection.java b/minecraft/world/level/entity/EntitySection.java
index 3c93b00..f564cf8 100644
--- a/minecraft/world/level/entity/EntitySection.java
+++ b/minecraft/world/level/entity/EntitySection.java
@@ -32,14 +32,14 @@ public class EntitySection<T extends EntityAccess> {
     public AbortableIterationConsumer.Continuation getEntities(AABB aabb, AbortableIterationConsumer<T> abortableiterationconsumer) {
         Iterator iterator = this.storage.iterator();
 
-        EntityAccess entityaccess;
+        T entityaccess;
 
         do {
             if (!iterator.hasNext()) {
                 return AbortableIterationConsumer.Continuation.CONTINUE;
             }
 
-            entityaccess = (EntityAccess) iterator.next();
+            entityaccess = (T) iterator.next();
         } while (!entityaccess.getBoundingBox().intersects(aabb) || !abortableiterationconsumer.accept(entityaccess).shouldAbort());
 
         return AbortableIterationConsumer.Continuation.ABORT;
@@ -53,16 +53,16 @@ public class EntitySection<T extends EntityAccess> {
         } else {
             Iterator iterator = collection.iterator();
 
-            EntityAccess entityaccess;
-            EntityAccess entityaccess1;
+            T entityaccess;
+            U entityaccess1;
 
             do {
                 if (!iterator.hasNext()) {
                     return AbortableIterationConsumer.Continuation.CONTINUE;
                 }
 
-                entityaccess = (EntityAccess) iterator.next();
-                entityaccess1 = (EntityAccess) entitytypetest.tryCast(entityaccess);
+                entityaccess = (T) iterator.next();
+                entityaccess1 = entitytypetest.tryCast(entityaccess);
             } while (entityaccess1 == null || !entityaccess.getBoundingBox().intersects(aabb) || !abortableiterationconsumer.accept(entityaccess1).shouldAbort());
 
             return AbortableIterationConsumer.Continuation.ABORT;
