diff --git a/minecraft/world/item/crafting/RecipeSerializer.java b/minecraft/world/item/crafting/RecipeSerializer.java
index 47b9205..c8d8ebd 100644
--- a/minecraft/world/item/crafting/RecipeSerializer.java
+++ b/minecraft/world/item/crafting/RecipeSerializer.java
@@ -38,6 +38,6 @@ public interface RecipeSerializer<T extends Recipe<?>> {
     void toNetwork(FriendlyByteBuf friendlybytebuf, T t0);
 
     static <S extends RecipeSerializer<T>, T extends Recipe<?>> S register(String s, S s0) {
-        return (RecipeSerializer) Registry.register(BuiltInRegistries.RECIPE_SERIALIZER, s, s0);
+        return Registry.register(BuiltInRegistries.RECIPE_SERIALIZER, s, s0);
     }
 }
