diff --git a/minecraft/world/item/crafting/RecipeCache.java b/minecraft/world/item/crafting/RecipeCache.java
index 4d8a11c..e5374e1 100644
--- a/minecraft/world/item/crafting/RecipeCache.java
+++ b/minecraft/world/item/crafting/RecipeCache.java
@@ -51,7 +51,7 @@ public class RecipeCache {
     private Optional<CraftingRecipe> compute(CraftingContainer craftingcontainer, Level level) {
         Optional<RecipeHolder<CraftingRecipe>> optional = level.getRecipeManager().getRecipeFor(RecipeType.CRAFTING, craftingcontainer, level);
 
-        this.insert(craftingcontainer.getItems(), (CraftingRecipe) optional.map(RecipeHolder::value).orElse((Object) null));
+        this.insert(craftingcontainer.getItems(), (CraftingRecipe) optional.map(RecipeHolder::value).orElse(null));
         return optional.map(RecipeHolder::value);
     }
 
