diff --git a/minecraft/world/entity/Entity.java b/minecraft/world/entity/Entity.java
index 6f5a210..3ebfa22 100644
--- a/minecraft/world/entity/Entity.java
+++ b/minecraft/world/entity/Entity.java
@@ -2664,7 +2664,7 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource {
                     }
 
                     return PortalShape.createPortalInfo(serverlevel, blockutil_foundrectangle, direction_axis, vec3, this, this.getDeltaMovement(), this.getYRot(), this.getXRot());
-                }).orElse((Object) null);
+                }).orElse(null);
             }
         } else {
             BlockPos blockpos1;
