diff --git a/minecraft/world/entity/LivingEntity.java b/minecraft/world/entity/LivingEntity.java
index 60353da..a35e32f 100644
--- a/minecraft/world/entity/LivingEntity.java
+++ b/minecraft/world/entity/LivingEntity.java
@@ -3439,7 +3439,7 @@ public abstract class LivingEntity extends Entity implements Attackable {
     }
 
     public void stopSleeping() {
-        Optional optional = this.getSleepingPos();
+        Optional<BlockPos> optional = this.getSleepingPos();
         Level level = this.level();
 
         java.util.Objects.requireNonNull(level);
@@ -3473,7 +3473,7 @@ public abstract class LivingEntity extends Entity implements Attackable {
 
     @Nullable
     public Direction getBedOrientation() {
-        BlockPos blockpos = (BlockPos) this.getSleepingPos().orElse((Object) null);
+        BlockPos blockpos = (BlockPos) this.getSleepingPos().orElse(null);
 
         return blockpos != null ? BedBlock.getBedOrientation(this.level(), blockpos) : null;
     }
