diff --git a/minecraft/commands/arguments/StringRepresentableArgument.java b/minecraft/commands/arguments/StringRepresentableArgument.java
index be7ea0a..ec09747 100644
--- a/minecraft/commands/arguments/StringRepresentableArgument.java
+++ b/minecraft/commands/arguments/StringRepresentableArgument.java
@@ -35,7 +35,7 @@ public class StringRepresentableArgument<T extends Enum<T> & StringRepresentable
     public T parse(StringReader stringreader) throws CommandSyntaxException {
         String s = stringreader.readUnquotedString();
 
-        return (Enum) this.codec.parse(JsonOps.INSTANCE, new JsonPrimitive(s)).result().orElseThrow(() -> {
+        return this.codec.parse(JsonOps.INSTANCE, new JsonPrimitive(s)).result().orElseThrow(() -> {
             return StringRepresentableArgument.ERROR_INVALID_VALUE.create(s);
         });
     }
