diff --git a/minecraft/world/ticks/LevelTicks.java b/minecraft/world/ticks/LevelTicks.java
index dbc9fb2..bdaea5a 100644
--- a/minecraft/world/ticks/LevelTicks.java
+++ b/minecraft/world/ticks/LevelTicks.java
@@ -305,7 +305,7 @@ public class LevelTicks<T> implements LevelTickAccess<T> {
         Predicate<ScheduledTick<T>> predicate = (scheduledtick) -> {
             return boundingbox.isInside(scheduledtick.pos());
         };
-        Stream stream = levelticks.alreadyRunThisTick.stream().filter(predicate);
+        Stream<ScheduledTick<T>> stream = levelticks.alreadyRunThisTick.stream().filter(predicate);
 
         Objects.requireNonNull(list);
         stream.forEach(list::add);
@@ -313,7 +313,7 @@ public class LevelTicks<T> implements LevelTickAccess<T> {
         Objects.requireNonNull(list);
         stream.forEach(list::add);
         levelticks.forContainersInArea(boundingbox, (i, levelchunkticks) -> {
-            Stream stream1 = levelchunkticks.getAll().filter(predicate);
+            Stream<ScheduledTick<T>> stream1 = levelchunkticks.getAll().filter(predicate);
 
             Objects.requireNonNull(list);
             stream1.forEach(list::add);
