diff --git a/minecraft/world/level/chunk/LevelChunkSection.java b/minecraft/world/level/chunk/LevelChunkSection.java
index 103177b..2f4d97f 100644
--- a/minecraft/world/level/chunk/LevelChunkSection.java
+++ b/minecraft/world/level/chunk/LevelChunkSection.java
@@ -110,13 +110,13 @@ public class LevelChunkSection {
     }
 
     public void recalcBlockCounts() {
-        class 1BlockCounter implements PalettedContainer.CountConsumer<BlockState> {
+        class BlockCounter implements PalettedContainer.CountConsumer<BlockState> {
 
             public int nonEmptyBlockCount;
             public int tickingBlockCount;
             public int tickingFluidCount;
 
-            _BlockCounter/* $FF was: 1BlockCounter*/() {
+            BlockCounter/* $FF was: 1BlockCounter*/() {
 }
 
             public void accept(BlockState blockstate, int i) {
@@ -139,12 +139,12 @@ public class LevelChunkSection {
             }
         }
 
-        1BlockCounter 1blockcounter = new 1BlockCounter();
+        BlockCounter blockcounter = new BlockCounter();
 
-        this.states.count(1blockcounter);
-        this.nonEmptyBlockCount = (short)1blockcounter.nonEmptyBlockCount;
-        this.tickingBlockCount = (short)1blockcounter.tickingBlockCount;
-        this.tickingFluidCount = (short)1blockcounter.tickingFluidCount;
+        this.states.count(blockcounter);
+        this.nonEmptyBlockCount = (short)blockcounter.nonEmptyBlockCount;
+        this.tickingBlockCount = (short)blockcounter.tickingBlockCount;
+        this.tickingFluidCount = (short)blockcounter.tickingFluidCount;
     }
 
     public PalettedContainer<BlockState> getStates() {
