diff --git a/minecraft/world/level/chunk/LinearPalette.java b/minecraft/world/level/chunk/LinearPalette.java
index 67386ba..386ae57 100644
--- a/minecraft/world/level/chunk/LinearPalette.java
+++ b/minecraft/world/level/chunk/LinearPalette.java
@@ -17,7 +17,7 @@ public class LinearPalette<T> implements Palette<T> {
 
     private LinearPalette(IdMap<T> idmap, int i, PaletteResize<T> paletteresize, List<T> list) {
         this.registry = idmap;
-        this.values = new Object[1 << i];
+        this.values = (T[]) new Object[1 << i];
         this.bits = i;
         this.resizeHandler = paletteresize;
         Validate.isTrue(list.size() <= this.values.length, "Can't initialize LinearPalette of size %d with %d entries", new Object[]{this.values.length, list.size()});
@@ -119,6 +119,6 @@ public class LinearPalette<T> implements Palette<T> {
 
     @Override
     public Palette<T> copy() {
-        return new LinearPalette<>(this.registry, (Object[]) this.values.clone(), this.resizeHandler, this.bits, this.size);
+        return new LinearPalette<>(this.registry, this.values.clone(), this.resizeHandler, this.bits, this.size);
     }
 }
