diff --git a/minecraft/util/datafix/fixes/EntityCodSalmonFix.java b/minecraft/util/datafix/fixes/EntityCodSalmonFix.java
index db73932..5f3a69a 100644
--- a/minecraft/util/datafix/fixes/EntityCodSalmonFix.java
+++ b/minecraft/util/datafix/fixes/EntityCodSalmonFix.java
@@ -6,8 +6,8 @@ import java.util.Map;
 
 public class EntityCodSalmonFix extends SimplestEntityRenameFix {
 
-    public static final Map<String, String> RENAMED_IDS = ImmutableMap.builder().put("minecraft:salmon_mob", "minecraft:salmon").put("minecraft:cod_mob", "minecraft:cod").build();
-    public static final Map<String, String> RENAMED_EGG_IDS = ImmutableMap.builder().put("minecraft:salmon_mob_spawn_egg", "minecraft:salmon_spawn_egg").put("minecraft:cod_mob_spawn_egg", "minecraft:cod_spawn_egg").build();
+    public static final Map<String, String> RENAMED_IDS = ImmutableMap.<String, String>builder().put("minecraft:salmon_mob", "minecraft:salmon").put("minecraft:cod_mob", "minecraft:cod").build();
+    public static final Map<String, String> RENAMED_EGG_IDS = ImmutableMap.<String, String>builder().put("minecraft:salmon_mob_spawn_egg", "minecraft:salmon_spawn_egg").put("minecraft:cod_mob_spawn_egg", "minecraft:cod_spawn_egg").build();
 
     public EntityCodSalmonFix(Schema schema, boolean flag) {
         super("EntityCodSalmonFix", schema, flag);
