diff --git a/minecraft/util/datafix/fixes/EntityPufferfishRenameFix.java b/minecraft/util/datafix/fixes/EntityPufferfishRenameFix.java
index eb74a2f..ced2713 100644
--- a/minecraft/util/datafix/fixes/EntityPufferfishRenameFix.java
+++ b/minecraft/util/datafix/fixes/EntityPufferfishRenameFix.java
@@ -7,7 +7,7 @@ import java.util.Objects;
 
 public class EntityPufferfishRenameFix extends SimplestEntityRenameFix {
 
-    public static final Map<String, String> RENAMED_IDS = ImmutableMap.builder().put("minecraft:puffer_fish_spawn_egg", "minecraft:pufferfish_spawn_egg").build();
+    public static final Map<String, String> RENAMED_IDS = ImmutableMap.<String, String>builder().put("minecraft:puffer_fish_spawn_egg", "minecraft:pufferfish_spawn_egg").build();
 
     public EntityPufferfishRenameFix(Schema schema, boolean flag) {
         super("EntityPufferfishRenameFix", schema, flag);
