diff --git a/minecraft/util/datafix/fixes/FixProjectileStoredItem.java b/minecraft/util/datafix/fixes/FixProjectileStoredItem.java
index 5f4e701..4202477 100644
--- a/minecraft/util/datafix/fixes/FixProjectileStoredItem.java
+++ b/minecraft/util/datafix/fixes/FixProjectileStoredItem.java
@@ -58,13 +58,13 @@ public class FixProjectileStoredItem extends DataFix {
 
         return (typed) -> {
             return typed.updateTyped(opticfinder, type1, (typed1) -> {
-                return fixprojectilestoreditem_subfixer.fix(typed1, type1);
+                return fixprojectilestoreditem_subfixer.fix(typed1, (Type) type1);
             });
         };
     }
 
     private static <T> Typed<T> fixArrow(Typed<?> typed, Type<T> type) {
-        DataResult dataresult = typed.write().map((dynamic) -> {
+        DataResult<Dynamic> dataresult = typed.write().map((dynamic) -> {
             return dynamic.set("item", createItemStack(dynamic, getArrowType(dynamic)));
         });
 
@@ -79,7 +79,7 @@ public class FixProjectileStoredItem extends DataFix {
     }
 
     private static <T> Typed<T> fixSpectralArrow(Typed<?> typed, Type<T> type) {
-        DataResult dataresult = typed.write().map((dynamic) -> {
+        DataResult<Dynamic> dataresult = typed.write().map((dynamic) -> {
             return dynamic.set("item", createItemStack(dynamic, "minecraft:spectral_arrow"));
         });
 
