diff --git a/minecraft/core/Direction.java b/minecraft/core/Direction.java
index f5fb3ca..05e8fc2 100644
--- a/minecraft/core/Direction.java
+++ b/minecraft/core/Direction.java
@@ -93,7 +93,7 @@ public enum Direction implements StringRepresentable {
     }
 
     public static Collection<Direction> allShuffled(RandomSource randomsource) {
-        return Util.shuffledCopy((Object[]) values(), randomsource);
+        return Util.shuffledCopy(values(), randomsource);
     }
 
     public static Stream<Direction> stream() {
@@ -671,7 +671,7 @@ public enum Direction implements StringRepresentable {
         }
 
         public List<Direction> shuffledCopy(RandomSource randomsource) {
-            return Util.shuffledCopy((Object[]) this.faces, randomsource);
+            return Util.shuffledCopy(this.faces, randomsource);
         }
     }
 }
