diff --git a/minecraft/data/tags/FluidTagsProvider.java b/minecraft/data/tags/FluidTagsProvider.java
index 58b11fa..d064125 100644
--- a/minecraft/data/tags/FluidTagsProvider.java
+++ b/minecraft/data/tags/FluidTagsProvider.java
@@ -18,7 +18,7 @@ public class FluidTagsProvider extends IntrinsicHolderTagsProvider<Fluid> {
 
     @Override
     protected void addTags(HolderLookup.Provider holderlookup_provider) {
-        this.tag(FluidTags.WATER).add((Object[])(Fluids.WATER, Fluids.FLOWING_WATER));
-        this.tag(FluidTags.LAVA).add((Object[])(Fluids.LAVA, Fluids.FLOWING_LAVA));
+        this.tag(FluidTags.WATER).add(Fluids.WATER, Fluids.FLOWING_WATER);
+        this.tag(FluidTags.LAVA).add(Fluids.LAVA, Fluids.FLOWING_LAVA);
     }
 }
