diff --git a/minecraft/advancements/critereon/ItemPredicate.java b/minecraft/advancements/critereon/ItemPredicate.java
index fb5cc1f..4c5538c 100644
--- a/minecraft/advancements/critereon/ItemPredicate.java
+++ b/minecraft/advancements/critereon/ItemPredicate.java
@@ -103,7 +103,7 @@ public record ItemPredicate(Optional<TagKey<Item>> tag, Optional<HolderSet<Item>
         public ItemPredicate.Builder of(ItemLike... aitemlike) {
             this.items = Optional.of(HolderSet.direct((itemlike) -> {
                 return itemlike.asItem().builtInRegistryHolder();
-            }, (Object[]) aitemlike));
+            }, aitemlike));
             return this;
         }
 
