diff --git a/minecraft/world/level/block/MultifaceBlock.java b/minecraft/world/level/block/MultifaceBlock.java
index 58dea77..452f5e8 100644
--- a/minecraft/world/level/block/MultifaceBlock.java
+++ b/minecraft/world/level/block/MultifaceBlock.java
@@ -180,7 +180,7 @@ public abstract class MultifaceBlock extends Block {
 
         return (BlockState) Arrays.stream(blockplacecontext.getNearestLookingDirections()).map((direction) -> {
             return this.getStateForPlacement(blockstate, level, blockpos, direction);
-        }).filter(Objects::nonNull).findFirst().orElse((Object) null);
+        }).filter(Objects::nonNull).findFirst().orElse(null);
     }
 
     public boolean isValidStateForPlacement(BlockGetter blockgetter, BlockState blockstate, BlockPos blockpos, Direction direction) {
