diff --git a/minecraft/world/level/block/TorchBlock.java b/minecraft/world/level/block/TorchBlock.java
index 7b641db..bebd49e 100644
--- a/minecraft/world/level/block/TorchBlock.java
+++ b/minecraft/world/level/block/TorchBlock.java
@@ -15,7 +15,7 @@ import net.minecraft.world.level.block.state.BlockState;
 public class TorchBlock extends BaseTorchBlock {
 
     protected static final MapCodec<SimpleParticleType> PARTICLE_OPTIONS_FIELD = BuiltInRegistries.PARTICLE_TYPE.byNameCodec().comapFlatMap((particletype) -> {
-        DataResult dataresult;
+        DataResult<SimpleParticleType> dataresult;
 
         if (particletype instanceof SimpleParticleType) {
             SimpleParticleType simpleparticletype = (SimpleParticleType) particletype;
