diff --git a/src/main/java/net/ouja/dish/entity/passive/DishTurtle.java b/src/main/java/net/ouja/dish/entity/passive/DishTurtle.java
new file mode 100644
index 0000000..e2f33e4
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/passive/DishTurtle.java
@@ -0,0 +1,24 @@
+package net.ouja.dish.entity.passive;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.animal.Turtle;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishTurtle extends DishEntity implements net.ouja.api.entity.passive.Turtle {
+    private Turtle turtle;
+
+    public DishTurtle(Entity entity) {
+        super(entity);
+        this.turtle = (Turtle)entity;
+    }
+
+    @Override
+    public boolean hasEgg() {
+        return this.turtle.hasEgg();
+    }
+
+    @Override
+    public boolean isLayingEgg() {
+        return this.turtle.isLayingEgg();
+    }
+}
