diff --git a/src/main/java/net/ouja/dish/entity/passive/DishDonkey.java b/src/main/java/net/ouja/dish/entity/passive/DishDonkey.java
new file mode 100644
index 0000000..3cc2465
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/passive/DishDonkey.java
@@ -0,0 +1,14 @@
+package net.ouja.dish.entity.passive;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.animal.horse.Donkey;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishDonkey extends DishEntity implements net.ouja.api.entity.passive.Donkey {
+    private Donkey donkey;
+
+    public DishDonkey(Entity entity) {
+        super(entity);
+        this.donkey = (Donkey)entity;
+    }
+}
