diff --git a/src/main/java/net/ouja/dish/entity/passive/DishMule.java b/src/main/java/net/ouja/dish/entity/passive/DishMule.java
new file mode 100644
index 0000000..810ef81
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/passive/DishMule.java
@@ -0,0 +1,14 @@
+package net.ouja.dish.entity.passive;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.animal.horse.Mule;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishMule extends DishEntity implements net.ouja.api.entity.passive.Mule {
+    private Mule mule;
+
+    public DishMule(Entity entity) {
+        super(entity);
+        this.mule = (Mule)entity;
+    }
+}
