diff --git a/src/main/java/net/ouja/dish/entity/passive/DishPufferfish.java b/src/main/java/net/ouja/dish/entity/passive/DishPufferfish.java
new file mode 100644
index 0000000..b0d590a
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/passive/DishPufferfish.java
@@ -0,0 +1,19 @@
+package net.ouja.dish.entity.passive;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.animal.Pufferfish;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishPufferfish extends DishEntity implements net.ouja.api.entity.passive.Pufferfish {
+    private Pufferfish pufferfish;
+
+    public DishPufferfish(Entity entity) {
+        super(entity);
+        this.pufferfish = (Pufferfish)entity;
+    }
+
+    @Override
+    public int getPuffState() {
+        return this.pufferfish.getPuffState();
+    }
+}
