diff --git a/src/main/java/net/ouja/dish/entity/passive/DishFox.java b/src/main/java/net/ouja/dish/entity/passive/DishFox.java
new file mode 100644
index 0000000..7fbdf7c
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/passive/DishFox.java
@@ -0,0 +1,19 @@
+package net.ouja.dish.entity.passive;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.animal.Fox;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishFox extends DishEntity implements net.ouja.api.entity.passive.Fox {
+    private Fox fox;
+
+    public DishFox(Entity entity) {
+        super(entity);
+        this.fox = (Fox)entity;
+    }
+
+    @Override
+    public String getVariant() {
+        return this.fox.getVariant().getSerializedName();
+    }
+}
