diff --git a/src/main/java/net/ouja/dish/entity/passive/DishCat.java b/src/main/java/net/ouja/dish/entity/passive/DishCat.java
new file mode 100644
index 0000000..890e64d
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/passive/DishCat.java
@@ -0,0 +1,34 @@
+package net.ouja.dish.entity.passive;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.animal.Cat;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishCat extends DishEntity implements net.ouja.api.entity.passive.Cat {
+    private Cat cat;
+
+    public DishCat(Entity entity) {
+        super(entity);
+        this.cat = (net.minecraft.world.entity.animal.Cat)entity;
+    }
+
+    @Override
+    public String getBreed() {
+        return cat.getVariant().toString();
+    }
+
+    @Override
+    public boolean isLaying() {
+        return cat.isLying();
+    }
+
+    @Override
+    public void hiss() {
+        cat.hiss();
+    }
+
+    @Override
+    public void setLaying(boolean b) {
+        cat.setLying(b);
+    }
+}
