diff --git a/src/main/java/net/ouja/dish/entity/passive/DishParrot.java b/src/main/java/net/ouja/dish/entity/passive/DishParrot.java
new file mode 100644
index 0000000..8fa208c
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/passive/DishParrot.java
@@ -0,0 +1,20 @@
+package net.ouja.dish.entity.passive;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.animal.Parrot;
+import net.ouja.api.entity.passive.ParrotTypes;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishParrot extends DishEntity implements net.ouja.api.entity.passive.Parrot {
+    private Parrot parrot;
+
+    public DishParrot(Entity entity) {
+        super(entity);
+        this.parrot = (Parrot)entity;
+    }
+
+    @Override
+    public ParrotTypes getVariant() {
+        return ParrotTypes.valueOf(this.parrot.getVariant().name());
+    }
+}
