diff --git a/src/main/java/net/ouja/dish/entity/passive/DishStrider.java b/src/main/java/net/ouja/dish/entity/passive/DishStrider.java
new file mode 100644
index 0000000..4f77ec7
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/passive/DishStrider.java
@@ -0,0 +1,30 @@
+package net.ouja.dish.entity.passive;
+
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.monster.Strider;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishStrider extends DishEntity implements net.ouja.api.entity.passive.Strider {
+    private Strider strider;
+
+    public DishStrider(Entity entity) {
+        super(entity);
+        this.strider = (Strider)entity;
+    }
+
+    @Override
+    public boolean canSaddle() {
+        return this.strider.isSaddleable();
+    }
+
+    @Override
+    public boolean isSaddled() {
+        return this.strider.isSaddled();
+    }
+
+    @Override
+    public void equipSaddle() {
+        this.strider.equipSaddle(SoundSource.BLOCKS);
+    }
+}
