diff --git a/src/main/java/net/ouja/dish/entity/passive/DishWanderingTrader.java b/src/main/java/net/ouja/dish/entity/passive/DishWanderingTrader.java
new file mode 100644
index 0000000..3fd0ff1
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/passive/DishWanderingTrader.java
@@ -0,0 +1,14 @@
+package net.ouja.dish.entity.passive;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.npc.WanderingTrader;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishWanderingTrader extends DishEntity implements net.ouja.api.entity.passive.WanderingTrader {
+    private WanderingTrader wanderingTrader;
+
+    public DishWanderingTrader(Entity entity) {
+        super(entity);
+        this.wanderingTrader = (WanderingTrader)entity;
+    }
+}
