diff --git a/src/main/java/net/ouja/dish/entity/passive/DishVillager.java b/src/main/java/net/ouja/dish/entity/passive/DishVillager.java
new file mode 100644
index 0000000..bb88906
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/passive/DishVillager.java
@@ -0,0 +1,14 @@
+package net.ouja.dish.entity.passive;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.npc.Villager;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishVillager extends DishEntity implements net.ouja.api.entity.passive.Villager {
+    private Villager villager;
+
+    public DishVillager(Entity entity) {
+        super(entity);
+        this.villager = (Villager)entity;
+    }
+}
