diff --git a/src/main/java/net/ouja/dish/entity/passive/DishChicken.java b/src/main/java/net/ouja/dish/entity/passive/DishChicken.java
new file mode 100644
index 0000000..f02ae62
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/passive/DishChicken.java
@@ -0,0 +1,14 @@
+package net.ouja.dish.entity.passive;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.animal.Chicken;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishChicken extends DishEntity implements net.ouja.api.entity.passive.Chicken {
+    private Chicken chicken;
+
+    public DishChicken(Entity entity) {
+        super(entity);
+        this.chicken = (Chicken)entity;
+    }
+}
