diff --git a/src/main/java/net/ouja/dish/entity/passive/DishCow.java b/src/main/java/net/ouja/dish/entity/passive/DishCow.java
new file mode 100644
index 0000000..7505b1d
--- /dev/null
+++ b/src/main/java/net/ouja/dish/entity/passive/DishCow.java
@@ -0,0 +1,14 @@
+package net.ouja.dish.entity.passive;
+
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.animal.Cow;
+import net.ouja.dish.entity.DishEntity;
+
+public class DishCow extends DishEntity implements net.ouja.api.entity.passive.Cow {
+    private Cow cow;
+
+    public DishCow(Entity entity) {
+        super(entity);
+        this.cow = (Cow)entity;
+    }
+}
