diff --git a/minecraft/nbt/NbtOps.java b/minecraft/nbt/NbtOps.java
index 05decde..b96febd 100644
--- a/minecraft/nbt/NbtOps.java
+++ b/minecraft/nbt/NbtOps.java
@@ -344,7 +344,7 @@ public class NbtOps implements DynamicOps<Tag> {
 
             return DataResult.success(ByteBuffer.wrap(bytearraytag.getAsByteArray()));
         } else {
-            return super.getByteBuffer(tag);
+            return DynamicOps.super.getByteBuffer(tag);
         }
     }
 
@@ -358,7 +358,7 @@ public class NbtOps implements DynamicOps<Tag> {
 
             return DataResult.success(Arrays.stream(intarraytag.getAsIntArray()));
         } else {
-            return super.getIntStream(tag);
+            return DynamicOps.super.getIntStream(tag);
         }
     }
 
@@ -372,7 +372,7 @@ public class NbtOps implements DynamicOps<Tag> {
 
             return DataResult.success(Arrays.stream(longarraytag.getAsLongArray()));
         } else {
-            return super.getLongStream(tag);
+            return DynamicOps.super.getLongStream(tag);
         }
     }
 
