diff --git a/minecraft/world/level/dimension/end/DragonRespawnAnimation.java b/minecraft/world/level/dimension/end/DragonRespawnAnimation.java
index ccd3747..3d02b92 100644
--- a/minecraft/world/level/dimension/end/DragonRespawnAnimation.java
+++ b/minecraft/world/level/dimension/end/DragonRespawnAnimation.java
@@ -27,7 +27,7 @@ public enum DragonRespawnAnimation {
                 endcrystal.setBeamTarget(blockpos1);
             }
 
-            enddragonfight.setRespawnStage(null.PREPARING_TO_SUMMON_PILLARS);
+            enddragonfight.setRespawnStage(PREPARING_TO_SUMMON_PILLARS);
         }
     },
     PREPARING_TO_SUMMON_PILLARS {
@@ -38,7 +38,7 @@ public enum DragonRespawnAnimation {
                     serverlevel.levelEvent(3001, new BlockPos(0, 128, 0), 0);
                 }
             } else {
-                enddragonfight.setRespawnStage(null.SUMMONING_PILLARS);
+                enddragonfight.setRespawnStage(SUMMONING_PILLARS);
             }
 
         }
@@ -81,7 +81,7 @@ public enum DragonRespawnAnimation {
                         Feature.END_SPIKE.place(spikeconfiguration, serverlevel, serverlevel.getChunkSource().getGenerator(), RandomSource.create(), new BlockPos(spikefeature_endspike.getCenterX(), 45, spikefeature_endspike.getCenterZ()));
                     }
                 } else if (flag1) {
-                    enddragonfight.setRespawnStage(null.SUMMONING_DRAGON);
+                    enddragonfight.setRespawnStage(SUMMONING_DRAGON);
                 }
             }
 
@@ -94,7 +94,7 @@ public enum DragonRespawnAnimation {
             EndCrystal endcrystal;
 
             if (i >= 100) {
-                enddragonfight.setRespawnStage(null.END);
+                enddragonfight.setRespawnStage(END);
                 enddragonfight.resetSpikeCrystals();
                 iterator = list.iterator();
 
