diff --git a/minecraft/world/ticks/ScheduledTick.java b/minecraft/world/ticks/ScheduledTick.java
index 1cdc451..c0708e0 100644
--- a/minecraft/world/ticks/ScheduledTick.java
+++ b/minecraft/world/ticks/ScheduledTick.java
@@ -36,13 +36,13 @@ public record ScheduledTick<T> (T type, BlockPos pos, long triggerTick, TickPrio
         this(t0, blockpos, i, TickPriority.NORMAL, j);
     }
 
-    public ScheduledTick(T t0, BlockPos blockpos, long i, TickPriority tickpriority, long j) {
-        blockpos = blockpos.immutable();
-        this.type = t0;
-        this.pos = blockpos;
-        this.triggerTick = i;
-        this.priority = tickpriority;
-        this.subTickOrder = j;
+    public ScheduledTick(T type, BlockPos pos, long triggerTick, TickPriority priority, long subTickOrder) {
+        pos = pos.immutable();
+        this.type = type;
+        this.pos = pos;
+        this.triggerTick = triggerTick;
+        this.priority = priority;
+        this.subTickOrder = subTickOrder;
     }
 
     public static <T> ScheduledTick<T> probe(T t0, BlockPos blockpos) {
